{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "#from mimseq import version\n",
    "#from mimseq.tRNAtools import modsToSNPIndex, generateGSNAPIndices, newModsParser, tidyFiles\n",
    "#from mimseq.tRNAmap import mainAlign\n",
    "#from mimseq.getCoverage import getCoverage, plotCoverage\n",
    "#from mimseq.mmQuant import generateModsTable, plotCCA\n",
    "#from mimseq.ssAlign import structureParser, modContext \n",
    "#from mimseq.splitClusters import splitIsodecoder, unsplitClusters, getIsodecoderSizes, writeIsodecoderTranscripts\n",
    "import sys, os, subprocess, logging, datetime, copy\n",
    "import argparse\n",
    "from pyfiglet import figlet_format\n",
    "from collections import defaultdict\n",
    "\n",
    "log = logging.getLogger(__name__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def restrictedFloat(x):\n",
    "## Method for restricting cluster_id and cov_diff argument to float between 0 and 1\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0 or x > 1.0:\n",
    "            raise argparse.ArgumentTypeError('{} not in range 0.0 - 1.0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "def restrictedFloat2(x):\n",
    "## Method for restricting min-cov argument to float between 0 and 1, or int greater than 1\n",
    "\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0:\n",
    "            raise argparse.ArgumentTypeError('{} not greater than 0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "\n",
    "\n",
    "parser = argparse.ArgumentParser(description = 'Custom high-throughput tRNA sequencing alignment and quantification pipeline\\\n",
    "    based on modification induced misincorporation cDNA synthesis.', add_help = True, usage = \"%(prog)s [options] sample data\")\n",
    "\n",
    "inputs = parser.add_argument_group(\"Input files\")\n",
    "inputs.add_argument('-s','--species', metavar='species', required = not ('-t' in sys.argv), dest = 'species', help = \\\n",
    "    'Species being analyzed for which to load pre-packaged data files (prioritized over -t, -o and -m). Options are: Hsap, Hsap38, Mmus, Scer, Spom, Dmel, Drer, Ecol', \\\n",
    "    choices = ['Hsap','Hsap38','Mmus','Scer','Spom','Dmel', 'Drer', 'Ecol'])\n",
    "inputs.add_argument('-t', '--trnas', metavar='genomic tRNAs', required = False, dest = 'trnas', help = \\\n",
    "    'Genomic tRNA fasta file, e.g. from gtRNAdb or tRNAscan-SE. Already avalable in data folder for a few model organisms.')\n",
    "inputs.add_argument('-o', '--trnaout', metavar = 'tRNA out file', required = (not '--species' or '-s' in sys.argv) or ('-t' in sys.argv), \n",
    "    dest = 'trnaout', help = 'tRNA.out file generated by tRNAscan-SE (also may be available on gtRNAdb). Contains information about tRNA features, including introns.')\n",
    "inputs.add_argument('-m', '--mito-trnas', metavar = 'mitochondrial tRNAs', required = False, dest = 'mito', \\\n",
    "    help = 'Mitochondrial tRNA fasta file. Should be downloaded from mitotRNAdb for species of interest. Already available in data folder for a few model organisms.')\n",
    "\n",
    "options = parser.add_argument_group(\"Program options\")\n",
    "options.add_argument('--pretRNAs', required = False, dest = 'pretrnas', action = 'store_true',\\\n",
    "    help = \"Input reference sequences are pretRNAs. Enabling this option will disable the removal of intron sequences and addition of 3'-CCA to generate \\\n",
    "    mature tRNA sequences. Useful for mapping and discovering pretRNA sequence reads.\")\n",
    "options.add_argument('--no-cluster', required = False, dest = 'cluster', action = 'store_false',\\\n",
    "    help = 'Disable usearch sequence clustering of tRNAs by isodecoder which drastically reduces the rate of multi-mapping reads. Default is enabled.')\n",
    "options.add_argument('--cluster-id', metavar = 'clustering identity threshold', dest = 'cluster_id', type = restrictedFloat, nargs = '?', default = 0.97,\\\n",
    "    required = False, help = 'Identity cutoff for usearch clustering between 0 and 1. Default is 0.97.')\n",
    "options.add_argument('--deconv-cov-ratio', metavar='deconvolution coverage threshold', dest='cov_diff', type = restrictedFloat, nargs = '?', default=0.5,\\\n",
    "    required=False, help=\"Threshold for ratio between coverage at 3' end and mismatch used for deconvolution. Coverage reductions greater than the threshold will result in non-deconvoluted sequences. \\\n",
    "        Default is 0.5 (i.e. less than 50%% reduction required for deconvolution).\")\n",
    "options.add_argument('--threads', metavar = 'thread number', required = False, dest = 'threads', type = int, \\\n",
    "    help = 'Set processor threads to use during read alignment and read counting.')\n",
    "options.add_argument('--posttrans-mod-off', required = False, dest = 'posttrans', action = 'store_true', \\\n",
    "    help = \"Disable post-transcriptional modification of tRNAs, i.e. addition of 3'-CCA and 5'-G (His) to mature sequences. Disable for certain \\\n",
    "    prokaryotes (e.g. E. coli) where this is genomically encoded. Leave enabled (default) for all eukaryotes.\")\n",
    "options.add_argument('--control-condition', metavar = 'control condition', required = True, dest = 'control_cond', \\\n",
    "    help = 'Name of control/wild-type condition as per user defined group specified in sample data input. This must exactly match the group name \\\n",
    "    specified in sample data. This is used for differential expression analysis so that results are always in the form mutant/treatment vs WT/control. REQUIRED')\n",
    "options.add_argument('--no-cca-analysis', required = False, dest = 'cca', action = 'store_false',\\\n",
    "    help = \"Disable analysis of 3'-CCA ends. When enabled, this calculates proportions of CC vs CCA ending reads per cluster and performs DESeq2 analysis. \\\n",
    "    Useful for comparing functional to non-functional mature tRNAs. Default is enabled.\")\n",
    "options.add_argument('--double-cca', required = False, dest = 'double_cca', action = \"store_true\",\\\n",
    "    help = \"Enable analysis of 3'-CCACCA tagging for tRNA degradation pathway. Note that this will alter the output of the CCA analysis pipeline.\")\n",
    "options.add_argument('--local-modomics', required=False, dest = 'local_mod', action='store_true',\\\n",
    "    help = \"Disable retrieval of Modomics data from online. Instead use older locally stored data. Warning - this leads\\\n",
    "        to usage of older Modomics data!\")\n",
    "\n",
    "align = parser.add_argument_group(\"GSNAP alignment options\")\n",
    "align.add_argument('--max-mismatches', metavar = 'allowed mismatches', required = False, dest = 'mismatches', type = float, \\\n",
    "    help = 'Maximum mismatches allowed. If specified between 0.0 and 1.0, then treated as a fraction of read length. Otherwise, treated as \\\n",
    "    integer number of mismatches. Default is an automatic ultrafast value calculated by GSNAP; see GSNAP help for more info.')\n",
    "align.add_argument('--remap-mismatches', metavar = 'allowed mismatches for remap', required = False, dest = 'remap_mismatches', type = float,\\\n",
    "    help = 'Maximum number of mismatches allowed during remapping of all reads. Treated similarly to --max-mismatches. This is important to control misalignment of reads to similar clusters/tRNAs \\\n",
    "    Note that the SNP index will be updated with new SNPs from the first round of alignment and so this should be relatively small to prohibit misalignment.')\n",
    "align.add_argument('--no-snp-tolerance', required = False, dest = 'snp_tolerance', action = 'store_false',\\\n",
    "    help = 'Disable GSNAP SNP-tolerant read alignment, where known modifications from Modomics are mapped as SNPs. Default is enabled.')\n",
    "\n",
    "\n",
    "outputs = parser.add_argument_group(\"Output options\")\n",
    "outputs.add_argument('-n', '--name', metavar = 'experiment name', required = True, dest = 'name', help = \\\n",
    "    'Name of experiment. Note, output files and indices will have this as a prefix. REQUIRED')\n",
    "outputs.add_argument('--out-dir', metavar = 'output directory', required = False, dest = 'out', help = \\\n",
    "    'Output directory. Default is current directory. Cannot be an existing directory.')\n",
    "outputs.add_argument('--keep-temp', required = False, dest='keep_temp', action = 'store_true', help = \\\n",
    "    'Keeps multi-mapping and unmapped bam files from GSNAP alignments. Default is false.')\n",
    "\n",
    "bedtools = parser.add_argument_group(\"Bedtools coverage options\")\n",
    "bedtools.add_argument('--min-cov', metavar = 'Minimum coverage per cluster', required = False, dest = 'min_cov', type = restrictedFloat2, default=0.0005, \\\n",
    "    help = \"Minimum coverage per cluster required to include this cluster in coverage plots, modification analysis, and 3'-CCA analysis. \\\n",
    "    Can be a fraction of total mapped reads between 0 and 1, or an integer of absolute coverage. Any cluster not meeting the threshold in 1 or more sample will be excluded. \\\n",
    "    Note that all clusters are included for differential expression analysis with DESeq2. Default = 0.0005 (0.05%% mapped reads).\")\n",
    "bedtools.add_argument('--max-multi', metavar = 'Bedtools coverage multithreading', required = False, dest = 'max_multi', type = int, \\\n",
    "    help = 'Maximum number of bam files to run bedtools coverage on simultaneously. Increasing this number reduces processing time\\\n",
    "    by increasing number of files processed simultaneously. However, depending on the size of the bam files to process and\\\n",
    "    available memory, too many files processed at once can cause termination of mim-tRNAseq due to insufficient memory. If\\\n",
    "    mim-tRNAseq fails during coverage calculation, lower this number. Increase at your own discretion. Default is 3.')\n",
    "\n",
    "remapping = parser.add_argument_group(\"Analysis of unannotated modifications and realignment\")\n",
    "remapping.add_argument('--remap', required = False, dest = 'remap', action = 'store_true',\\\n",
    "    help = 'Enable detection of unannotated (potential) modifications from misincorporation data. These are defined as having a total misincorporation rate\\\n",
    "    higher than the threshold set with --misinc-thresh. These modifications are then appended to already known ones, and read alignment is reperformed.\\\n",
    "    Very useful for poorly annotated species in Modomics. Due to realignment and misincorporation parsing, enabling this option slows the analysis down considerably.')\n",
    "remapping.add_argument('--misinc-thresh', metavar = 'threshold for unannotated mods', dest = 'misinc_thresh', type = restrictedFloat, nargs = '?', default = 0.1,\\\n",
    "    required = False, help = 'Threshold of total misincorporation rate at a position in a cluster used to call unannotated modifications. Value between 0 and 1, default is 0.1  (10%% misincorporation).')\n",
    "\n",
    "parser.add_argument('--version', action='version', version='%(prog)s {}'.format(version.__version__), help = 'Show version number and exit')\n",
    "parser.add_argument('sampledata', help = 'Sample data sheet in text format, tab-separated. Column 1: full path to fastq (or fastq.gz). Column 2: condition/group.')\n",
    "\n",
    "parser.set_defaults(threads=1, out=\"./\", max_multi = 3, min_cov = 0, mito = '', cov_diff = 0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "args = parser.parse_args('--species Hsap --cluster-id 0.95 --threads 1 --min-cov 2000 --max-mismatches 0.1 --control-condition HEK293T -n hg19_test --out-dir hg19_HEK239vsK562 --max-multi 1 --remap --remap-mismatches 0.075 sampleData_HEKvsK562.txt'.split())\n",
    "\n",
    "\n",
    "try:\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(__file__))\n",
    "except NameError:\n",
    "    filepath = '/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/mimseq.py'\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(filepath))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           _                 _   ____  _   _    _                   \n",
      " _ __ ___ (_)_ __ ___       | |_|  _ \\| \\ | |  / \\   ___  ___  __ _ \n",
      "| '_ ` _ \\| | '_ ` _ \\ _____| __| |_) |  \\| | / _ \\ / __|/ _ \\/ _` |\n",
      "| | | | | | | | | | | |_____| |_|  _ <| |\\  |/ ___ \\\\__ \\  __/ (_| |\n",
      "|_| |_| |_|_|_| |_| |_|      \\__|_| \\_\\_| \\_/_/   \\_\\___/\\___|\\__, |\n",
      "                                                                 |_|\n",
      "\n",
      " Modification-induced misincorporation analysis of tRNA sequencing data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(figlet_format('mim-tRNAseq', font='standard'))\n",
    "print(\" Modification-induced misincorporation analysis of tRNA sequencing data\\n\")\n",
    "if args.pretrnas:\n",
    "    if args.cca:\n",
    "        log.warning(\"Disabling CCA analysis in pre-tRNA mode...\")\n",
    "        args.cca = False\n",
    "    if args.cluster:\n",
    "        log.warning(\"Disabling tRNA clustering in pre-tRNA mode...\")\n",
    "        args.cluster = False\n",
    "# Check that control_cond exists in sample data\n",
    "conditions = list()\n",
    "with open(args.sampledata, \"r\") as sampleData:\n",
    "    for line in sampleData:\n",
    "        line = line.strip()\n",
    "        if not line.startswith(\"#\"):\n",
    "            conditions.append(line.split(\"\\t\")[1])\n",
    "if args.control_cond not in conditions:\n",
    "    raise argparse.ArgumentTypeError('{} not a valid condition in {}'.format(args.control_cond, args.sampledata))\n",
    "if not args.species and not (args.trnas or args.trnaout):\n",
    "    parser.error('Must specify valid --species argument or supply -t (tRNA sequences) and -o (tRNAscan out file)!')                        \n",
    "else:\n",
    "    if args.species:\n",
    "        if args.species == 'Hsap':\n",
    "            args.trnas = args.repo_path + \"/data/hg19-eColitK/hg19_eColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg19-eColitK/hg19_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg19-eColitK/hg19-mitotRNAs.fa\"\n",
    "        if args.species == 'Hsap38':\n",
    "            args.trnas = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-detailed.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg38-eColitK/hg38-mitotRNAs.fa\"\n",
    "        if args.species == 'Scer':\n",
    "            args.trnas = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/sacCer3-eColitK/sacCer3-mitotRNAs.fa\"\n",
    "        if args.species == 'Mmus':\n",
    "            args.trnas = args.repo_path + \"/data/mm10-eColitK/mm10_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/mm10-eColitK/mm10_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/mm10-eColitK/mm10-mitotRNAs.fa\"\n",
    "        if args.species == 'Spom':\n",
    "            args.trnas = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_972H-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/schiPomb-eColitK/schiPomb-mitotRNAs.fa\"\n",
    "        if args.species == 'Dmel':\n",
    "            args.trnas = args.repo_path + \"/data/dm6-eColitK/dm6_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/dm6-eColitK/dm6_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/dm6-eColitK/dm6-mitotRNAs.fa\"\n",
    "        if args.species == 'Drer':\n",
    "            args.trnas = args.repo_path + \"/data/danRer11-eColitK/danRer11_eColitK_filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/danRer11-eColitK/danRer11_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/danRer11-eColitK/danRer11-mitotRNAs.fa\"\n",
    "        if args.species == 'Ecol':\n",
    "            args.trnas = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.out\"\n",
    "            args.mito = ''\n",
    "    else:\n",
    "        args.species = args.trnas.split(\"/\")[-1].split(\".\")[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "trnas, trnaout, name, species, out, cluster, cluster_id, cov_diff, posttrans, control_cond, threads, max_multi, snp_tolerance, keep_temp, cca, double_cca, min_cov, mismatches, remap, remap_mismatches, misinc_thresh, mito_trnas, pretrnas, local_mod, sample_data = args.trnas, args.trnaout, args.name, args.species, args.out, args.cluster, args.cluster_id, args.cov_diff, args.posttrans, args.control_cond, args.threads, args.max_multi, args.snp_tolerance, args.keep_temp, args.cca, args.double_cca, args.min_cov, args.mismatches, args.remap, args.remap_mismatches, args.misinc_thresh, args.mito, args.pretrnas, args.local_mod, args.sampledata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=False, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='hg19_HEK239vsK562', posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, repo_path='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq', sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_round = 1 #first round of mapping\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def intronRemover (Intron_dict, seqIO_dict, seqIO_record, posttrans_mod_off, double_cca):\n",
    "# Use Intron_dict to find and remove introns plus add CCA and 5' G for His (if eukaryotic)\n",
    "\n",
    "    # Find a match, slice intron and add G and CCA\n",
    "    ID = re.search(\"tRNAscan-SE ID: (.*?)\\).|\\((chr.*?)-\", seqIO_dict[seqIO_record].description).groups()\n",
    "    ID = list(filter(None, ID))[0]\n",
    "    if ID in Intron_dict:\n",
    "        seq = str(seqIO_dict[seqIO_record].seq[:Intron_dict[ID]['intron_start']] + seqIO_dict[seqIO_record].seq[Intron_dict[ID]['intron_stop']:])\n",
    "    else:\n",
    "        seq = str(seqIO_dict[seqIO_record].seq)\n",
    "    if not posttrans_mod_off:\n",
    "        if double_cca:\n",
    "            seq = seq + 'CCACCA'\n",
    "        else:\n",
    "            seq = seq + 'CCA'\n",
    "        if 'His' in seqIO_record:\n",
    "            seq = 'G' + seq\n",
    "\n",
    "    return(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getModomics(local_mod):\n",
    "    # Get full Modomics modified tRNA data from web\n",
    "    fetch = False\n",
    "    if not local_mod:\n",
    "        try:\n",
    "            response = requests.get(\"http://www.genesilico.pl/modomics/api/sequences?&RNAtype=tRNA\")\n",
    "            response.raise_for_status()\n",
    "            modomics = response.json()\n",
    "            fetch = True\n",
    "            log.info(\"Modomics retrieved...\")\n",
    "        except HTTPError as http_err:\n",
    "            log.error(\"Unable to connect to Modomics database! HTTP error: {}. Check status of Modomics webpage. Using local Modomics files...\".format(http_err))\n",
    "            modomics_path = os.path.dirname(os.path.realpath(__file__)) + '/data/modomics'\n",
    "            modomics = open(modomics_path, \"r+\", encoding = \"utf-8\")\n",
    "        except Exception as err:\n",
    "            log.error(\"Error in connecting to Modomics: {}. Using local Modomics files...\".format(err))\n",
    "            modomics_path = os.path.dirname(os.path.realpath(__file__)) + '/data/modomics'\n",
    "            modomics = open(modomics_path, \"r+\", encoding = \"utf-8\")\n",
    "    else:\n",
    "        log.warning(\"Retrieval of Modomics database disabled. Using local files instead...\")\n",
    "        modomics_path = os.path.dirname(os.path.realpath(__file__)) + '/data/modomics'\n",
    "        modomics = open(modomics_path, \"r+\", encoding = \"utf-8\")\n",
    "\n",
    "    return(modomics, fetch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getUnmodSeq(seq, modification_table):\n",
    "# Change modified bases into standard ACGT in input sequence\n",
    "\n",
    "    new_seq = []\n",
    "    for char in seq:\n",
    "        # for insertions ('_') make reference N - this is not described in the modifications table\n",
    "        if char == '_':\n",
    "            char = 'N'\n",
    "        else:\n",
    "            char = modification_table[char]['ref']\n",
    "            # Change queuosine to G (reference is preQ0base in modification file)\n",
    "            if char == 'preQ0base':\n",
    "                char = 'G'\n",
    "\n",
    "        new_seq.append(char)\n",
    "\n",
    "    new_seq = ''.join(new_seq)\n",
    "    new_seq = new_seq.replace('U', 'T')\n",
    "    return(new_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def processModomics(modomics_file, fetch, species, modifications):\n",
    "\n",
    "    modomics_dict = dict()\n",
    "    perSpecies_count = defaultdict(int)\n",
    "\n",
    "    # build modomics_dict from JSON data through API\n",
    "    if fetch:\n",
    "        log.info(\"Parsing Modomics JSON data...\")\n",
    "        for data in modomics_file.values():\n",
    "            sameIDcount = 0\n",
    "\n",
    "            mod_species = data['organism']\n",
    "            if not mod_species in species:\n",
    "                continue\n",
    "            else:\n",
    "                perSpecies_count[mod_species] += 1\n",
    "                anticodon = data['anticodon']\n",
    "                new_anticodon = getUnmodSeq(anticodon, modifications)\n",
    "                if \"N\" in new_anticodon:\n",
    "                    continue\n",
    "                \n",
    "                # Check amino acid name in modomics - set to iMet if equal to Ini to match gtRNAdb\n",
    "                amino = data['subtype']\n",
    "                if amino == 'Ini':\n",
    "                    amino = 'iMet'\n",
    "\n",
    "                curr_id = str(mod_species.replace(' ','_') + '_tRNA-' + amino + '-' + new_anticodon)\n",
    "                #Unique names for duplicates\n",
    "                while curr_id in modomics_dict:\n",
    "                    sameIDcount += 1\n",
    "                    curr_id = \"-\".join(curr_id.split('-')[0:3]) + '-' + str(sameIDcount)\n",
    "\n",
    "                tRNA_type = data['organellum']\n",
    "                tRNA_type = \"cytosolic\" if re.search(\"cytosol\",tRNA_type) else tRNA_type\n",
    "                \n",
    "                sequence = data['seq'].replace('U','T').replace('-','')\n",
    "                unmod_sequence = getUnmodSeq(sequence, modifications)\n",
    "                # Return list of modified nucl and inosines indices and add to modomics_dict\n",
    "                # add unmodified seq to modomics_dict by lookup to modifications\n",
    "                Mods = list('''\"KR'OYW⊆X*[''')\n",
    "                modPos = [i for i, x in enumerate(sequence) if x in Mods]\n",
    "                inosinePos = [i for i, x in enumerate(sequence) if x == 'I']\n",
    "                \n",
    "                modomics_dict[curr_id] = {'sequence':sequence, 'type':tRNA_type, 'anticodon':new_anticodon, 'modified':modPos, 'unmod_sequence':unmod_sequence, 'InosinePos':inosinePos}\n",
    "\n",
    "    # build modomics_dict if data was not fetched from API - i.e. using local txt version\n",
    "    elif not fetch:\n",
    "        log.info(\"Parsing local Modomics data...\")\n",
    "        for line in modomics_file:\n",
    "            line = line.strip()\n",
    "            sameIDcount = 0\n",
    "\n",
    "            if line.startswith('>'):\n",
    "                line = line.replace(' | ','|')\n",
    "                # Check if species matches those from gtRNAdb input, otherwise skip entry\n",
    "                mod_species = line.split('|')[3]\n",
    "                if not mod_species in species:\n",
    "                    continue\n",
    "                else:\n",
    "                    # Add to perSpecies_count to tally total modomics entries per species of interest\n",
    "                    perSpecies_count[mod_species] += 1\n",
    "                    # Replace modomics antidon with normal ACGT codon, keep \".\" for pattern matching\n",
    "                    anticodon = str(line.split('|')[2])\n",
    "                    new_anticodon = getUnmodSeq(anticodon, modifications)\n",
    "                    if \"N\" in new_anticodon:\n",
    "                        continue\n",
    "                    #new_anticodon = new_anticodon.replace(\"N\", \".\")\n",
    "\n",
    "                    # Check amino acid name in modomics - set to iMet if equal to Ini to match gtRNAdb\n",
    "                    amino = str(line.split('|')[1])\n",
    "                    if amino == 'Ini' :\n",
    "                        amino = 'iMet'\n",
    "\n",
    "                    curr_id = str(line.split('|')[3].split(' ')[0]) + '_' + str(line.split('|')[3].split(' ')[1]) + '_' + str(line.split('|')[0].split(' ')[1]) + '-' + amino + '-' + new_anticodon\n",
    "                    #Unique names for duplicates\n",
    "                    while curr_id in modomics_dict:\n",
    "                        sameIDcount += 1\n",
    "                        curr_id = \"-\".join(curr_id.split('-')[0:3]) + '-' + str(sameIDcount)\n",
    "\n",
    "                    tRNA_type = str(line.split('|')[4])\n",
    "                    tRNA_type = \"cytosolic\" if re.search(\"cytosol\",tRNA_type) else tRNA_type\n",
    "                    modomics_dict[curr_id] = {'sequence':'','type':tRNA_type, 'anticodon':new_anticodon}\n",
    "            \n",
    "            else:\n",
    "                if not mod_species in species:\n",
    "                    continue\n",
    "                else:\n",
    "                    if \"N\" in new_anticodon:\n",
    "                        continue\n",
    "\n",
    "                    sequence = line.strip().replace('U','T').replace('-','')\n",
    "                    modomics_dict[curr_id]['sequence'] = sequence\n",
    "                    unmod_sequence = getUnmodSeq(sequence, modifications)\n",
    "\n",
    "                    # Return list of modified nucl indices and add to modomics_dict\n",
    "                    # add unmodified seq to modomics_dict by lookup to modifications\n",
    "                    Mods = list('''\"KR'OYW⊆X*[''')\n",
    "                    modPos = [i for i, x in enumerate(modomics_dict[curr_id]['sequence']) if x in Mods]\n",
    "                    inosinePos = [i for i, x in enumerate(modomics_dict[curr_id]['sequence']) if x == 'I']\n",
    "                    modomics_dict[curr_id]['modified'] = modPos\n",
    "                    modomics_dict[curr_id]['unmod_sequence'] = unmod_sequence\n",
    "                    modomics_dict[curr_id][\"InosinePos\"] = inosinePos\n",
    "        \n",
    "    return(modomics_dict, perSpecies_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tRNAparser (gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, posttrans_mod_off, double_cca, pretrnas, local_mod):\n",
    "# tRNA sequence files parser and dictionary building\n",
    "\n",
    "    # Generate modification reference table\n",
    "    modifications = modificationParser(modifications_table)\n",
    "    temp_name = gtRNAdb.split(\"/\")[-1]\n",
    "                \n",
    "    log.info(\"\\n+\" + (\"-\" * (len(temp_name)+24)) + \"+\\\n",
    "        \\n| Starting analysis for {} |\\\n",
    "        \\n+\".format(temp_name) + (\"-\" * (len(temp_name)+24)) + \"+\")\n",
    "          \n",
    "    log.info(\"Processing tRNA sequences...\")\n",
    "\n",
    "    # Initialise intron dictionary\n",
    "    Intron_dict = initIntronDict(tRNAscan_out)\n",
    "\n",
    "    # Build dictionary of sequences from gtRNAdb fasta\n",
    "    tRNA_dict = defaultdict(dd)\n",
    "    temp_dict = SeqIO.to_dict(SeqIO.parse(gtRNAdb, \"fasta\"))\n",
    "    \n",
    "    species = set()\n",
    "    for seq in temp_dict:\n",
    "        # Get species of input tRNA seqs to subset full Modomics table\n",
    "        species.add(' '.join(seq.split('_')[0:2]))\n",
    "        # only add to dictionary if not nmt or undetermined sequence\n",
    "        if not (re.search('Und', seq) or re.search('nmt', seq)):\n",
    "            if not pretrnas:\n",
    "                tRNAseq = intronRemover(Intron_dict, temp_dict, seq, posttrans_mod_off, double_cca)\n",
    "            else:\n",
    "                tRNAseq = str(temp_dict[seq].seq)\n",
    "            tRNA_dict[seq]['sequence'] = tRNAseq\n",
    "            tRNA_dict[seq]['species'] = ' '.join(seq.split('_')[0:2])\n",
    "            tRNA_dict[seq]['type'] = \"cytosolic\"\n",
    "\n",
    "    # add mitochondrial tRNAs if given\n",
    "    if mitotRNAs:\n",
    "        temp_dict = SeqIO.to_dict(SeqIO.parse(mitotRNAs, \"fasta\"))\n",
    "        mito_count = defaultdict(int)\n",
    "        # read each mito tRNA, edit sequence header to match nuclear genes as above and add to tRNA_dict\n",
    "        for seq in temp_dict:\n",
    "            seq_parts = seq.split(\"|\")\n",
    "            anticodon = seq_parts[4]\n",
    "            amino = re.search(\"[a-zA-z]+\", seq_parts[3]).group(0)\n",
    "            mito_count[anticodon] += 1\n",
    "            new_seq = seq_parts[1] + \"_mito_tRNA-\" + amino + \"-\" + seq_parts[4] + \"-\" + str(mito_count[anticodon]) + \"-1\"\n",
    "            tRNAseq = str(temp_dict[seq].seq) + \"CCA\" if not double_cca else str(temp_dict[seq].seq) + \"CCACCA\"\n",
    "            tRNA_dict[new_seq]['sequence'] = tRNAseq\n",
    "            tRNA_dict[new_seq]['type'] = 'mitochondrial'\n",
    "            tRNA_dict[new_seq]['species'] = ' '.join(seq.split('_')[0:2])\n",
    "\n",
    "        num_cytosilic = len([k for k in tRNA_dict.keys() if tRNA_dict[k]['type'] == \"cytosolic\"])\n",
    "        num_mito = len([k for k in tRNA_dict.keys() if tRNA_dict[k]['type'] == \"mitochondrial\"])\n",
    "\n",
    "        log.info(\"{} cytosolic and {} mitochondrial tRNA sequences imported\".format(num_cytosilic, num_mito))\n",
    "\n",
    "    # Read in and parse modomics file to contain similar headers to tRNA_dict\n",
    "    # Save in new dict\n",
    "\n",
    "    log.info(\"Processing modomics database...\")\n",
    "    modomics_file, fetch = getModomics(local_mod)\n",
    "    modomics_dict, perSpecies_count = processModomics(modomics_file, fetch, species, modifications)\n",
    "\n",
    "    for s in species:\n",
    "        log.info('Number of Modomics entries for {}: {}'.format(s, perSpecies_count[s]))\n",
    "\n",
    "    return(tRNA_dict, modomics_dict, species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq'"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args.repo_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Here make tmp folder, start logging etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "import re, copy, sys, os, shutil, subprocess, logging, glob\n",
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "\n",
    "#log = logging.getLogger(__name__)\n",
    "\n",
    "class TRNA():\n",
    "    '''Object for tRNAs, their sequences, modifications and other information.'''\n",
    "    \n",
    "    def __init__(self, cmd_args):\n",
    "        self.inp_repo_path = cmd_args.repo_path\n",
    "        self.inp_gtRNAdb = cmd_args.trnas\n",
    "        self.inp_tRNAscan_out = cmd_args.trnaout\n",
    "        self.inp_mitotRNAs = cmd_args.mito\n",
    "        self.inp_ptm_off = cmd_args.posttrans\n",
    "        self.inp_double_cca = cmd_args.double_cca\n",
    "        self.inp_pretrnas = cmd_args.pretrnas\n",
    "        self.inp_local_mod = cmd_args.local_mod\n",
    "        \n",
    "        # Read intron boundaries into dict\n",
    "        self.intron_dict = self.read_introns(cmd_args.trnaout)\n",
    "        # Read tRNAs, minus introns, into dict\n",
    "        self.tRNA_dict = self.read_tRNAs(cmd_args.trnas, cmd_args.mito, cmd_args.posttrans, cmd_args.double_cca, cmd_args.pretrnas)\n",
    "        # Read table of tRNA modifications into dict\n",
    "        self.mod_dict = self.mod_parser(cmd_args.repo_path)\n",
    "        \n",
    "        \n",
    "\n",
    "    # Private methods for default dict.\n",
    "    # Used to create nested defaultdicts.\n",
    "    # Can be done with lambda functions (e.g. tRNA_dict = defaultdict(lambda: defaultdict()))\n",
    "    # But lambda functions cannot be pickled, and pickling is required for parallelization with multiprocessing.\n",
    "    def __dd(self):\n",
    "        return(defaultdict())\n",
    "    def __dd_list(self):\n",
    "        return(defaultdict(list))\n",
    "\n",
    "    def mod_parser(self, repo_path):\n",
    "        '''Parse tRNA and modifications.'''\n",
    "        mod_path = Path(repo_path + '/modifications.txt')\n",
    "        try:\n",
    "            _ = mod_path.resolve(strict=True)\n",
    "        except FileNotFoundError:\n",
    "            print('Expected modifications file but found none at this location: {}'.format(mod_path))\n",
    "        \n",
    "        # Read modifications and build dict\n",
    "        with open(mod_path, 'r', encoding='utf-8') as mod_file:\n",
    "            mod_dict = dict()\n",
    "            for line in mod_file:\n",
    "                if not line.startswith(\"#\"):\n",
    "                    name, abbr, ref, mod = list(map(str.strip, line.split('\\t')))\n",
    "                    # Replace unknown modifications with reference of N\n",
    "                    if not ref:\n",
    "                        ref = 'N'\n",
    "                    if mod:\n",
    "                        mod_dict[mod] = {'name':name, 'abbr':abbr, 'ref':ref}\n",
    "        return(mod_dict)\n",
    "\n",
    "    def read_introns(self, trnaout_path):\n",
    "        '''Build dictionary of intron locations.'''\n",
    "        intron_dict = {}\n",
    "        with open(trnaout_path, 'r') as trnaout_file:\n",
    "            for line in trnaout_file:\n",
    "                cols = line.split()\n",
    "                if line.startswith(\"chr\"):\n",
    "                    tRNA_ID = cols[0] + \".trna\" + cols[1]\n",
    "                    tRNA_start = int(cols[2])\n",
    "                    intron_start = int(cols[6])\n",
    "                    intron_stop = int(cols[7])\n",
    "                    # If intron boundaries are not 0, i.e. there is an intron then add to dict\n",
    "                    if (intron_start > 0) & (intron_stop > 0):\n",
    "                        if tRNA_start > intron_start: # reverse strand\n",
    "                            intron_start = tRNA_start - intron_start\n",
    "                            intron_stop = tRNA_start - intron_stop + 1 # python 0 indexing\n",
    "                        else: # forward strand\n",
    "                            intron_start = intron_start - tRNA_start\n",
    "                            intron_stop = intron_stop - tRNA_start + 1 # python 0 indexing\n",
    "                        intron_dict[tRNA_ID] = {'intron_start': intron_start, 'intron_stop': intron_stop}\n",
    "\n",
    "            # log.info(\"{} introns registered...\".format(len(intron_dict)))\n",
    "            print(\"{} introns registered...\".format(len(intron_dict)))\n",
    "            return(intron_dict)\n",
    "\n",
    "\n",
    "    def read_tRNAs(self, trna_path, mitotrna_path, ptm_off, double_cca, pretrnas):\n",
    "        '''Read tRNA sequences from input files, remove introns and add CCA.'''\n",
    "        def remove_intron (intron_dict, seq_id, seq_obj, ptm_off, double_cca):\n",
    "            '''Use intron_dict to remove introns plus add CCA and 5' G for His (if eukaryotic).'''\n",
    "            # Find a match, slice intron and add G and CCA\n",
    "            ID = re.search(\"tRNAscan-SE ID: (.*?)\\).|\\((chr.*?)-\", seq_obj.description).groups()\n",
    "            ID = list(filter(None, ID))[0]\n",
    "            if ID in intron_dict:\n",
    "                seq = str(seq_obj.seq[:intron_dict[ID]['intron_start']] + seq_obj.seq[intron_dict[ID]['intron_stop']:])\n",
    "            else:\n",
    "                seq = str(seq_obj.seq)\n",
    "            if not ptm_off:\n",
    "                if double_cca:\n",
    "                    seq = seq + 'CCACCA'\n",
    "                else:\n",
    "                    seq = seq + 'CCA'\n",
    "                if '-His-' in seq_id:\n",
    "                    seq = 'G' + seq\n",
    "            return(seq)\n",
    "\n",
    "        # Build dict of sequences from input tRNA fasta\n",
    "        # Start with cyto\n",
    "        tRNA_dict = defaultdict(self.__dd)\n",
    "        intron_count = 0\n",
    "        for seq_id, seq_obj in SeqIO.to_dict(SeqIO.parse(trna_path, \"fasta\")).items():\n",
    "            # Only add to dict if not undetermined sequence (Und)\n",
    "            # or nuclear-encoded mitochondrial tRNA (mnt)\n",
    "            if not (re.search('Und', seq_id) or re.search('nmt', seq_id)):\n",
    "                if not pretrnas:\n",
    "                    tRNAseq = remove_intron(self.intron_dict, seq_id, seq_obj, ptm_off, double_cca)\n",
    "                    intron_count += 1 if len(tRNAseq) < len(seq_obj.seq) else 0\n",
    "                else:\n",
    "                    tRNAseq = str(seq_obj.seq)\n",
    "                tRNA_dict[seq_id]['sequence'] = tRNAseq\n",
    "                tRNA_dict[seq_id]['species'] = ' '.join(seq_id.split('_')[0:2])\n",
    "                tRNA_dict[seq_id]['type'] = 'cyto'\n",
    "\n",
    "        # Then add mito (if given)\n",
    "        if mitotrna_path:\n",
    "            mito_count = defaultdict(int) # for unique tRNA naming\n",
    "            # Read each mito tRNA, edit sequence header to match nuclear genes as above and add to tRNA_dict\n",
    "            for seq_id, seq_obj in SeqIO.to_dict(SeqIO.parse(mitotrna_path, 'fasta')).items():\n",
    "                id_parts = seq_id.split(\"|\")\n",
    "                anticodon = id_parts[4]\n",
    "                amino_acid = id_parts[3][0:3] # e.g. extract Leu from Leu1/Leu2\n",
    "                mito_count[anticodon] += 1\n",
    "                new_id = id_parts[1] + \"_mito_tRNA-\" + amino_acid + \"-\" + id_parts[4] + \"-\" + str(mito_count[anticodon]) + \"-1\"\n",
    "                if double_cca:\n",
    "                    tRNAseq = str(seq_obj.seq) + \"CCACCA\"\n",
    "                else:\n",
    "                    tRNAseq = str(seq_obj.seq) + \"CCA\"\n",
    "                tRNA_dict[new_id]['sequence'] = tRNAseq\n",
    "                tRNA_dict[new_id]['type'] = 'mito'\n",
    "                tRNA_dict[new_id]['species'] = ' '.join(id_parts[1].split('_')[0:2])\n",
    "\n",
    "            num_cyto = sum(1 for val in tRNA_dict.values() if val['type'] == 'cyto')\n",
    "            num_mito = sum(1 for val in tRNA_dict.values() if val['type'] == 'mito')\n",
    "            # log.info(\"Removed {} introns (including for duplicate tRNAs)\".format(intron_count))\n",
    "            # log.info(\"{} cytosolic and {} mitochondrial tRNA sequences imported\".format(num_cytosilic, num_mito))\n",
    "            print(\"Removed {} introns (including for duplicate tRNAs)\".format(intron_count))\n",
    "            print(\"{} cytosolic and {} mitochondrial tRNA sequences imported\".format(num_cyto, num_mito))\n",
    "            return(tRNA_dict)\n",
    "\n",
    "\n",
    "\n",
    "#            # Get species of input tRNA seqs to subset full Modomics table\n",
    "#            species_set = {val['species'] for val in tRNA_dict.values()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34 introns registered...\n",
      "Removed 33 introns (including for duplicate tRNAs)\n",
      "574 cytosolic and 22 mitochondrial tRNA sequences imported\n"
     ]
    }
   ],
   "source": [
    "trna_obj = TRNA(args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(<bound method TRNA.__dd of <__main__.TRNA object at 0x146b69650>>,\n",
       "            {'Escherichia_coli_str_K_12_substr_MG1655_tRNA-eColiLys-TTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGTCGTTAGCTCAGTTGGTAGAGCAGTTGACTTTTAATCAATTGGtCGCAGGTTCGAATCCTGCACGACCCACCA',\n",
       "                          'species': 'Escherichia coli',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGtCCTGGGTTCGATCCCCAGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCACGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGATGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-12-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-12-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-12-3': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-13-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-14-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCAATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-15-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGTGGGATCGATGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-16-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCCCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-17-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCGCAAGTGGTAGAGTGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-18-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTGGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGCAGGATCGACGCCTGCACTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-18-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTGGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGCAGGATCGACGCCTGCACTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-19-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGATTAGCTCAAGCGGTAGAGCGCCTGCTTAGCATGCAAGAGGtAGCAGGATCGATGCCTGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCCGGGTTCAATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCCGGGTTCAATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-20-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGATCAGTGGTAGGGCGCACGCTTAGCATGCATGAGGcCCTGGGTCAATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-21-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGATTAGCTCAAGCGGTAGGGTGCCTGCTTAGCATGCAAGAGGtAGCAGGATCGACGCCTGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-22-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAAGCGGTAGAGCGCTCCCTTAGCATGCGAGAGGtAGCGGGATCGACGCCCCCATTCTCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-23-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGAcAGAGCACGTGGTTAGCATGCGTGAGGtCCTGCGTTCAACTTCCAGTATTTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGTACGAGGtCCCGGGTTCAATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTAGCATGCATGAGGtCCCGGGTTCGATCCCCAGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCTGGGTTCAATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGCGGTAGAGCGCGTGCTTAGCATGCACGAGGtCCTGGGTTCAATCCCCAATACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCCGGGTTCAATCCCTGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGATGCCCGCATCCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-8-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGATGCCCGCATCCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAGGCGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGACGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-9-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGAATTAGCTCAGGCGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGACGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-chr6-93': defaultdict(None,\n",
       "                         {'sequence': 'GAAGAAGTAGCTCAAGTGGTCGAGTGCTCGCTTAGCATGTGAGAGGtAGTGGGATTGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTCGCATGTATGAGGtCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTCGCATGTATGAGGcCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCGCGCTTCGCATGTGTGAGGtCCCGGGTTCAATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTCGCATGTACGAGGcCCCGGGTTCGACCCCCGGCTCCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGATCAGTGGTAGAGCGCATGCTTCGCATGTACGAGGtCCCTGGTTCAATCCCTGGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-GGC-chr19-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCTTTGTGGGTtAAAGTACTTGTCTGGCAAACAGGAGAtGCTGGAtTTCAATCCCAGCAGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGtCCCGGGTTCGATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGtCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCACGTATGAGGcCCCGGGTTCAATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCTCGGGTTCGATCCCCGACACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCACATGCTTTGCATGTGTGAGGcCCCGGGTTCGATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCTCGGTTCGATCCCCGACACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCGGATGCTTTGCATGTATGAGAcTTTGGGTTGGATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GTGGATGTAGTTTAGTGGTAGAACGCGCGCTTTGCATGTATGAGGtCCCGGTTTCGATCCCTGGCGTTTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCCAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCCAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCCAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCTAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCTAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCTAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCTAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGATTCCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGATTCCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGATTCCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCCAGTGGCCTAATGGAtAAGGCATCAGCCTCCGGAGCTGGGGAtTGTGGGTTCGAGTCCCATCTGGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCCAGTGGCCTAATGGAtAAGGCACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCTGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCCAGTGGCCTAATGGAtAAGGCACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCTGGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCCGGTGGCCTAATGGAtAAGGCATTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCCGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCCAGTGGCCTAATGGAtAAGGCATTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCATCTGGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCCAGTGGCCTGATGGAtAAGGTACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTTCCACCTGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGCAGGTTCGAGTCCTGCCGCGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTCCGTGGTTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCGCGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTTCGTGGTTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTTCGTGGTTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAAAAGAtTGCAGGTTTGAGTTCTGCCACGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTCCGGGTTCGAGTCCCGGCGGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGTGGGTTCGAATCCCACCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGTGGGTTCGAGTCCCACCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGTGGGTTCGAGTCCCACCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATGGAcgAGCGCGCTGGACTTCTAATCCAGAGGtTCCGGGTTCGAGTCCCGGCAGAGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGCGGGTTCGAGTCCCTCCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-chr5-4': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGCATAGCAGAGTGGTtCAGGTTACAGGTTCTAGATGTAAACTGAGTTCAAATCCCAGTTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-ATT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTcAGAGCGTTCGGCTATTAACCGAACGGtGAGTAGTTCAAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-ATT-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTcAGAGCGTTCGGCTATTAACCGAACGGtGAGTAGTTCAAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGATCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAGGtTGGCGGTTCGAACCCACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCCGCTGTTAACCGAAAGCtTGGTGGTTCGAGCCCACCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-11-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCCGCTGTTAACCGAAAGCtTGGTGGTTCGAGCCCACCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-12-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAAGtTGGTGGTTCGAACACACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCGGCTGTTAACCATAAGGtTGGTGGTTAGAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-14-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCCGCTGTTAACCGAAAGCtTGGTGGTTTGAGCCCACCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-15-1': defaultdict(None,\n",
       "                         {'sequence': 'ATCTCCGTGGAGCAATTGGTtAGCGCGTTCGGCTGTTAACCGGAAAGtTGGTGGTTCGAGCCTACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-3': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-4': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-5': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-17-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCTGTAGCGCGATCGGTtAGCGCCTTCGGCTGTTAAACGAAAGGtTGGTGGTTCGTTCCCACCCCGGGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-18-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCGGCTGTTAACCATAAGGtTGGTGGTTACAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-19-1': defaultdict(None,\n",
       "                         {'sequence': 'ATCTCCGTGGAGCAATTGGTtAGCGCGTTCGGCCGTTAACCGGAAAGtTGGTGGTTCGAGCCTACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-6': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-21-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCACAATCGGTtAGCGCGTTCGGCTGTTAATCTAGAGGtTGGTGGTTAGAGCCCACTGAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-22-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCACAATCGGTtAGAGCGTTCGGCTGTTAATCTAAAGGtTGGTGGCTAGAACCCACTGAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCATTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCATTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGAtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACTGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATGGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCATCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGTAGTCGGTtAGCGCGTTCGGCTGTTAACCGAAAAGtTGGTGGTTCGAGCCCACCCAGGAACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAGGtTGGTGGTTCGAACCCACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACTGAAAGGtTAGTGGTTCGAGCCCACCCGGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-9-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACTGAAAGGtTAGTGGTTCGAGCCCACCCGGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-ATC-chr6-103': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGAGTGTACTTATCATGCACGAGGtCTTGGGctGATTcCCCAGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTtAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCAATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-10': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-11': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-3': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-4': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-5': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-6': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-7': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-8': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-9': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATAGTGGTgAGTGTCCCCGTCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCGTTAGTATGGTGGTgAGTATCCCTGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCAACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTCATCAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCTGGGGTTCGATTCCCTGAGGAGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'TTCTTGTTAATATAGTGGTgAGTATTCCCACCTGTCATGCGGGAGACGGGGTTCAATTCCCTGATGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTTGTTAGTATAGTGGTgAGTGTTTCTGCCTGTCATGTGGAGACTGGAGTTTGAGTCCCCAACAGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-ACA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTGTATAGCTCAGTGGTAGAGCATTTGACTACAGCTTTAAAGTCATAtGTAGcTGGGTTCAAATCCTGTTTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGCATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTTAGCGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-12-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTTAGGGGTAGAGCATTTGACTGCAGATCAAAAGGtCCCTGGTTCAAATCCAGGTGCCCCTTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-14-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAAGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-15-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCTCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-16-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCACTTGACTGCAGATCAAGAAGtCCTTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-17-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGATATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-18-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGTTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-19-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAAATCAAGAGGtCCCTGATTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-20-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGCGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-21-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCACAGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-22-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGCGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-23-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCACAGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGTTACTCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-24-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTAGGGCTCAGGGAtAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCGAATCTAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-25-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTATATCTCAGGGGGcAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTGAAATCCGGGTGCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCACTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGTGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTGTAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-9-2': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-9-3': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-9-4': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr11-21': defaultdict(None,\n",
       "                         {'sequence': 'GAAGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAAAATACTCCTTTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr12-14': defaultdict(None,\n",
       "                         {'sequence': 'GAGGGTATAGCTCAGAGGTAGAGCACTGGACTGCAGCTTCAGTTCAGCTGCCTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr5-2': defaultdict(None,\n",
       "                         {'sequence': 'AGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAAAAAAAAAAGTTTCCCCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr7-25': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTGAGGAGTAAACCATTTGGCTGCAGATCAAGTGGtCCCTGGTTCAAATATGGGTGCCCTCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr8-6': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAACATTAGACTGCAAATCTGAAGaTGGAGGTTAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chrX-3': defaultdict(None,\n",
       "                         {'sequence': 'GTGGGTATAGCTCAGAGGTAGAGCATTCAACTGCAGATCGAAGTGgCTGTACCATTTTGCATTCCCACCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgACCACTTTGGACTCTGAATACAGTGATCAGAGTTCAAGTCTCACTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGGGCTTTGGACTCTGACTACAGTGaTCAGAGTTCAAGTCTCAGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-12-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTaAGCACCCTGGACTCTGAATCCAGCAaCCAGAGTTCCAGTCTCAGCGtGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGTGTAGTCTACTGGTTAAACGCTTGGgCTCTGACATTAAcGtCCTGGGTTCAAATCCCAGCTTTGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-16-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCAGTGGTgcAGGGCATGGCTCTGAcACCTGGCGGCCTGGGTTCAAATCCCAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTaAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-4-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTaAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCGGTAaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCCATGGTGTAATGGTcAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTaAGCACTCTGGACTCTGAATCCAGCCATCTGAGTTCGAGTCTCTGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGCACTTTGGACTCTGAATACAGTGATCAGAGTTCAAGTCTCACTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-8-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGCACTTTGGACTCTGAATACAGTGATCAGAGTTCAAGTCTCACTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGgGTTAATGGTgAGCACCCTGGACTCTGAATCAAGCGaTCCGAGTTCAAATCTCGGTGGTACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-chr19-4': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCATGGTGTAATGGTgAGCACTCTGGACTCTGAATCCAGAAATACATTCAAAGAATTAAGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCAaTCCGAGTTCGAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-3-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGGCTTTGAATCCAGCAaTCCGAGTTCGAATCTTGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTGTGTACCTCAGTGGGcAAGGGTATGGACTTTGAAGCCAGACTaTTTGGGTTCAAATCCCAGCTTGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCATGTGGCCTAAGGGAaAAGACATCTCACTTTGGGTCAGAAGAtTGAGGGTTCAAGTCCTTTCATGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-chr8-10': defaultdict(None,\n",
       "                         {'sequence': 'GGTACAGTGTTAAAGGGGagaAAAATTGCTGACTTTGAATaCAGTAGaCCTAGGTTTGAATCCTGGCTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-3': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-4': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-5': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-6': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-7': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGAAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'CCCCTGGTGGTCTAGTGCTtAGGATTCGGTGCTCTCACCGCTGCTGCCTGCGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTTGATGTCTAGTGGTtAGGATTTGGTGCTCTCACTGCAGCAGCCTGGGTTCATTTCTCAGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCATATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGTGGCCCGGGTTCGACTCCCGGTATGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'TCCCATATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGTGGCCCGGGTTCGACTCCCGGTATGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-10-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCGTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCTGCAGCTCGAGTTCGATTCCTGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-11-1': defaultdict(None,\n",
       "                         {'sequence': 'CCCTGTGGTCTAGTGGCtAAGACTTTGTGCTTTCATTGCTGCAtCCTAGGTTCAATTCCCAGTCAGGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCACATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGCGGCCCGGGTTCGACTCCCGGTGTGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'TCCCACATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGCGGCCCGGGTTCGACTCCCGGTGTGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCGCGGCCCGGGTTCGATTCCCGGCCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-4-2': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTGTAGTGGTgAGCACAGCTGCCTTTCAAGCAGTTAaCGCGGGTTCGATTCCCGGGTAACGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCTTGGTGGTCTAGTGGCtAGGATTCGGTGCTTTCACCTGTGCGGCCCGGGTTCAATTCCCGATGAAGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'TGTCTGGTGGTCAAGTGGCtAGGATTTGGCGCTTTCACTGCCGCGGCCCGCGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCTGCAGCTCGAGTTCGATTCCTGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GCAATGGTGGTTCAGTGGTAGAATTCTCGCCTTTCACACAGGAGaCCCGGGTTCAATTCCTGACCCATGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTCCCACGCGGGAGaCCCGGGTTCAATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTCCCACGCGGGAGaCCCGGGTTCAATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGCCGCTGGTGTAGTGGTATCATGCAAGATTCCCATTCTTGCGaCCCGGGTTCGATTCCCGGGCGGCGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GCGCCGCTGGTGTAGTGGTATCATGCAAGATTCCCATTCTTGCGaCCCGGGTTCGATTCCCGGGCGGCGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAATGGTAGAATTCTCGCCTCCCACGCAGGAGaCCCAGGTTCGATTCCTGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTTTAGTGGTAGAATTCTCGCCTCCCATGCGGGAGaCCCGGGTTCAATTCCCGGCCACTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTTGGTGGTGCAGTGGTAGAATTCTCGCCTCCCACGTGGGAGaCCCGGGTTCAATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTGGTTCAGTGGTAGAATTCTCGCCTCCCACGCGGGAGaCCCGGGTTTAATTCCCGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GTGGTCTAGTGGTtAGGATTCAGCGCTCCCACCGCCGCAGCCCGGGTTCGATTCCCGGtCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTCAGTGGTTTAGTGGTgGAATTCCTGCCTCCCATGCACGAGaTCCGTGTTCAACTCCTGGTTGGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-chr1-102': defaultdict(None,\n",
       "                         {'sequence': 'GCGTCAGTGgTTTTAGTGGTgGAATTCCTGCCTCCCATGCACGAGaTCCGTGTTCAACTCCTGGTTGGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-chr1-16': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGCAGTTCAGTGGTAGAATTCTCGCCTCCCACCCGGGAGaCCTGGATTCCATTTCCGGCAAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-chr1-19': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGCAGTTCAGTGGTAGAATTCTCGCCTCCCACCCGGGAGaCCTGGATTCCATTTCCGGCAAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-6': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTTGATTCCCGGCCAGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATAGGTGGTTCAGTGGTAGAATTCTTGCCTGCCACGCAGGAGGCCCAGGTTTGATTCCTGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCATGCGGGCGGCCGGGCTTCGATTCCTgGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GCATGGGTGATTCAGTGGTAGAATTTTCACCTGCCATGCAGGAGGTCCAGGTTCATTTCCTGGCCTATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTtAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-6': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTaAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCGTTGGTGGTATAGTGGTgAGCATAGTTGCCTTCCAAGCAGTTGaCCCGGGCTCGATTCCCGCCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-6': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-7': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-8': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-9': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCATGATCGTATAGTGGTtAGTACTCTGCGCTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGCGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAACTTAATTGGTtAGAGCGTGGTGCTAATAATGTCAAGGtTGCGGGTTGGATCCCCGAACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-11-1': defaultdict(None,\n",
       "                         {'sequence': 'TGGCTGATGaGCTCAGCTGGTgGGAGCATGGTGTTAATGAGGCTGAGGtCGTGGGTTCAATCCCCACCGGGCTATCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGAACCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-4': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-5': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCTAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTcAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-7-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTcAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTCGGCtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTGGTTAGTTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGTGGGTTCGATCCCCATATCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-chr6-37': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCGGTTGCTtAGAGTGTTGTGTTAATTACGCtAAAGGtCGTGGGTTTGATCCTCTTACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-chr6-83': defaultdict(None,\n",
       "                         {'sequence': 'GTTAGCTTAGCTGGTgAGAGCGTGCTGCTAATAACACCAAGGtCGTGGTTTCTATCCCATtACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTaAGAGCGTGGTGCTGATAACACCAAGGtCGCGGGCTCGACTCCCGCACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTaAGAGCGTGGTGCTGATAACACCAAGGtCGCGGGCTCGACTCCCGCACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTaAGAGCGTGGTGCTGATAACACCAAGGtCGCGGGCTCGACTCCCGCACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-chrX-5': defaultdict(None,\n",
       "                         {'sequence': 'GGTCGGTTATCTCAGTTGGTgAGCGCGTGGTGCTGATACCGCCATGGtCGCGGGCTTGATCCCCGCACCGGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-chrX-7': defaultdict(None,\n",
       "                         {'sequence': 'GGTCGGTTATCTCAGTTGGTgAGCGCGTGGTGCTGATACCGCCATGGtCGCGGGCTTGATCCCCGCACCGGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-chrX-9': defaultdict(None,\n",
       "                         {'sequence': 'GGTCGGTTATCTCAGTTGGTgAGCGCGTGGTGCTGATACCGCCATGGtCGCAGGCTTGATCCCCGCACCGGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGATCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCAAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCAAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGTGGTctAAGACGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTTGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGCCAGTGGCTCAATGGAtAATGCGTCTGACTAAGAATCAGAAGAtTCCAGCCTTGACTCCTGGCTGGCTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGGGTGGCCGAGCGGTctAAGGCACTGTATTAAGACTCCAGTCTCTTCAGAGGCATGGGTTTGAATCCCACTGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAGCGGTctAAGGCTCCGGATTAAGGCGCCGGTGTCTTCGGAGgCATGGGTTCGAATTCCACCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGTATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGCATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGGATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCAGTcttAAGGCGCTGCGTTCAAATCGCACCCTCCGCTGGAGGCGTGGGTTCGAATCCCACTTTTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-7-1': defaultdict(None,\n",
       "                         {'sequence': 'CATAGGGTAGTGGCtAAGAACCTAAACTCTAAATTTAGATGtCCTGAGTTCAAATCCCAGCTGTATGCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-chr1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGTGTcAGGTAGGGCATGGATTTGGGAGCCAGAAAgCCCAGGTTCAAATCCTGACTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-chr14-11': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGTGTAGCCAAGAGGCcaggAAGGTGCACCCTGGAGCCCCACTGCCTAGGTTCAAATCCCAGCTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-6': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-7': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-chr9-3': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATGGTGTAATGGTgAGCACTCTGGACTCAGAATCCAGAAGtAGTgCTGGAACAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-chr9-7': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGGGTGGCTGAGCAGTctGAGGGGCTGCGTTCAGGTCGCAGTCTGCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGAAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCAGGATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGACATATGTCCGCGTGGGTTCGAACCCCACTCCTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCGGGATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGGCTGGTGCCCGCGTGGGTTCGAACCCCACTCTCGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCAGAATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGATTCATATCCGCGTGGGTTCGAACCCCACTTCTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCGGGATGGCTGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGACAGGTGTCCGCGTGGGTTCGAGCCCCACTCCCGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'ACTCATTTGGCTGAGTGGTtAAGGCATTGGACTTAAGATCCAATGGAGTAGTGGCTGTGTGGGTTTAAACCCCACTACTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-chr1-9': defaultdict(None,\n",
       "                         {'sequence': 'GAGAAAGTcATCGTAGTTACGAAGTTGGCTTAAACCCAGTTTtGGGAGGTTCAATTCCTTCCTTTCTCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-chr11-12': defaultdict(None,\n",
       "                         {'sequence': 'ACCAGGATGGCCAAGTAGTTaAAGGCACTGGACTTAAGAGCCAATGGACATATGTCTGTGTGGGTTTGAACCCCACTCCTGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGTGTGGCCGAGCGGTctAAGGCGCTGGATTTAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCACTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTTAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGCGTGGCCGAGTGGTctAAGGCGCTGGATTTAGGCTCCAGTCATTTCGATGgCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAGTGTGGTTGAATGGTctAAGGCACTGAATTTAGGCTCCAGTCTCTTTGGGGaCGTGGGTTTAAATCCCACTGCTGCAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCAGCTAGCTCAGTCGGTAGAGCATAAGACTCTTAATCTCAGGGtTGTGGATTCGTGCCCCATGCTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-11-1': defaultdict(None,\n",
       "                         {'sequence': 'CTGCAGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCATGGGTTCGTGCCCCATGTTGGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-12-1': defaultdict(None,\n",
       "                         {'sequence': 'CCAGCATGTCTCAGTCGGTATAGTGTGAGACTCTTAATCTCAGGGtCGTGGGTTCAAGCCCCACATTGGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCTAGCTAGATCAGTTGGTAGAGCATAAGACTCTTAATCTCAGGGtCATGGGTTTGAGCCCTACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-14-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCAGCTAGCTCAGCCGGTAGAGCACAAGACTCTTAATCTCAGGGtCGTGGGTTTGAGCCCTGTGTTGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-15-1': defaultdict(None,\n",
       "                         {'sequence': 'CCGAATAGCTTAGTTGATgAAGCGTGAGACTCTTAATCTCAGGGtAGTGGGTTCAAGCCCCACATTGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-16-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGCTACCTCAGTTGGTAGAGCATGGGACTCTTAATCCCAGAGtcAGTGGGTTCAAGCCTCACATTGAGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACCCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGCTAGCTCAGTCGATAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCGCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTTGAGCCCCATGTTGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCAGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCATGGGTTTGAGCCCCACGTTTGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGCTAGCTCAGTCGGCAAAGCATGAGACTCTTAATCTCAGGGtCGTGGGCTCGAGCTCCATGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGACTACCTCAGTCGGTgGAGCATGGGACTCTTCATCCCAGGGtTGTGGGTTCGAGCCCCACATTGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr15-5': defaultdict(None,\n",
       "                         {'sequence': 'GCACAGTCAGTcagtAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTCAAGCCCCACCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr16-1': defaultdict(None,\n",
       "                         {'sequence': 'CCCCGGCTGGCTCAGTCAGTAGATCATGAGACTCTTAATCTCAGGGtCGTGGGTTCACGCCCCACACTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr7-30': defaultdict(None,\n",
       "                         {'sequence': 'GCGCTAGTCAGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACATCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGATAGCTCAGTTGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCAGGATAGTTCAGGTGGTAGAGCATCAGACTTTTAACCTGAGGGtTCAGGGTTCAAGTCTCTGTTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-11-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCCAGATAGCTCAGTCAGTAGAGCATCAGACTTTTAATCTGAGGGtCCAAGGTTCATGTCCCTTTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-12-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCTGGGTAGCTTAGTTGGTAGAGCATTGGACTTTTAATTTGAGGGcCCAGGTTTCAAGTCCCTGTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-13-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGGTAGCTCAGTCGGTAGAGCTaTCAGACTTTTAGCCTGAGGAtTCAGGGTTCAATCCCTTGCTGGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-14-1': defaultdict(None,\n",
       "                         {'sequence': 'GATAGCTCAGTTGATAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCATGTCCCTGTTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-15-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTGGGGTAACTCAGTTGGTAGAGTAGCAGACTTTTCATCTGAGGGtCCAGGGTTTAAGTCCATGTCCAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGATAGCTCAGTTGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-3': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-4': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-5': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCGGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGGTAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTCCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGATAGCTCAGTTGGTAGAACATCAGACTTTTAATCTGACGGtGCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGAGAGCTCAGTGGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCTCGTTCGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-9-1': defaultdict(None,\n",
       "                         {'sequence': 'ACCTGGGTAGCTCAGTAGGTAGAACATCAGACTTTTAATCTGAGGGtCTAGGGTTCAAGTCCCTGTCCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-chr3-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCTGGATAGCTCCTTCGGTAGAGCATCATcagACTTTTAATGTGAGGGtCCAGGGTTCAAGTTCCTGTTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTCGTTAGCGCAGTAGGTAGCGCGTCAGTCTCATAATCTGAAGGtCGTGAGTTCGATCCTCACACGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGCGCAGTGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTCCTTAGCGCAGTAGGCAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAACCTCAGAGGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCTCCTTAGCGCAGTAGGCAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAACCTCAGAGGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGCGCAGCGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-4-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGCGCAGCGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-4-3': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGCGCAGCGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGCGCAGCTGGCAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCAAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCTCGTTAGCGCAGTAGGCAGCGCGTCAGTCTCATAATCTGAAGGtCGTGAGTTCGAGCCTCACACGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCTCTTAGTGCAGCTGGCAGCGCGTCAGTTTCATAATCTGAAAGtCCTGAGTTCAAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-6': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTGTCTAGTGGCtAGGATTCGGTGCTGAAAGCGTCACGgCCCGGGTTCGATTCCCGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTGAGATAGCTCGGTTGGGAGGGCATCAGCCTGAAGATCTAAAGGaGACTGGTTCAATTCTGGGTTTTGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAGATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACCGAAGATCTtAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCAAAATTGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTTAAAGttCCCTGGTTCAACCCTGGGTTTCAGCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCGAAATAGCTCAATTGGGAGAGTGTTAGACTGAAGATCTTCTGCAGGtCTCTGGTTCAATTCCGGGTTTCGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCAAAATAGCTCAGCTGGGAGAGCATTAGACTGAAGATCTAAAGGtCTCTGGTTTGATCCTGGGTTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGATGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-6': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-7': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-8': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTgTGGTTCTCGCTTAGGGCGGGAGACCCAAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGTGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-GGG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGTGTGGCATAGTGGTtagaAATGTGCGCTCTGGGGCTGCTGAtCCCAGGCTCAAACCCTGGCGCTGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGTGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGGTTTGGGTCCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-4': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-5': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATGATCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGCGACAGAGTGGTTCAATTCCACCTTTCGGGCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTCGGATGATCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGTGACAGAGTGGTTCAATTCCACCTTTGTAGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCGTGTGGCCTTAATGGAtAAGGTGTCTGACTTCAGATCAGAAGAtTGAGGGTTTGAGTCCCTTTGTGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-ACT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGCTGCTACTAATGCCAGGGtCGAGGTTTCGATCCCCGTACGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-3': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-4': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-5': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-6': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTTTCCCCACGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGTGATGGACTAGAAACCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGTGTATGGCTCAGGGGTAGAGAATTTGACTAGAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-6-1': defaultdict(None,\n",
       "                         {'sequence': 'AGTTGTAGCTGAGTGGTtAAGGCAACGAGCTAGAAATTCGTTGGTTTCTCTCTgTGCAGGTTTGAATCCTGCTAATTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-chr11-8': defaultdict(None,\n",
       "                         {'sequence': 'CAAGAAATTCATAGAGGTTATGGGATTGGCTAGAAACCAGTTTcAGGAGGTTCGATTCCTTCCTTTTTGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTGTGATGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTGTGATGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTCTCCCCGCGCAGGTTCAAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTGTGATGGCCGAGTGGTtAAGGTGTTGGACTCGAAATCCAATGGGGGTTCCCCGCGCAGGTTCAAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCACGGTGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTTTCCCCGCACAGGTTCGAATCCTGTTCGTGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCACCTTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTTTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-4-2': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-4-3': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACACGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GAGGCCTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCAGCGATGGCCGAGTGGTtAAGGCGTTGGACTTGAAATCCAATGGGGTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGTCGGCTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Sup-CTA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGGGTAGGGTAGAGGTtAAAACCATGGATTCTAGAGCCAGATGGGTTCAAATCCCGGCTCTGCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Sup-CTA-chr5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGTGTGATGTGGAGGAagggAGCATGTACTCTACTTCAGAGGGgCTTGGGTTCAAATCCCAGCTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Sup-TTA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GCCCGGATAGTTCAGTTGGTAGAGCATCAGACTTAATCAGAGGGtCCAGGGTTCAAGTCCCTGTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCCGTGGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCCGTGGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCCGTGGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-2-2': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCGTAGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGACTCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTTCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGAGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCTGTGGCTTAGCTGGTcAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-7-1': defaultdict(None,\n",
       "                         {'sequence': 'AGCACCATGGCTTAGCTGGTtAAAGCACCTGTCTAGTAAACAGGAGAtCCTGAGtTTCAATTcCAATGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTCGTAAACAGGAGAtCCTGGGTTCGACTCCCAGTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCGGTGGCCAAGTGGTAAGGCGTCGGTCTCGTAAACCGAAGAtCACGGGTTCGAACCCCGTCCGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTGTGGCTTAGTTGGCtAAAGCGCCTGTCTCGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCGCGGTGGCCAAGTGGTAAGGCGTCGGTCTCGTAAACCGAAGAtCGCGGGTTCGAACCCCGTCCGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCTGTAGCTCAGCGGTtGGAGCGCTGGTCTCGTAAACCtAGGGGtCGTGAGTTCAAATCTCACCAGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCAGAGTGGTGCAGCGGAAGCGTGCTGGGCCCGTAACCCAGAGGtCAATGGATCGAAGCCATCCTTGGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-chr1-118': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTCGTAAGCAGGAGGATCGCTTGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTTGTAAACAGGAGAtCCTGGGTTCGAATCCCAGTAGAGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCATAGCTCAGTGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCGATCCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCATAGCTCAGGGGTtAGAGCGCTGGTCTTGTAAACCAGGGGtCGCGAGTTCAATTCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCCTATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCTCCATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGCCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAATGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAGTCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-3-2': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-3-3': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGcTGCGTGTTCGAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GACCTCGTGGCGCAACGGCAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-chr7-31': defaultdict(None,\n",
       "                         {'sequence': 'GGCCTCATGGTGCAACAGTAGTGTGTCTGACTCCAGATCAGAAGGtTGTATGTTCAAATCACGTAGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr1-11': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCTTTAGACTATAAATCTAAAGAGAGATTGAGCTCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr1-52': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGACTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr11-9': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr9-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGGCCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GGGGGTATAGCTCAGGGCtAGAGCTtTTTGACTGTAGAGCAAGAGGtCCCTGGTTCAAATCCAGGTTCTCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-11-1': defaultdict(None,\n",
       "                         {'sequence': 'TATAGCTCAGTGGTAGAGCATTTAACTGTAGATCAAGAGGtCCCTGGATCAACTCTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-12-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCAGTGTTGCACAACGGTtaAGTGAAGAGGCTGTAAACCCAGACTGGATGGGTTCAATTCCCATCTCTGCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-2-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-3-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGGAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-4-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCAGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-2': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-3': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-4': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-5': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-6-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGGAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-7-1': defaultdict(None,\n",
       "                         {'sequence': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCAATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-8-1': defaultdict(None,\n",
       "                         {'sequence': 'CTTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-9-1': defaultdict(None,\n",
       "                         {'sequence': 'TCTTCAATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATTCTTAGGTGCTGGTTTGATTCCGACTTGGAGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr10-3': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGCAAGCATTGGACTGTAAATCTAAAGaCAGATGTTGAGCCATCTTTTTAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr14-8': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGaCAGGGGCTAAGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr2-12': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGCAAGCATTAGACTGTAAATCTAAAGaCAGAGGTTAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr21-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGaCAGGGGTTAAGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCATTAGACTGTAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTTCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGCAAGCATTAGACTGTAAATCTGAAAaCAGAGGTCAAAGgTCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-6': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCATTAGACTGTAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr8-7': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGACTGAATAAGCCTTAGACTGTAAATCTGAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr9-10': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAGCATTGGACTGTAAATCTAAAGaCAGAGGTCAAGACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr9-4': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTAAAGCATTAGACTGTAAATCTAAGGaCAGAGGCTAAACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTcATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCTGGATCAAAACCAGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCGCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTTGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCAGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCATAGTGTACTGGTtATCACATTCACCTAACACGCGAAAGGtCCTTGGTTTGAAACCAGGCAGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-chr6-84': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCATAGCGTAGTAATtATCAAGTTTGCCTAACACGCGAAATGtCCCCGGTTCAAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-3': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-4': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-5': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-6': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-10-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCTGTGGTGTAGTGGTtATCATGTTCGCCTCACACGAGAAAAGtCCCTGATTCGAGACTGGGTGGGAACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-11-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCTGTGGTGTAGTGGTtATTATGTTCGCTTCACATATGAAAGGtCTCTGGTTCGAGACTGCGTGGGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-11-2': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCTGTGGTGTAGTGGTtATTATGTTCGCTTCACATATGAAAGGtCTCTGGTTCGAGACTGCGTGGGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GCTTCTGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCAGAAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGCGGTtATCACATTCGCCTCACACGCGAAAGGtCCCCGGTTCGATCCCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACTGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-5-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGtAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-6-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGGAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTTGAAACCAGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-7-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTAGTGTAGTGGTtATTATGTTCGCCTCACACGCGAAAAGtCCCCGGTTCGAAATCAGGCGGGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-8-1': defaultdict(None,\n",
       "                         {'sequence': 'GCACTGGTGGTTCAGTGGTAGAATTCTCGCCTCACACGCGGGACACCCGGGTTCAATTCCCGGTCAAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-9-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCTGTAGTATGGTGGTtATCACGTTAGTCTCACACGTGAAAGGtCCCTGGTTCGAAACCAGGTGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATAGTGTAGTGGTtATCACGTCTGCTTTACACGCAGAAGGtCCTGGGTTCGAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-1-2': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATAGTGTAGTGGTtATCACGTCTGCTTTACACGCAGAAGGtCCTGGGTTCGAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-2-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATAGTGTAGCGGTtATCACGTCTGCTTTACACGCAGAAGGtCCTGGGTTCGAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-3-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATAGTGTAGTGGTtATCACATCTGCTTTACACGCAGAAGGtCCTGGGTTCAAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-4-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTCCGTGGTGTAGTGGTtATCACATTCGCCTTACACGCGAAAGGtCCTCGGGTCGAAACCGAGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-chr17-24': defaultdict(None,\n",
       "                         {'sequence': 'GATTCTATAGTGTAGCAGTtATCACATCTGCTTTACATGCAGAAGGtCCTAGGTTTGAGCCGCGGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-chr2-6': defaultdict(None,\n",
       "                         {'sequence': 'GGTTCCATAGTGTAACAGTtATCATGTCTGCTTTACACACTGAAGGtCCTGGGTTCAAACTCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-2': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-3': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-4': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-5': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-6': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-7': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-8': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-2-1': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCTAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-3-1': defaultdict(None,\n",
       "                         {'sequence': 'AGCAGAGTGGTGCAGTGGAAGCATACctaTGGGCCCATAACCCAGAGGtTGATGGATGGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ala-TGC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AAGGGCTTAGCTTAATTAAAGTGGCTGATTTGCGTTCAGTTGATGCAGAGTGGGGTTTTGCAGTCCTTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Arg-TCG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TGGTATATAGTTTAAACAAAACGAATGATTTCGACTCATTAAATTATGATAATCATATTTACCAACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Asn-GTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TAGATTGAAGCCAGTTGATTAGGGTGCTTAGCTGTTAACTAAGTGTTTGTGGGTTTAAGTCCCATTGGTCTAGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Asp-GTC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AAGGTATTAGAAAAACCATTTCATAACTTTGTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Cys-GCA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AGCTCCGAGGTGATTTTCATATTGAATTGCAAATTCGAAGAAGCAGCTTCAAACCTGCCGGGGCTTCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Gln-TTG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'TAGGATGGGGTGTGATAGGTGGCACGGAGAATTTTGGATTCTCAGGGATGGGTTCGATTCTCATAGTCCTAGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Glu-TTC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTCTTGTAGTTGAAATACAACGATGGTTTTTCATATCATTGGTCGTGGTTGTAGTCCGTGCGAGAATACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Gly-TCC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'ACTCTTTTAGTATAAATAGTACCGTTAACTTCCAATTAACTAGTTTTGACAACATTCAAAAAAGAGTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-His-GTG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTAAATATAGTTTAACCAAAACATCAGATTGTGAATCTGACAACAGAGGCTTACGACCCCTTATTTACCCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ile-GAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Leu-TAG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'ACTTTTAAAGGATAACAGCTATCCATTGGTCTTAGGCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Leu-TAA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTAAGATGGCAGAGCCCGGTAATCGCATAAAACTTAAAACTTTACAGTCAGAGGTTCAATTCCTCTTCTTAACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Lys-TTT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'CACTGTAAAGCTAACTTAGCATTAACCTTTTAAGTTAAAGATTAAGAGAACCAACACCTCTTTACAGTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Met-CAT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AGTAAGGTCAGCTAAATAAGCTATCGGGCCCATACCCCGAAAATGTTGGTTATACCCTTCCCGTACTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Phe-GAA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTTTATGTAGCTTACCTCCTCAAAGCAATACACTGAAAATGTTTAGACGGGCTCACATCACCCCATAAACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Pro-TGG-1-1': defaultdict(None,\n",
       "                         {'sequence': 'CAGAGAATAGTTTAAATTAGAATCTTAGCTTTGGGTGCTAATGGTGGAGTTAAAGACTTTTTCTCTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ser-GCT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GAGAAAGCTCACAAGAACTGCTAACTCATGCCCCCATGTCTAACAACATGGCTTTCTCACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ser-TGA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GAAAAAGTCATGGAGGCCATGGGGTTGGCTTGAAACCAGCTTTGGGGGGTTCGATTCCTTCCTTTTTTGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Thr-TGT-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GTCCTTGTAGTATAAACTAATACACCAGTCTTGTAAACCGGAGATGAAAACCTTTTTCCAAGGACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Trp-TCA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'AGAAATTTAGGTTAAATACAGACCAAGAGCCTTCAAAGCCCTCAGTAAGTTGCAATACTTAATTTCTGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Tyr-GTA-1-1': defaultdict(None,\n",
       "                         {'sequence': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGACAGGGGTTAGGCCTCTTTTTACCCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'}),\n",
       "             'Homo_sapiens_mito_tRNA-Val-TAC-1-1': defaultdict(None,\n",
       "                         {'sequence': 'CAGAGTGTAGCTTAACACAAAGCACCCAACTTACACTTAGGAGATTTCAACTTAACTTGACCGCTCTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens'})})"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trna_obj.tRNA_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Look at CCA for e.coli\n",
    "# Remove duplicate entries for cyto tRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "import re, copy, sys, os, shutil, subprocess, logging, glob\n",
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "from mimseq.ssAlign import aligntRNA, extraCCA, tRNAclassifier, tRNAclassifier_nogaps, getAnticodon, clusterAnticodon\n",
    "\n",
    "def modsToSNPIndex(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, experiment_name, out_dir, double_cca, snp_tolerance = False, cluster = False, cluster_id = 0.95, posttrans_mod_off = False, pretrnas = False, local_mod = False):\n",
    "# Builds SNP index needed for GSNAP based on modificaiton data for each tRNA and clusters tRNAs\n",
    "\n",
    "    nomatch_count = 0\n",
    "    match_count = 0\n",
    "    total_snps = 0\n",
    "    total_inosines = 0\n",
    "    snp_records = list()\n",
    "    seq_records = defaultdict()\n",
    "    anticodon_list = list()\n",
    "    tRNAbed = open(out_dir + experiment_name + \"_maturetRNA.bed\",\"w\")\n",
    "    # generate modomics_dict and tRNA_dict\n",
    "    tRNA_dict, modomics_dict, species = tRNAparser(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, posttrans_mod_off, double_cca, pretrnas, local_mod)\n",
    "    temp_dir = out_dir + \"/tmp/\"\n",
    "\n",
    "    try:\n",
    "        os.mkdir(temp_dir)\n",
    "    except FileExistsError:\n",
    "        log.warning(\"Temp folder present - previous run interrupted? Overwriting old temp files...\\n\")\n",
    "\n",
    "    ###################################################\n",
    "    ## Main code for matching and SNP index building ##\n",
    "    ###################################################\n",
    "\n",
    "    # remove spurious CCA seen in some sequences (i.e. genomically encoded or artifact from gtRNAdb or tRNAScan-SE prediction)\n",
    "    # to do this, remove previously added CCA (intronRemover()) that falls out of canonical structure using ssAlign module\n",
    "\n",
    "    with open(out_dir + 'temptRNAseqs.fa', 'w') as tempSeqs:\n",
    "        for seq in tRNA_dict:\n",
    "            tempSeqs.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "\n",
    "    aligntRNA(tempSeqs.name, out_dir)\n",
    "    extra_cca = extraCCA()\n",
    "\n",
    "    for record in extra_cca:\n",
    "        tRNA_dict[record]['sequence'] = tRNA_dict[record]['sequence'][:-3]\n",
    "\n",
    "    os.remove(tempSeqs.name)\n",
    "\n",
    "    # match each sequence in tRNA_dict to value in modomics_dict using BLAST\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "##### Up to here constitutes an object #####\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    log.info(\"\\n+------------------------+ \\\n",
    "        \\n| Beginning SNP indexing |\\\n",
    "        \\n+------------------------+\")    \n",
    "\n",
    "    for seq in tRNA_dict:\n",
    "        # Initialise list of modified sites for each tRNA\n",
    "        tRNA_dict[seq]['modified'] = []\n",
    "        tRNA_dict[seq]['InosinePos'] = []\n",
    "        temp_tRNAFasta = open(temp_dir + seq + \".fa\",\"w\")\n",
    "        temp_tRNAFasta.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "        temp_tRNAFasta.close()\n",
    "        tRNA_dict[seq]['anticodon'] = anticodon = re.search('.*tR(NA|X)-.*?-(.*?)-', seq).group(2)\n",
    "        if not anticodon in anticodon_list:\n",
    "            anticodon_list.append(anticodon)\n",
    "        # find initial possible matches to modomics where anticodons match and types are the same (here regex is used to match anticodons with \".\" in modomics to all possible matching sequences from input tRNAs)\n",
    "        match = {k:v for k,v in modomics_dict.items() if re.match(\"^\" + v['anticodon'] + \"+$\", anticodon) and tRNA_dict[seq]['type'] == v['type']}\n",
    "        if len(match) >= 1:\n",
    "            temp_matchFasta = open(temp_dir + \"modomicsMatch.fasta\",\"w\")\n",
    "            for i in match:    \n",
    "                temp_matchFasta.write(\">\" + i + \"\\n\" + match[i]['unmod_sequence'] + \"\\n\")\n",
    "            temp_matchFasta.close()\n",
    "\n",
    "            #blast\n",
    "            blastn_cline = NcbiblastnCommandline(query = temp_tRNAFasta.name, subject = temp_matchFasta.name, task = 'blastn-short', out = temp_dir + \"blast_temp.xml\", outfmt = 5)\n",
    "            blastn_cline()\n",
    "\n",
    "            #parse XML result and store hit with highest bitscore    \n",
    "            blast_record = NCBIXML.read(open(temp_dir + \"blast_temp.xml\",\"r\"))\n",
    "            maxbit = 0\n",
    "            tophit = ''\n",
    "            for alignment in blast_record.alignments:\n",
    "                for hsp in alignment.hsps:\n",
    "                    if (hsp.bits > maxbit) and (hsp.align_length / alignment.length == 1) and (hsp.identities / alignment.length == 1):\n",
    "                        maxbit = hsp.bits\n",
    "                        tophit = alignment.title.split(' ')[0]\n",
    "            \n",
    "            # return list of all modified positions for the match as long as there is only 1, add to tRNA_dict\n",
    "            if tophit:\n",
    "                match_count += 1\n",
    "                tRNA_dict[seq]['modified'] = match[tophit]['modified']\n",
    "                tRNA_dict[seq]['InosinePos'] = match[tophit]['InosinePos']\n",
    "            elif len(tophit) == 0:\n",
    "                nomatch_count += 1\n",
    "        if len(match) == 0:\n",
    "            nomatch_count += 1\n",
    "\n",
    "        # Build seqrecord list for writing\n",
    "        seq_records[str(seq)] = SeqRecord(Seq(tRNA_dict[seq]['sequence'].upper()), id = str(seq))\n",
    "\n",
    "        tRNAbed.write(seq + \"\\t0\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t\" + seq + \"\\t1000\\t+\\n\" )\n",
    "\n",
    "    tRNAbed.close()\n",
    "\n",
    "    log.info(\"{} total tRNA gene sequences (undetermined and nmt sequences excluded)\".format(len(tRNA_dict)))\n",
    "    log.info(\"{} sequences with a match to Modomics dataset\".format(match_count))\n",
    "\n",
    "    with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "        SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "    # if clustering is not activated then write full gff and report on total SNPs written \n",
    "    if not cluster:\n",
    "        coverage_bed = tRNAbed.name\n",
    "        mod_lists = defaultdict(list)\n",
    "        Inosine_lists = defaultdict(list)\n",
    "        with open(out_dir + experiment_name + \"_tRNA.gff\",\"w\") as tRNAgff, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:    \n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for seq in tRNA_dict:\n",
    "                mod_lists[seq] = tRNA_dict[seq]['modified']\n",
    "                Inosine_lists[seq] = tRNA_dict[seq]['InosinePos']\n",
    "                tRNAgff.write(seq + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + seq + \"'\\n\")\n",
    "                isoacceptor_group = '-'.join(seq.split(\"-\")[:-2])\n",
    "                isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        # generate Stockholm alignment file for all tRNA transcripts and parse additional mods file\n",
    "        aligntRNA(temptRNATranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "        # add additional SNPs from extra file to list of modified positions, and ensure non-redundancy with set()\n",
    "        # index SNPs\n",
    "        # Format for SNP index (space separated):\n",
    "        # >snpID chromosomeName:position(1-based) RefMod\n",
    "        # e.g. >rs111 Homo_sapiens_nmt_tRNA-Leu-TAA-1-1_exp0:29 GN\n",
    "        for seq in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[seq]['modified'] = list(set(tRNA_dict[seq]['modified'] + additionalMods_sub[seq]['mods']))\n",
    "                mod_lists[seq] = list(set(mod_lists[seq] + additionalMods_sub[seq]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[seq])\n",
    "\n",
    "            # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "            # Position is 1-based for iit_store i.e. pos + 1\n",
    "            for (index, pos) in enumerate(mod_lists[seq]):\n",
    "                #if \"Gln-TTG\" in seq and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for seq in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[seq]['InosinePos'] = list(set(tRNA_dict[seq]['InosinePos'] + additionalInosines_sub[seq]['InosinePos']))\n",
    "                Inosine_lists[seq] = list(set(Inosine_lists[seq] + additionalInosines_sub[seq]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[seq])\n",
    "            total_snps += len(Inosine_lists[seq])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[seq]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions only if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for seq in Inosine_lists:\n",
    "                for pos in Inosine_lists[seq]:\n",
    "                    seq_records[seq].seq = seq_records[seq].seq[0:pos] + \"G\" + seq_records[seq].seq[pos+1:]\n",
    "\n",
    "        with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "            SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False\n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))\n",
    "        # empty mismatch dict to avoid error when returning it from this function\n",
    "        mismatch_dict = defaultdict(list)\n",
    "        insert_dict = defaultdict(dd_list)\n",
    "        del_dict = defaultdict(dd_list)\n",
    "        mod_lists = {tRNA:list() for tRNA in tRNA_dict.keys()}\n",
    "        Inosine_lists = {tRNA:data['InosinePos'] for tRNA, data in tRNA_dict.items()}\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list)\n",
    "        cluster_dict = defaultdict(list)\n",
    "\n",
    "    ##########################\n",
    "    # Cluster tRNA sequences #\n",
    "    ##########################\n",
    "\n",
    "    elif cluster:\n",
    "\n",
    "        log.info(\"**** Clustering tRNA sequences ****\")\n",
    "        log.info(\"Clustering tRNA sequences by {:.0%} similarity...\".format(cluster_id))\n",
    "        # dictionary of final centroid sequences\n",
    "        final_centroids = defaultdict()\n",
    "        # get dictionary of sequences for each anticodon and write to fastas\n",
    "        for anticodon in anticodon_list:\n",
    "            seq_set = {k:{'sequence':v['sequence'],'modified':v['modified']} for k,v in tRNA_dict.items() if v['anticodon'] == anticodon}\n",
    "            with open(temp_dir + anticodon + \"_allseqs.fa\",\"w\") as anticodon_seqs:\n",
    "                for sequence in seq_set:\n",
    "                    anticodon_seqs.write(\">\" + sequence + \"\\n\" + seq_set[sequence]['sequence'] + \"\\n\")\n",
    "            # run usearch on each anticodon sequence fatsa to cluster\n",
    "            cluster_cmd = [\"usearch\", \"-cluster_fast\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]\n",
    "            #cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"--sortedby\", \"other\" ,\"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]            #cluster_cmd = \"usearch -cluster_fast \" + temp_dir + anticodon + \"_allseqs.fa -sort length -id \" + str(cluster_id) + \" -centroids \" + temp_dir + anticodon + \"_centroids.fa -uc \" + temp_dir + anticodon + \"_clusters.uc &> /dev/null\"\n",
    "            subprocess.check_call(cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # sort clusters by size (i.e. number of members in cluster)\n",
    "            sort_cmd = [\"usearch\", \"-sortbysize\", temp_dir + anticodon + \"_centroids.fa\", \"-fastaout\", temp_dir + anticodon + \"_centroids_sort.fa\"]\n",
    "            subprocess.check_call(sort_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # recluster based on sorted by size clusters\n",
    "            final_cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_centroids_sort.fa\", \"-sortedby\", \"size\", \"-id\", str(cluster_id), \"-centroids\", temp_dir + anticodon + \"_centroidsFinal.fa\"]\n",
    "            subprocess.check_call(final_cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "        # combine centroids files into one file\n",
    "        for filename in glob.glob(temp_dir + \"*_centroidsFinal.fa\"):\n",
    "            with open(filename, \"r\") as fileh:\n",
    "                with open(temp_dir + \"all_centroids.fa\", \"a\") as outh:\n",
    "                    outh.write(fileh.read())\n",
    "        centroids = SeqIO.parse(temp_dir + \"all_centroids.fa\", \"fasta\")\n",
    "        for centroid in centroids:\n",
    "            centroid.id = centroid.id.split(\";\")[0]\n",
    "            final_centroids[centroid.id] = SeqRecord(Seq(str(centroid.seq).upper()), id = centroid.id) \n",
    "\n",
    "        # read cluster files, get nonredudant set of mod positions of all members of a cluster, create snp_records for writing SNP index\n",
    "        cluster_pathlist = Path(temp_dir).glob(\"**/*_clusters.uc\")\n",
    "        mod_lists = defaultdict(list) # stores non-redundant sets of mismatches and mod positions for clusters\n",
    "        Inosine_lists = defaultdict(list) # stores positions of inosines for clusters\n",
    "        snp_records = list()\n",
    "        cluster_dict = defaultdict(list) # info about clusters and members\n",
    "        mismatch_dict = defaultdict(list) # dictionary of mismatches only (not mod positions - required for exclusion from misincorporation analysis in mmQuant)\n",
    "        insert_dict = defaultdict(dd_list) # dictionary of insertions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        del_dict = defaultdict(dd_list) # dictionary of deletions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list) # for each cluster, list of members that mismatch at each position corresonding to mismatch dict - used for splitting clusters into isodecoders (splitClusters.py)\n",
    "        cluster_num = 0\n",
    "        total_snps = 0\n",
    "        total_inosines = 0\n",
    "        clusterbed = open(out_dir + experiment_name + \"_clusters.bed\",\"w\")\n",
    "        coverage_bed = clusterbed.name\n",
    "        clustergff = open(out_dir + experiment_name + \"_tRNA.gff\",\"w\")\n",
    "        for path in cluster_pathlist:\n",
    "            with open(str(path),\"r\") as cluster_file:\n",
    "                for line in cluster_file:\n",
    "                    line = line.strip()\n",
    "\n",
    "                    # Handle cluster centroids and initialise modified positions list\n",
    "                    if line.split(\"\\t\")[0] == \"S\":\n",
    "                        cluster_num += 1\n",
    "                        cluster_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        mod_lists[cluster_name] = tRNA_dict[cluster_name][\"modified\"]\n",
    "                        Inosine_lists[cluster_name] = tRNA_dict[cluster_name]['InosinePos']\n",
    "                        clusterbed.write(cluster_name + \"\\t0\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t\" + cluster_name + \"\\t1000\\t+\\n\" )\n",
    "                        clustergff.write(cluster_name + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + cluster_name + \"'\\n\")\n",
    "                        cluster_dict[cluster_name].append(cluster_name)\n",
    "                \n",
    "                    # Handle members of clusters\n",
    "                    elif line.split(\"\\t\")[0] == \"H\":\n",
    "                        member_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        cluster_name = line.split(\"\\t\")[9].split(\";\")[0]\n",
    "                        compr_aln = line.split(\"\\t\")[7]\n",
    "                        # if member of cluster is 100% identical (i.e. \"=\" in 8th column of cluster file)\n",
    "                        if compr_aln == \"=\":\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + tRNA_dict[member_name][\"modified\"]))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + tRNA_dict[member_name][\"InosinePos\"]))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "                        \n",
    "                        # if there are insertions or deletions in the centroid, edit member or centroid sequences to ignore these positions\n",
    "                        # and edit modified positions list in order to make non-redundant positions list, similar to next else statement\n",
    "                        elif re.search(\"[ID]\", compr_aln):\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            pos = 0\n",
    "                            adjust_pos_del = 0\n",
    "                            adjust_pos_ins = 0\n",
    "                            insertion_pos = list()\n",
    "                            deletion_pos = list()\n",
    "                            aln_list = re.split('(.*?[A-Z])', compr_aln)\n",
    "                            aln_list = list(filter(None, aln_list))\n",
    "\n",
    "                            for phrase in aln_list:\n",
    "                                \n",
    "                                if (\"M\" in phrase) and (phrase.split(\"M\")[0] != \"\"):\n",
    "                                    pos += int(phrase.split(\"M\")[0])\n",
    "                                elif (\"M\" in phrase) and (phrase.split(\"M\")[0] == \"\"):\n",
    "                                    pos += 1\n",
    "\n",
    "                                if (\"I\" in phrase) and (phrase.split(\"I\")[0] != \"\"):\n",
    "                                    insert_len = int(phrase.split(\"I\")[0])\n",
    "\n",
    "                                    for i in range(insert_len):\n",
    "                                        insertion_pos.append(pos+i)\n",
    "                                        #pos += 1\n",
    "                                elif (\"I\" in phrase) and (phrase.split(\"I\")[0] == \"\"):\n",
    "                                    insertion_pos.append(pos)\n",
    "                                    #pos += 1\n",
    "\n",
    "                                if (\"D\" in phrase) and (phrase.split(\"D\")[0] != \"\"):\n",
    "                                    delete_len = int(phrase.split(\"D\")[0])\n",
    "\n",
    "                                    for i in range(delete_len):\n",
    "                                        deletion_pos.append(pos)\n",
    "                                        pos += 1\n",
    "                                elif (\"D\" in phrase) and (phrase.split(\"D\")[0] == \"\"):\n",
    "                                    deletion_pos.append(pos)\n",
    "                                    pos += 1\n",
    "\n",
    "                            for delete in deletion_pos:\n",
    "                                adjust_pos_len = 0\n",
    "                                for insert in insertion_pos:\n",
    "                                    if insert < delete:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                new_delete = delete + adjust_pos_del + adjust_pos_len\n",
    "                                member_seq = member_seq[ :new_delete] + member_seq[new_delete+1: ]\n",
    "                                del_dict[cluster_name][new_delete].append(member_name)\n",
    "                                adjust_pos_del -= 1\n",
    "\n",
    "                            for index, insert in enumerate(insertion_pos):\n",
    "                                adjust_pos_len = 0\n",
    "                                for delete in deletion_pos:\n",
    "                                    if delete < insert:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                #if index != 0:\n",
    "                                #    if insert == insertion_pos[index-1] + 1:\n",
    "                                #        adjust_pos_ins -= 1\n",
    "                                new_insert = insert + adjust_pos_len + adjust_pos_ins\n",
    "                                member_seq = member_seq[ :new_insert] + cluster_seq[insert] + member_seq[new_insert: ]\n",
    "                                #insert_dict[cluster_name][new_insert-1].append(member_name)\n",
    "                                insert_dict[cluster_name][new_insert].append(member_name)\n",
    "                                #cluster_seq = cluster_seq[ :new_insert] + cluster_seq[new_insert+1: ]\n",
    "\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "                        # handle members that are not exact sequence matches but have no indels either\n",
    "                        # find mismatches and build non-redundant set\n",
    "                        else:\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "        clusterbed.close()\n",
    "\n",
    "        # Write cluster information to tsv and get number of unique sequences per cluster (i.e. isodecoders) for read count splitting\n",
    "        with open(out_dir + experiment_name + \"clusterInfo.txt\",\"w\") as clusterInfo, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:\n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            clusterInfo.write(\"tRNA\\tcluster_num\\tcluster_size\\tparent\\n\")\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for key, value in cluster_dict.items():\n",
    "                cluster_num = list(cluster_dict.keys()).index(key)\n",
    "                for member in value:\n",
    "                    clusterInfo.write(\"{}\\t{}\\t{}\\t{}\\n\".format(member, cluster_num, len(value), key))\n",
    "                    isoacceptor_group = '-'.join(member.split(\"-\")[:-2])\n",
    "                    isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        # generate Stockholm alignment file for cluster transcripts and process additional mods file\n",
    "        aligntRNA(clusterTranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "\n",
    "        log.info(\"{} clusters created from {} tRNA sequences\".format(len(cluster_dict),len(tRNA_dict)))\n",
    "\n",
    "        # update mod_lists with additional mods and write SNP index\n",
    "        for cluster in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[cluster]['modified'] = list(set(tRNA_dict[cluster]['modified'] + additionalMods_sub[cluster]['mods']))\n",
    "                mod_lists[cluster] = list(set(mod_lists[cluster] + additionalMods_sub[cluster]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(mod_lists[cluster]):\n",
    "                # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "                # Position is 1-based for iit_store i.e. pos + 1\n",
    "                #if \"Gln-TTG\" in cluster and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for cluster in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[cluster]['InosinePos'] = list(set(tRNA_dict[cluster]['InosinePos'] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "                Inosine_lists[cluster] = list(set(Inosine_lists[cluster] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[cluster])\n",
    "            total_snps += len(Inosine_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[cluster]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for cluster in Inosine_lists:\n",
    "                for pos in Inosine_lists[cluster]:\n",
    "                    final_centroids[cluster].seq = final_centroids[cluster].seq[0:pos] + \"G\" + final_centroids[cluster].seq[pos+1:]\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # rewrite edited cluster transcripts\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False        \n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))        \n",
    "    \n",
    "    # write outputs for indexing \n",
    "    with open(out_dir + experiment_name + \"_modificationSNPs.txt\", \"w\") as snp_file:\n",
    "        for item in snp_records:\n",
    "            snp_file.write('{}\\n'.format(item))\n",
    "    \n",
    "    shutil.rmtree(temp_dir)\n",
    "    \n",
    "    # Return coverage_bed (either tRNAbed or clusterbed depending on --cluster) for coverage calculation method\n",
    "    return(coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers = modsToSNPIndex(trnas, trnaout, mito_trnas, modifications, name, out, double_cca, snp_tolerance, cluster, cluster_id, posttrans, pretrnas, local_mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=False, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='hg19_HEK239vsK562', posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
