{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from mimseq import version\n",
    "from mimseq.tRNAtools import TRNA\n",
    "#from mimseq.tRNAtools import modsToSNPIndex, generateGSNAPIndices, newModsParser, tidyFiles\n",
    "#from mimseq.tRNAmap import mainAlign\n",
    "#from mimseq.getCoverage import getCoverage, plotCoverage\n",
    "#from mimseq.mmQuant import generateModsTable, plotCCA\n",
    "#from mimseq.ssAlign import structureParser, modContext \n",
    "#from mimseq.splitClusters import splitIsodecoder, unsplitClusters, getIsodecoderSizes, writeIsodecoderTranscripts\n",
    "import sys, os, subprocess, logging, datetime, copy, shutil, re, glob\n",
    "import argparse\n",
    "from pyfiglet import figlet_format\n",
    "from collections import defaultdict\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "log = logging.getLogger(__name__)\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def restrictedFloat(x):\n",
    "## Method for restricting cluster_id and cov_diff argument to float between 0 and 1\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0 or x > 1.0:\n",
    "            raise argparse.ArgumentTypeError('{} not in range 0.0 - 1.0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "def restrictedFloat2(x):\n",
    "## Method for restricting min-cov argument to float between 0 and 1, or int greater than 1\n",
    "\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0:\n",
    "            raise argparse.ArgumentTypeError('{} not greater than 0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "\n",
    "\n",
    "parser = argparse.ArgumentParser(description = 'Custom high-throughput tRNA sequencing alignment and quantification pipeline\\\n",
    "    based on modification induced misincorporation cDNA synthesis.', add_help = True, usage = \"%(prog)s [options] sample data\")\n",
    "\n",
    "inputs = parser.add_argument_group(\"Input files\")\n",
    "inputs.add_argument('-s','--species', metavar='species', required = not ('-t' in sys.argv), dest = 'species', help = \\\n",
    "    'Species being analyzed for which to load pre-packaged data files (prioritized over -t, -o and -m). Options are: Hsap, Hsap38, Mmus, Scer, Spom, Dmel, Drer, Ecol', \\\n",
    "    choices = ['Hsap','Hsap38','Mmus','Scer','Spom','Dmel', 'Drer', 'Ecol'])\n",
    "inputs.add_argument('-t', '--trnas', metavar='genomic tRNAs', required = False, dest = 'trnas', help = \\\n",
    "    'Genomic tRNA fasta file, e.g. from gtRNAdb or tRNAscan-SE. Already avalable in data folder for a few model organisms.')\n",
    "inputs.add_argument('-o', '--trnaout', metavar = 'tRNA out file', required = (not '--species' or '-s' in sys.argv) or ('-t' in sys.argv), \n",
    "    dest = 'trnaout', help = 'tRNA.out file generated by tRNAscan-SE (also may be available on gtRNAdb). Contains information about tRNA features, including introns.')\n",
    "inputs.add_argument('-m', '--mito-trnas', metavar = 'mitochondrial tRNAs', required = False, dest = 'mito', \\\n",
    "    help = 'Mitochondrial tRNA fasta file. Should be downloaded from mitotRNAdb for species of interest. Already available in data folder for a few model organisms.')\n",
    "\n",
    "options = parser.add_argument_group(\"Program options\")\n",
    "options.add_argument('--pretRNAs', required = False, dest = 'pretrnas', action = 'store_true',\\\n",
    "    help = \"Input reference sequences are pretRNAs. Enabling this option will disable the removal of intron sequences and addition of 3'-CCA to generate \\\n",
    "    mature tRNA sequences. Useful for mapping and discovering pretRNA sequence reads.\")\n",
    "options.add_argument('--no-cluster', required = False, dest = 'cluster', action = 'store_false',\\\n",
    "    help = 'Disable usearch sequence clustering of tRNAs by isodecoder which drastically reduces the rate of multi-mapping reads. Default is enabled.')\n",
    "options.add_argument('--cluster-id', metavar = 'clustering identity threshold', dest = 'cluster_id', type = restrictedFloat, nargs = '?', default = 0.97,\\\n",
    "    required = False, help = 'Identity cutoff for usearch clustering between 0 and 1. Default is 0.97.')\n",
    "options.add_argument('--deconv-cov-ratio', metavar='deconvolution coverage threshold', dest='cov_diff', type = restrictedFloat, nargs = '?', default=0.5,\\\n",
    "    required=False, help=\"Threshold for ratio between coverage at 3' end and mismatch used for deconvolution. Coverage reductions greater than the threshold will result in non-deconvoluted sequences. \\\n",
    "        Default is 0.5 (i.e. less than 50%% reduction required for deconvolution).\")\n",
    "options.add_argument('--threads', metavar = 'thread number', required = False, dest = 'threads', type = int, \\\n",
    "    help = 'Set processor threads to use during read alignment and read counting.')\n",
    "options.add_argument('--posttrans-mod-off', required = False, dest = 'posttrans', action = 'store_true', \\\n",
    "    help = \"Disable post-transcriptional modification of tRNAs, i.e. addition of 3'-CCA and 5'-G (His) to mature sequences. Disable for certain \\\n",
    "    prokaryotes (e.g. E. coli) where this is genomically encoded. Leave enabled (default) for all eukaryotes.\")\n",
    "options.add_argument('--control-condition', metavar = 'control condition', required = True, dest = 'control_cond', \\\n",
    "    help = 'Name of control/wild-type condition as per user defined group specified in sample data input. This must exactly match the group name \\\n",
    "    specified in sample data. This is used for differential expression analysis so that results are always in the form mutant/treatment vs WT/control. REQUIRED')\n",
    "options.add_argument('--no-cca-analysis', required = False, dest = 'cca', action = 'store_false',\\\n",
    "    help = \"Disable analysis of 3'-CCA ends. When enabled, this calculates proportions of CC vs CCA ending reads per cluster and performs DESeq2 analysis. \\\n",
    "    Useful for comparing functional to non-functional mature tRNAs. Default is enabled.\")\n",
    "options.add_argument('--double-cca', required = False, dest = 'double_cca', action = \"store_true\",\\\n",
    "    help = \"Enable analysis of 3'-CCACCA tagging for tRNA degradation pathway. Note that this will alter the output of the CCA analysis pipeline.\")\n",
    "options.add_argument('--local-modomics', required=False, dest = 'local_mod', action='store_true',\\\n",
    "    help = \"Disable retrieval of Modomics data from online. Instead use older locally stored data. Warning - this leads\\\n",
    "        to usage of older Modomics data!\")\n",
    "\n",
    "align = parser.add_argument_group(\"GSNAP alignment options\")\n",
    "align.add_argument('--max-mismatches', metavar = 'allowed mismatches', required = False, dest = 'mismatches', type = float, \\\n",
    "    help = 'Maximum mismatches allowed. If specified between 0.0 and 1.0, then treated as a fraction of read length. Otherwise, treated as \\\n",
    "    integer number of mismatches. Default is an automatic ultrafast value calculated by GSNAP; see GSNAP help for more info.')\n",
    "align.add_argument('--remap-mismatches', metavar = 'allowed mismatches for remap', required = False, dest = 'remap_mismatches', type = float,\\\n",
    "    help = 'Maximum number of mismatches allowed during remapping of all reads. Treated similarly to --max-mismatches. This is important to control misalignment of reads to similar clusters/tRNAs \\\n",
    "    Note that the SNP index will be updated with new SNPs from the first round of alignment and so this should be relatively small to prohibit misalignment.')\n",
    "align.add_argument('--no-snp-tolerance', required = False, dest = 'snp_tolerance', action = 'store_false',\\\n",
    "    help = 'Disable GSNAP SNP-tolerant read alignment, where known modifications from Modomics are mapped as SNPs. Default is enabled.')\n",
    "\n",
    "\n",
    "outputs = parser.add_argument_group(\"Output options\")\n",
    "outputs.add_argument('-n', '--name', metavar = 'experiment name', required = True, dest = 'name', help = \\\n",
    "    'Name of experiment. Note, output files and indices will have this as a prefix. REQUIRED')\n",
    "outputs.add_argument('--out-dir', metavar = 'output directory', default=\"./\", required = False, dest = 'outdir', help = \\\n",
    "    'Output directory. Default is current directory. Cannot be an existing directory.')\n",
    "outputs.add_argument('--keep-temp', required = False, dest='keep_temp', action = 'store_true', help = \\\n",
    "    'Keeps multi-mapping and unmapped bam files from GSNAP alignments. Default is false.')\n",
    "outputs.add_argument('--overwrite', required = False, dest='overwrite', action = 'store_true', help = \\\n",
    "    'Overwrite existing output directory if overlaps with --our-dir.')\n",
    "\n",
    "bedtools = parser.add_argument_group(\"Bedtools coverage options\")\n",
    "bedtools.add_argument('--min-cov', metavar = 'Minimum coverage per cluster', required = False, dest = 'min_cov', type = restrictedFloat2, default=0.0005, \\\n",
    "    help = \"Minimum coverage per cluster required to include this cluster in coverage plots, modification analysis, and 3'-CCA analysis. \\\n",
    "    Can be a fraction of total mapped reads between 0 and 1, or an integer of absolute coverage. Any cluster not meeting the threshold in 1 or more sample will be excluded. \\\n",
    "    Note that all clusters are included for differential expression analysis with DESeq2. Default = 0.0005 (0.05%% mapped reads).\")\n",
    "bedtools.add_argument('--max-multi', metavar = 'Bedtools coverage multithreading', required = False, dest = 'max_multi', type = int, \\\n",
    "    help = 'Maximum number of bam files to run bedtools coverage on simultaneously. Increasing this number reduces processing time\\\n",
    "    by increasing number of files processed simultaneously. However, depending on the size of the bam files to process and\\\n",
    "    available memory, too many files processed at once can cause termination of mim-tRNAseq due to insufficient memory. If\\\n",
    "    mim-tRNAseq fails during coverage calculation, lower this number. Increase at your own discretion. Default is 3.')\n",
    "\n",
    "remapping = parser.add_argument_group(\"Analysis of unannotated modifications and realignment\")\n",
    "remapping.add_argument('--remap', required = False, dest = 'remap', action = 'store_true',\\\n",
    "    help = 'Enable detection of unannotated (potential) modifications from misincorporation data. These are defined as having a total misincorporation rate\\\n",
    "    higher than the threshold set with --misinc-thresh. These modifications are then appended to already known ones, and read alignment is reperformed.\\\n",
    "    Very useful for poorly annotated species in Modomics. Due to realignment and misincorporation parsing, enabling this option slows the analysis down considerably.')\n",
    "remapping.add_argument('--misinc-thresh', metavar = 'threshold for unannotated mods', dest = 'misinc_thresh', type = restrictedFloat, nargs = '?', default = 0.1,\\\n",
    "    required = False, help = 'Threshold of total misincorporation rate at a position in a cluster used to call unannotated modifications. Value between 0 and 1, default is 0.1  (10%% misincorporation).')\n",
    "\n",
    "parser.add_argument('--version', action='version', version='%(prog)s {}'.format(version.__version__), help = 'Show version number and exit')\n",
    "parser.add_argument('sampledata', help = 'Sample data sheet in text format, tab-separated. Column 1: full path to fastq (or fastq.gz). Column 2: condition/group.')\n",
    "\n",
    "parser.set_defaults(threads=1, out=\"./\", max_multi = 3, min_cov = 0, mito = '', cov_diff = 0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "args = parser.parse_args('--species Hsap --cluster-id 0.95 --threads 1 --min-cov 2000 --max-mismatches 0.1 --control-condition HEK293T -n hg19_test --out-dir hg19_HEK239vsK562_test --overwrite --max-multi 1 --remap --remap-mismatches 0.075 sampleData_HEKvsK562.txt'.split())\n",
    "\n",
    "\n",
    "try:\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(__file__))\n",
    "except NameError:\n",
    "    filepath = '/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/mimseq.py'\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(filepath))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           _                 _   ____  _   _    _                   \n",
      " _ __ ___ (_)_ __ ___       | |_|  _ \\| \\ | |  / \\   ___  ___  __ _ \n",
      "| '_ ` _ \\| | '_ ` _ \\ _____| __| |_) |  \\| | / _ \\ / __|/ _ \\/ _` |\n",
      "| | | | | | | | | | | |_____| |_|  _ <| |\\  |/ ___ \\\\__ \\  __/ (_| |\n",
      "|_| |_| |_|_|_| |_| |_|      \\__|_| \\_\\_| \\_/_/   \\_\\___/\\___|\\__, |\n",
      "                                                                 |_|\n",
      "\n",
      " Modification-induced misincorporation analysis of tRNA sequencing data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(figlet_format('mim-tRNAseq', font='standard'))\n",
    "print(\" Modification-induced misincorporation analysis of tRNA sequencing data\\n\")\n",
    "if args.pretrnas:\n",
    "    if args.cca:\n",
    "        log.warning(\"Disabling CCA analysis in pre-tRNA mode...\")\n",
    "        args.cca = False\n",
    "    if args.cluster:\n",
    "        log.warning(\"Disabling tRNA clustering in pre-tRNA mode...\")\n",
    "        args.cluster = False\n",
    "# Check that control_cond exists in sample data\n",
    "conditions = list()\n",
    "with open(args.sampledata, \"r\") as sampleData:\n",
    "    for line in sampleData:\n",
    "        line = line.strip()\n",
    "        if not line.startswith(\"#\"):\n",
    "            conditions.append(line.split(\"\\t\")[1])\n",
    "if args.control_cond not in conditions:\n",
    "    raise argparse.ArgumentTypeError('{} not a valid condition in {}'.format(args.control_cond, args.sampledata))\n",
    "if not args.species and not (args.trnas or args.trnaout):\n",
    "    parser.error('Must specify valid --species argument or supply -t (tRNA sequences) and -o (tRNAscan out file)!')                        \n",
    "else:\n",
    "    if args.species:\n",
    "        if args.species == 'Hsap':\n",
    "            args.trnas = args.repo_path + \"/data/hg19-eColitK/hg19_eColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg19-eColitK/hg19_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg19-eColitK/hg19-mitotRNAs.fa\"\n",
    "        if args.species == 'Hsap38':\n",
    "            args.trnas = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-detailed.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg38-eColitK/hg38-mitotRNAs.fa\"\n",
    "        if args.species == 'Scer':\n",
    "            args.trnas = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/sacCer3-eColitK/sacCer3-mitotRNAs.fa\"\n",
    "        if args.species == 'Mmus':\n",
    "            args.trnas = args.repo_path + \"/data/mm10-eColitK/mm10_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/mm10-eColitK/mm10_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/mm10-eColitK/mm10-mitotRNAs.fa\"\n",
    "        if args.species == 'Spom':\n",
    "            args.trnas = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_972H-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/schiPomb-eColitK/schiPomb-mitotRNAs.fa\"\n",
    "        if args.species == 'Dmel':\n",
    "            args.trnas = args.repo_path + \"/data/dm6-eColitK/dm6_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/dm6-eColitK/dm6_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/dm6-eColitK/dm6-mitotRNAs.fa\"\n",
    "        if args.species == 'Drer':\n",
    "            args.trnas = args.repo_path + \"/data/danRer11-eColitK/danRer11_eColitK_filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/danRer11-eColitK/danRer11_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/danRer11-eColitK/danRer11-mitotRNAs.fa\"\n",
    "        if args.species == 'Ecol':\n",
    "            args.trnas = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.out\"\n",
    "            args.mito = ''\n",
    "    else:\n",
    "        args.species = args.trnas.split(\"/\")[-1].split(\".\")[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# trnas, trnaout, name, species, out, cluster, cluster_id, cov_diff, posttrans, control_cond, threads, max_multi, snp_tolerance, keep_temp, cca, double_cca, min_cov, mismatches, remap, remap_mismatches, misinc_thresh, mito_trnas, pretrnas, local_mod, sample_data = args.trnas, args.trnaout, args.name, args.species, args.out, args.cluster, args.cluster_id, args.cov_diff, args.posttrans, args.control_cond, args.threads, args.max_multi, args.snp_tolerance, args.keep_temp, args.cca, args.double_cca, args.min_cov, args.mismatches, args.remap, args.remap_mismatches, args.misinc_thresh, args.mito, args.pretrnas, args.local_mod, args.sampledata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=True, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='./', outdir='hg19_HEK239vsK562_test', overwrite=True, posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, repo_path='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq', sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args.local_mod = True\n",
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-05-25 11:16:45,212 [INFO ] mim-tRNAseq v0.3.4.1 run with command:\n",
      "2021-05-25 11:16:45,213 [INFO ] /Users/krdav/anaconda3/envs/mimseq/lib/python3.7/site-packages/ipykernel_launcher.py -f /Users/krdav/Library/Jupyter/runtime/kernel-33d2f614-513a-403d-a3f0-8585d86fae22.json\n",
      "2021-05-25 11:16:45,223 [INFO ] 34 introns registered...\n",
      "2021-05-25 11:16:45,243 [INFO ] Removed 33 introns (including for duplicate tRNAs)\n",
      "2021-05-25 11:16:45,244 [INFO ] 399 cytosolic and 22 mitochondrial tRNA sequences imported\n",
      "2021-05-25 11:16:45,247 [INFO ] Downloading modomics database...\n",
      "2021-05-25 11:16:45,247 [WARNI] Retrieval of Modomics database disabled. Using local files instead...\n",
      "2021-05-25 11:16:45,252 [INFO ] Parsing Modomics JSON data...\n",
      "2021-05-25 11:16:45,259 [INFO ] Number of Modomics entries for Escherichia coli: 43\n",
      "2021-05-25 11:16:45,260 [INFO ] Number of Modomics entries for Homo sapiens: 45\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting: hg19_HEK239vsK562_test/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-05-25 11:17:47,833 [INFO ] Cliped extra CCA from 0 sequences.\n",
      "2021-05-25 11:17:47,859 [INFO ] \n",
      "+------------------------+     \n",
      "| Beginning SNP indexing |    \n",
      "+------------------------+\n"
     ]
    }
   ],
   "source": [
    "# This will be wrapped as a function with \"args\" as input\n",
    "\n",
    "#####################\n",
    "# Results directory #\n",
    "#####################\n",
    "\n",
    "if not args.outdir.endswith(\"/\"):\n",
    "    args.outdir = args.outdir + \"/\"\n",
    "# Main wrapper\n",
    "# Integrity check for output folder argument...\n",
    "\n",
    "try:\n",
    "    os.mkdir(args.outdir)\n",
    "except FileExistsError:\n",
    "    if args.overwrite:\n",
    "        print('Overwriting: {}'.format(args.outdir))\n",
    "        shutil.rmtree(args.outdir)\n",
    "        os.mkdir(args.outdir)\n",
    "    else:\n",
    "        raise FileExistsError(\"Output folder already exists!\")\n",
    "tmp_dir = args.outdir + \"/tmp/\"\n",
    "os.mkdir(tmp_dir)\n",
    "        \n",
    "\n",
    "###########\n",
    "# Logging #\n",
    "###########\n",
    "\n",
    "now = datetime.datetime.now()\n",
    "logging.basicConfig(\n",
    "    format=\"%(asctime)s [%(levelname)-5.5s] %(message)s\",\n",
    "    level=logging.INFO,\n",
    "    handlers=[\n",
    "        logging.FileHandler(args.outdir + \"mim-tRNAseq_{}.log\".format(now.strftime(\"%H-%M-%S\"))),\n",
    "        logging.StreamHandler()\n",
    "    ])\n",
    "log.info(\"mim-tRNAseq v{} run with command:\".format(version.__version__))\n",
    "log.info(\" \".join(sys.argv))\n",
    "\n",
    "########\n",
    "# main #\n",
    "########\n",
    "\n",
    "# Load tRNAs and their modifications\n",
    "trna_obj = TRNA(args)\n",
    "\n",
    "# Match each sequence in tRNA_dict to value in modomics_dict using BLAST\n",
    "log.info(\"\\n+------------------------+ \\\n",
    "    \\n| Beginning SNP indexing |\\\n",
    "    \\n+------------------------+\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "map_round = 1 #first round of mapping\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "class SNPIndex:\n",
    "    '''Object for SNP index.'''\n",
    "\n",
    "    def __init__(self, tRNA_dict, modomics_dict, species_set, tmp_dir, args):\n",
    "        self.tRNA_dict = tRNA_dict\n",
    "        self.modomics_dict = modomics_dict\n",
    "        self.species_set = species_set\n",
    "        self.tmp_dir = tmp_dir\n",
    "        self.inp_outdir = args.outdir\n",
    "        self.inp_cluster = args.cluster\n",
    "        self.inp_cluster_id = args.cluster_id\n",
    "        self.inp_expnam = args.name\n",
    "\n",
    "        # Write BED file for alignment\n",
    "        self.write_bed_file()\n",
    "        # For use in clustering\n",
    "        self.anticodon_list = list({tRNA['anticodon'] for tRNA in self.tRNA_dict.values()})\n",
    "        # Cluster, or not\n",
    "        if self.inp_cluster:\n",
    "            self.clust()\n",
    "        else:\n",
    "            self.no_clust()\n",
    "\n",
    "    \n",
    "    def write_bed_file(self):\n",
    "        seq_records = defaultdict()\n",
    "        tRNAbed_fh = open(self.inp_outdir + self.inp_expnam + \"_maturetRNA.bed\",\"w\")\n",
    "        for tRNA_id, tRNA in self.tRNA_dict.items():\n",
    "            # Initialize list of modified sites for each tRNA\n",
    "            tRNA['modified_idx'] = list()\n",
    "            tRNA['insosine_idx'] = list()\n",
    "            with open(self.tmp_dir + tRNA_id + \".fa\", \"w\") as fh_tRNA:\n",
    "                print('>{}\\n{}'.format(tRNA_id, tRNA['seq']), file=fh_tRNA)\n",
    "            anticodon = re.search('.*tR(NA|X)-.*?-(.*?)-', tRNA_id).group(2)\n",
    "            tRNA['anticodon'] = anticodon\n",
    "            # Find matches to modomics where anticodons match\n",
    "            # and types are the same \n",
    "            # (here regex is used to match anticodons with \".\" in modomics\n",
    "            # to all possible matching sequences from input tRNAs)\n",
    "            match = dict()\n",
    "            for key, mod in self.modomics_dict.items():\n",
    "                if re.match(\"^\" + mod['anticodon'] + \"+$\", anticodon) and tRNA['type'] == mod['type']:\n",
    "                    match[key] = mod\n",
    "            if len(match) > 0:\n",
    "                with open(self.tmp_dir + \"modomics_match.fasta\", \"w\") as match_fh:\n",
    "                    for key, mod in match.items():  \n",
    "                        print('>{}\\n{}'.format(key, mod['unmod_seq']), file=match_fh)\n",
    "\n",
    "                # Call BLAST\n",
    "                blastn_cline = NcbiblastnCommandline(query=fh_tRNA.name, subject=match_fh.name, task='blastn-short', out=self.tmp_dir + \"blast_temp.xml\", outfmt=5)\n",
    "                blastn_cline()\n",
    "\n",
    "                # Parse XML result and store hit with highest bitscore    \n",
    "                blast_rec = NCBIXML.read(open(self.tmp_dir + \"blast_temp.xml\", \"r\"))\n",
    "                maxbit = 0\n",
    "                tophit = ''\n",
    "                for aln in blast_rec.alignments:\n",
    "                    for hsp in aln.hsps:\n",
    "                        if (hsp.bits > maxbit) and (hsp.align_length / aln.length == 1) and (hsp.identities / aln.length == 1):\n",
    "                            maxbit = hsp.bits\n",
    "                            tophit = aln.title.split(' ')[0]\n",
    "\n",
    "                # Return list of all modified positions for the match as long as there is only 1, add to tRNA_dict\n",
    "                if tophit:\n",
    "                    tRNA['modified_idx'].append(match[tophit]['modified_idx'])\n",
    "                    tRNA['insosine_idx'].append(match[tophit]['insosine_idx'])\n",
    "\n",
    "            # Build seqrecord list for writing\n",
    "            seq_records[str(tRNA_id)] = SeqRecord(Seq(tRNA['seq'].upper()), id = str(tRNA_id))\n",
    "            print('{}\\t0\\t{}\\t{}\\t1000\\t+\\n'.format(tRNA_id, str(len(tRNA['seq'])), tRNA_id), file=tRNAbed_fh)\n",
    "        tRNAbed_fh.close()\n",
    "\n",
    "        match_count = sum(len(tRNA['modified_idx'])+len(tRNA['insosine_idx']) for tRNA in self.tRNA_dict.values())\n",
    "        log.info(\"{} total tRNA gene sequences (undetermined and nmt sequences excluded)\".format(len(self.tRNA_dict)))\n",
    "        log.info(\"{} sequences with a match to Modomics dataset\".format(match_count))\n",
    "\n",
    "        with open(str(self.inp_outdir + self.inp_expnam + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "            SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "    def no_clust(self):\n",
    "        pass\n",
    "\n",
    "    def clust(self):\n",
    "        pass\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-05-25 11:41:48,077 [INFO ] 421 total tRNA gene sequences (undetermined and nmt sequences excluded)\n",
      "2021-05-25 11:41:48,083 [INFO ] 36 sequences with a match to Modomics dataset\n"
     ]
    }
   ],
   "source": [
    "snpidx_obj = SNPIndex(trna_obj.tRNA_dict, trna_obj.modomics_dict, trna_obj.species_set, tmp_dir, args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Escherichia_coli_tRNA-Ala-TGC': {'seq': 'GGGGCTATAGCTCAGCDGGGAGAGCGCCTGCTTVGCACGCAGGAG7TCTGCGGTPCGATCCCGCATAGCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGGCTATAGCTCAGCTGGGAGAGCGCCTGCTTTGCACGCAGGAGGTCTGCGGTTCGATCCCGCATAGCTCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ala-GGC': {'seq': 'GGGGCTATAGCTCAGCDGGGAGAGCGCTTGCATGGCATGCAAGAG7TCAGCGGTPCGATCCCGCTTAGCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGGCTATAGCTCAGCTGGGAGAGCGCTTGCATGGCATGCAAGAGGTCAGCGGTTCGATCCCGCTTAGCTCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Arg-ACG': {'seq': 'GCATCCG4AGCTCAGCDGGADAGAGTACTCGG%TICG/ACCGAGCG7XCGGAGGTPCGAATCCTCCCGGATGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'ACG',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GCATCCGTAGCTCAGCTGGATAGAGTACTCGGCTACGAACCGAGCGGTCGGAGGTTCGAATCCTCCCGGATGCACCA',\n",
       "  'insosine_idx': [34]},\n",
       " 'Escherichia_coli_tRNA-Arg-CCG': {'seq': 'GCGCCCGTAGCTCAGCDGGADAGAGCGCTGCC%TCCGKAGGCAGAG7TCTCAGGTPCGAATCCTGTCGGGCGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGCCCGTAGCTCAGCTGGATAGAGCGCTGCCCTCCGGAGGCAGAGGTCTCAGGTTCGAATCCTGTCGGGCGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Arg-TCT': {'seq': 'GCGCCCTTAGCTCAGTTGGATAGAGCAACGAC%T{CT6AGPCGTGGGCCGCAGGTPCGAATCCTGCAGGGCGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGCCCTTAGCTCAGTTGGATAGAGCAACGACCTTCTAAGTCGTGGGCCGCAGGTTCGAATCCTGCAGGGCGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Asn-GTT': {'seq': 'TCCTCTG4AGTTCAGDCGGDAGAACGGCGGACTQTT6APCCGTAT7TCACTGGTPCGAGTCCAGTCAGAGGAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TCCTCTGTAGTTCAGTCGGTAGAACGGCGGACTGTTAATCCGTATGTCACTGGTTCGAGTCCAGTCAGAGGAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Asp-GTC': {'seq': 'GGAGCGG4AGTTCAGDCGGDDAGAATACCTGCCT⊄TC/CGCAGGGG7TCGCGGGTPCGAGTCCCGPCCGTTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGAGCGGTAGTTCAGTCGGTTAGAATACCTGCCTGTCACGCAGGGGGTCGCGGGTTCGAGTCCCGTCCGTTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Cys-GCA': {'seq': 'GGCGCGT4AACAAAGCGGDDATGTAGCGGAPTGCA*APCCGTCTAGTCCGGTPCGACTCCGGAACGCGCCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCA',\n",
       "  'modified_idx': [35],\n",
       "  'unmod_seq': 'GGCGCGTTAACAAAGCGGTTATGTAGCGGATTGCAAATCCGTCTAGTCCGGTTCGACTCCGGAACGCGCCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gln-TTG': {'seq': 'TGGGGTA4CGCCAAGC#GDAAGGCACCGGTJT$TG/PACCGGCATTCCCTGGTPCGAATCCAGGTACCCCAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TGGGGTATCGCCAAGCGGTAAGGCACCGGTTTTTGATACCGGCATTCCCTGGTTCGAATCCAGGTACCCCAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gln-CTG': {'seq': 'TGGGGTA4CGCCAAGC#GDAAGGCACCGGAJTCTG/PPCCGGCATTCCGAGGTPCGAATCCTCGTACCCCAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TGGGGTATCGCCAAGCGGTAAGGCACCGGATTCTGATTCCGGCATTCCGAGGTTCGAATCCTCGTACCCCAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Glu-TTC': {'seq': 'GTCCCCT4CGTCPAGAGGCCCAGGACACCGCCCTSTC/CGGCGGTAACAGGGGTPCGAATCCCCTAGGGGACGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GTCCCCTTCGTCTAGAGGCCCAGGACACCGCCCTTTCACGGCGGTAACAGGGGTTCGAATCCCCTAGGGGACGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-CCC': {'seq': 'GCGGGCG4AGTTCAATGGDAGAACGAGAGCTTCCCAAGCTCTATACGAGGGTPCGATTCCCTTCGCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGCGTAGTTCAATGGTAGAACGAGAGCTTCCCAAGCTCTATACGAGGGTTCGATTCCCTTCGCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-TCC': {'seq': 'GCGGGCATCGTATAATGGCTATTACCTCAGCCT{CCAAGCTGATGATGCGGGTPCGATTCCCGCTGCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGCATCGTATAATGGCTATTACCTCAGCCTTCCAAGCTGATGATGCGGGTTCGATTCCCGCTGCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-GCC': {'seq': 'GCGGGAA4AGCTCAGDDGGDAGAGCACGACCTTGCCAAGGTCGGG7TCGCGAGTPCGAGTCTCGTTTCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGAATAGCTCAGTTGGTAGAGCACGACCTTGCCAAGGTCGGGGTCGCGAGTTCGAGTCTCGTTTCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-His-GTG': {'seq': 'GGTGGCTA4AGCTCAGDDGGDAGAGCCCTGGATTQTG/PPCCAGTT7TCGTGGGTPCGAATCCCATTAGCCACCCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGGCTATAGCTCAGTTGGTAGAGCCCTGGATTGTGATTCCAGTTGTCGTGGGTTCGAATCCCATTAGCCACCCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ile-GAT': {'seq': 'AGGCTTGTAGCTCAGGDGGDDAGAGCGCACCCCTGAT6AGGGTGAG7XCGGTGGTPCAAGTCCACPCAGGCCTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAT',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'AGGCTTGTAGCTCAGGTGGTTAGAGCGCACCCCTGATAAGGGTGAGGTCGGTGGTTCAAGTCCACTCAGGCCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ile-CAT': {'seq': 'GGCCCCT4AGCTCAGT#GDDAGAGCAGGCGACT}AT6APCGCTTG7XCGCTGGTPCAAGTCCAGCAGGGGCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GGCCCCTTAGCTCAGTGGTTAGAGCAGGCGACTCATAATCGCTTGGTCGCTGGTTCAAGTCCAGCAGGGGCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-CAG': {'seq': 'GCGAAGGTGGCGGAADD#GDAGACGCGCTAGCTTCAGKPGPTAGTGTCCTTACGGACGTGGGGGTPCAAGTCCCCCCCCTCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGAAGGTGGCGGAATTGGTAGACGCGCTAGCTTCAGGTGTTAGTGTCCTTACGGACGTGGGGGTTCAAGTCCCCCCCCTCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-GAG': {'seq': 'GCCGAGGTGGTGGAADD#GDAGACACGCTACCTTGAGKPGGTAGTGCCCAATAGGGCTTACGGGTPCAAGTCCCGTCCTCGGTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCGAGGTGGTGGAATTGGTAGACACGCTACCTTGAGGTGGTAGTGCCCAATAGGGCTTACGGGTTCAAGTCCCGTCCTCGGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-TAA': {'seq': 'GCCCGGA4GGTGGAADC#GDAGACACAAGGGAPT)AA*APCCCTCGGCGTTCGCGCTGTGCGGGTPCAAGTCCCGCTCCGGGTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCCGGATGGTGGAATCGGTAGACACAAGGGATTTAAAATCCCTCGGCGTTCGCGCTGTGCGGGTTCAAGTCCCGCTCCGGGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-CAA': {'seq': 'GCCGAAG4GGCGAAADC#GDAGACGCAGTTGAPTBAA*APCAACCGTAGAAATACGTGCCGGTPCGAGTCCGGCCTTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCGAAGTGGCGAAATCGGTAGACGCAGTTGATTCAAAATCAACCGTAGAAATACGTGCCGGTTCGAGTCCGGCCTTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Lys-TTT': {'seq': 'GGGTCGTTAGCTCAGDDGGDAGAGCAGTTGACTSTT6APCAATTG7XCGCAGGTPCGAATCCTGCACGACCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GGGTCGTTAGCTCAGTTGGTAGAGCAGTTGACTTTTAATCAATTGGTCGCAGGTTCGAATCCTGCACGACCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Met-CAT': {'seq': 'GGCTACG4AGCTCAGDD#GDDAGAGCACATCACTMAT6APGATGGG7XCACAGGTPCGAATCCCGTCGTAGCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GGCTACGTAGCTCAGTTGGTTAGAGCACATCACTCATAATGATGGGGTCACAGGTTCGAATCCCGTCGTAGCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Phe-GAA': {'seq': 'GCCCGGA4AGCTCAGDCGGDAGAGCAGGGGAPTGAA*APCCCCGT7XCCTTGGTPCGATTCCGAGTCCGGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAA',\n",
       "  'modified_idx': [36, 46],\n",
       "  'unmod_seq': 'GCCCGGATAGCTCAGTCGGTAGAGCAGGGGATTGAAAATCCCCGTGTCCTTGGTTCGATTCCGAGTCCGGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Pro-CGG': {'seq': 'CGGTGATTGGCGCAGCCTGGDAGCGCACTTCGTTCGGKACGAAGGG7TCGGAGGTPCGAATCCTCTATCACCGACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'CGGTGATTGGCGCAGCCTGGTAGCGCACTTCGTTCGGGACGAAGGGGTCGGAGGTTCGAATCCTCTATCACCGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Sec-TCA': {'seq': '_AAGATCG4CGTCTCCGGDGAGGCGGCTGGACTTCA+ATCCAGTTGGGGCCGC_GCGGTCCCGGGCAGGTPCGACTCCTGTGATCTT_GCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'NAAGATCGTCGTCTCCGGTGAGGCGGCTGGACTTCAAATCCAGTTGGGGCCGCNGCGGTCCCGGGCAGGTTCGACTCCTGTGATCTTNGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-TGA': {'seq': 'GGAAGTG4GGCCGAGC#GDDGAAGGCACCGGTBTVGA*AACCGGCGACCCGAAAGGGTTCCAGAGTPCGAATCTCTGCGCTTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGAAGTGTGGCCGAGCGGTTGAAGGCACCGGTCTTGAAAACCGGCGACCCGAAAGGGTTCCAGAGTTCGAATCTCTGCGCTTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-CGA': {'seq': 'GGAGAGATGCCGGAGC#GCDGAACGGACCGGTCTCGA*AACCGGAGTAGGGGCAACTCTACCGGGGGTPCAAATCCCCCTCTCTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGAGAGATGCCGGAGCGGCTGAACGGACCGGTCTCGAAAACCGGAGTAGGGGCAACTCTACCGGGGGTTCAAATCCCCCTCTCTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-GCT': {'seq': 'GGTGAGG4GGCCGAGAGGCDGAAGGCGCTCCC%TGCT6AGGGAGTATGCGGTCAAAAGCTGCATCCGGGGTPCGAATCCCCGCCTCACCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGAGGTGGCCGAGAGGCTGAAGGCGCTCCCCTGCTAAGGGAGTATGCGGTCAAAAGCTGCATCCGGGGTTCGAATCCCCGCCTCACCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-GGA': {'seq': 'GGTGAGG4GTCCGAGT#GDDGAAGGAGCACGCCTGGAAAGPGTGTATACGGCAACGTATCGGGGGTPCGAATCCCCCCCTCACCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGAGGTGTCCGAGTGGTTGAAGGAGCACGCCTGGAAAGTGTGTATACGGCAACGTATCGGGGGTTCGAATCCCCCCCTCACCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Thr-GGT': {'seq': 'GCTGATATAGCTCAGDDGGDAGAGCGCACCCTTGGTEAGGGTGAG7TCGGCAGTPCGAATCTGCCTATCAGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCTGATATAGCTCAGTTGGTAGAGCGCACCCTTGGTAAGGGTGAGGTCGGCAGTTCGAATCTGCCTATCAGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Trp-CCA': {'seq': 'AGGGGCG4AGTTCAADDGGDAGAGCACCGGTBTCCA*AACCGGGT7TTGGGAGTPCGAGTCTCTCCGCCCCTGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCA',\n",
       "  'modified_idx': [36],\n",
       "  'unmod_seq': 'AGGGGCGTAGTTCAATTGGTAGAGCACCGGTCTCCAAAACCGGGTGTTGGGAGTTCGAGTCTCTCCGCCCCTGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Tyr-GTA': {'seq': 'GGTGGGG44CCCGAGC#GCCAAAGGGAGCAGACTQPA*APCTGCCGTCACAGACTTCGAAGGTPCGAATCCTTCCCCCACCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGTGGGGTTCCCGAGCGGCCAAAGGGAGCAGACTGTAAATCTGCCGTCACAGACTTCGAAGGTTCGAATCCTTCCCCCACCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Val-TAC': {'seq': 'GGGTGAT4AGCTCAGCDGGGAGAGCACCTCCCTVAC=AGGAGGGG7TCGGCGGTPCGATCCCGTCATCACCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGTGATTAGCTCAGCTGGGAGAGCACCTCCCTTACAAGGAGGGGGTCGGCGGTTCGATCCCGTCATCACCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Val-GAC': {'seq': 'GCGTTCA4AGCTCAGDDGGDDAGAGCACCACCTTGACATGGTGGGG7XCGTTGGTPCGAGTCCAATTGAACGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAC',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GCGTTCATAGCTCAGTTGGTTAGAGCACCACCTTGACATGGTGGGGGTCGTTGGTTCGAGTCCAATTGAACGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-iMet-CAT': {'seq': 'CGCGGGG4GGAGCAGCCTGGDAGCTCGTCGGGBTCATAACCCGAAGATCGTCGGTPCAAATCCGGCCCCCGCAACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'CGCGGGGTGGAGCAGCCTGGTAGCTCGTCGGGCTCATAACCCGAAGATCGTCGGTTCAAATCCGGCCCCCGCAACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ala-AGC': {'seq': 'GGGGAATTALCTCAAADGGDAGAGCRCTCGCJTIGCOP#CGAGAG7TAGCGGGAPCG\"TGCCCGCATTCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGC',\n",
       "  'modified_idx': [25, 36, 57],\n",
       "  'unmod_seq': 'GGGGAATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGTAGCGGGATCGATGCCCGCATTCTCCACCA',\n",
       "  'insosine_idx': [33]},\n",
       " 'Homo_sapiens_tRNA-Glu-CTC': {'seq': 'TCCCTGGTGLTCPAGTGGDPAGGATTCGGCGCTCTCACCGCCGCGGC??GGG\\\\PCGATTCCCGGTCAGGGAACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TCCCTGGTGGTCTAGTGGTTAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Phe-GAA': {'seq': 'GCCGAAATALCTC\"GDDGGGAGAGCRPPAGABT#AAWAPCTAAAG7DC?CTGGTPCG\"TCCCGGGTTTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAA',\n",
       "  'modified_idx': [13, 25, 36, 57],\n",
       "  'unmod_seq': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGTCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gly-CCC': {'seq': 'GCGBCLCTGGTGPAGTGGDATCATGCAAGAJTCCCATZCTTGCGAC??GGGTPCG\"TTCCCGGGCGGCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCC',\n",
       "  'modified_idx': [55],\n",
       "  'unmod_seq': 'GCGCCGCTGGTGTAGTGGTATCATGCAAGATTCCCATTCTTGCGACCCGGGTTCGATTCCCGGGCGGCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gly-GCC': {'seq': 'GCAJTLGTGGTTCAGTGGDAGAATTCTCGCCTGCCA?GCGGGAGG???GGGTPCG\"TTCCCGGCCAATGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCC',\n",
       "  'modified_idx': [55],\n",
       "  'unmod_seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-His-GTG': {'seq': ';GCCGTGATCGTAPAGDGGDDAGTACTCTGCGPTGTGGCCGCAGCAA??TCGGTPCG\"ATCCGAGTCACGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTG',\n",
       "  'modified_idx': [57],\n",
       "  'unmod_seq': 'GGCCGTGATCGTATAGTGGTTAGTACTCTGCGTTGTGGCCGCAGCAACCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Met-CAT': {'seq': \"GCCCTCTTAGCGCAGTGGG'AGCGCGTCAGTCTCATAATCTGAAGGTCCTGAGTTCGAGCCTCAGAGAGGGCA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [19],\n",
       "  'unmod_seq': 'GCCCTCTTAGCGCAGTGGGCAGCGCGTCAGTCTCATAATCTGAAGGTCCTGAGTTCGAGCCTCAGAGAGGGCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Asn-GTT': {'seq': 'GTCTCTGTKLCGCAADCGGDXAGCGCRPPCGGCTQTT6ACCGAAAG7DTGGTGG.PCG\"GCCCACCCAGGGACGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTT',\n",
       "  'modified_idx': [8, 20, 26, 58],\n",
       "  'unmod_seq': 'GTCTCTGTGGCGCAATCGGTTAGCGCGTTCGGCTGTTAACCGAAAGGTTGGTGGNTCGAGCCCACCCAGGGACGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gln-TTG': {'seq': 'GGTCCCATKGTGPAAT#GDDAGCACTCPGGABTNTGAAPCCAGCGAT??GAGPPCA\"ATCTCGGTGGGACCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTG',\n",
       "  'modified_idx': [8, 56],\n",
       "  'unmod_seq': 'GGTCCCATGGTGTAATGGTTAGCACTCTGGACTTTGAATCCAGCGATCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-TGA': {'seq': \"GCAGCGATGGCCGAGTGGTTAAGGCGTTGGA'TTGAAATCCAATGG'GTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GCAGCGATGGCCGAGTGGTTAAGGCGTTGGACTTGAAATCCAATGGCGTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-iMet-CAT': {'seq': 'AGCAGAGTKLCGCAGCGGAAGCGTLCTGGGCCCAT6ACCCAGAG7D?GATGGATCG\"AACCATCCTCTGCTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [8, 56],\n",
       "  'unmod_seq': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGTCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Tyr-GTA': {'seq': 'CCTTCLATALCTCAGCDGGXAGAGCRRAGGACT9PAKA]CCTTAG7D?GCTGGTPCG\"TTCCGGCTCGAAGGACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTA',\n",
       "  'modified_idx': [19, 25, 26, 36, 57],\n",
       "  'unmod_seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGTCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Sec-TCA': {'seq': 'GCCCGGAT_CCTCAGTGGTCTGGGGTGCAGGCTNCA+ACCTGTAGCTGTCTAGCGACAGAGTGGTPCA\"TTCCA_TTTCGGGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCA',\n",
       "  'modified_idx': [68],\n",
       "  'unmod_seq': 'GCCCGGATNCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGCGACAGAGTGGTTCAATTCCANTTTCGGGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ile-GAT': {'seq': 'AGAAATATKTCTGATAAAAGARPPACTTTGAT6GAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'GAT',\n",
       "  'modified_idx': [8, 21],\n",
       "  'unmod_seq': 'AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-TAA': {'seq': 'GTTAAGATKLCAGAGCCCGGDAATCGCAPAAAACTÊAAAACTTTACAGT?AGAGGTPCA\"TTCCTCTTCTTAACACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TAA',\n",
       "  'modified_idx': [8, 59],\n",
       "  'unmod_seq': 'GTTAAGATGGCAGAGCCCGGTAATCGCATAAAACTTAAAACTTTACAGTCAGAGGTTCAATTCCTCTTCTTAACACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-GCT': {'seq': \"GACGAGGTGGCCGAGTGGTTAAGGCGATGGACTGCTEATCCATTGT'CTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GACGAGGTGGCCGAGTGGTTAAGGCGATGGACTGCTAATCCATTGTCCTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Lys-TTT': {'seq': 'CACTGTAA\"LCTAACTTAGCAPPAACCT∃TT6AGTTAAAGATTAAGAGAACCAA_CTCTTTACAGTGACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TTT',\n",
       "  'modified_idx': [8],\n",
       "  'unmod_seq': 'CACTGTAAAGCTAACTTAGCATTAACCTTTTAAGTTAAAGATTAAGAGAACCAANCTCTTTACAGTGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Pro-TGG': {'seq': 'CAGAGAAT\"GTTTAAATTAGAATCPPAGCPTTGGKPGCTAATGGTGGAGTPAAAGACTTTTTCTCTGACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TGG',\n",
       "  'modified_idx': [8, 34],\n",
       "  'unmod_seq': 'CAGAGAATAGTTTAAATTAGAATCTTAGCTTTGGGTGCTAATGGTGGAGTTAAAGACTTTTTCTCTGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Asp-GTC': {'seq': 'AAGGTATT\"LAAAAACCATTTCAPAACTTTQTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'GTC',\n",
       "  'modified_idx': [8],\n",
       "  'unmod_seq': 'AAGGTATTAGAAAAACCATTTCATAACTTTGTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-TAG': {'seq': 'ACTTTTAA\"LGATAACAGCTATCCAPPGGPCTTAGKCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TAG',\n",
       "  'modified_idx': [8, 35],\n",
       "  'unmod_seq': 'ACTTTTAAAGGATAACAGCTATCCATTGGTCTTAGGCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-TAG': {'seq': 'GCGGGAGTGGCGAAADD#GTAGACGCACCAGATTVAGKPTCTGGCGCCGCAAGGTGTGCGAGTPCAAGTCTCGCCTCCCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGGGAGTGGCGAAATTGGTAGACGCACCAGATTTAGGTTCTGGCGCCGCAAGGTGTGCGAGTTCAAGTCTCGCCTCCCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Thr-TGT': {'seq': 'GCCGACTTAGCTCAGTAGGDAGAGCAACTGABTVGT6ATCAGTAG7TCACCAGTPCGATTCCGGTAGTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCCGACTTAGCTCAGTAGGTAGAGCAACTGACTTGTAATCAGTAGGTCACCAGTTCGATTCCGGTAGTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-CAG': {'seq': \"GTCAGGATGGCCGAGCGGTCTAAGGCGCTGCGTTCAGGTCGCAGTC'CCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAG',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GTCAGGATGGCCGAGCGGTCTAAGGCGCTGCGTTCAGGTCGCAGTCCCCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-CAA': {'seq': 'GTCAGGATGGCCGAGTGGTCTAAGGCGCCAGAC?CAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GTCAGGATGGCCGAGTGGTCTAAGGCGCCAGACCCAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Arg-CCT': {'seq': \"GCCCCAGTGGCCTAATGGATAAGGCACTGGC'TCCTAAGCCAGGGATTGTGGGTTCGAGTCCCACCTGGGGTA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GCCCCAGTGGCCTAATGGATAAGGCACTGGCCTCCTAAGCCAGGGATTGTGGGTTCGAGTCCCACCTGGGGTA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Arg-TCT': {'seq': \"GGCTCCGTGGCGCAATGGATAGCGCATTGGA'TTCTAATTCAAAGGTTCCGGGTTCGAGTCCCGGCGGAGTCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCTCCGTGGCGCAATGGATAGCGCATTGGACTTCTAATTCAAAGGTTCCGGGTTCGAGTCCCGGCGGAGTCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-AGA': {'seq': \"GTAGTCGTGGCCGAGTGGTTAAGGCGATGGA'TAGAAATCCATTGG'GTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GTAGTCGTGGCCGAGTGGTTAAGGCGATGGACTAGAAATCCATTGGCGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-CGA': {'seq': \"GCTGTGATGGCCGAGTGGTTAAGGCGTTGGA'TCGAAATCCAATGG'GTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GCTGTGATGGCCGAGTGGTTAAGGCGTTGGACTCGAAATCCAATGGCGTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-AGT': {'seq': \"GGCGCCGTGGCTTAGTTGGTTAAAGCGCCTG'CTAGTAAACAGGAGATCCTGGGTTCGAATCCCAGCGGTGCCT\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCGCCGTGGCTTAGTTGGTTAAAGCGCCTGCCTAGTAAACAGGAGATCCTGGGTTCGAATCCCAGCGGTGCCT',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-CGT': {'seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGTCTCGTAAACAGGAGATCCTGGGTTCGACTCCCAGTGGGGCCT',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGTCTCGTAAACAGGAGATCCTGGGTTCGACTCCCAGTGGGGCCT',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-TGT': {'seq': \"GGCTCTATGGCTTAGTTGGTTAAAGCGCCTG'CTTGTAAACAGGAGATCCTGGGTTCGAATCCCAGTAGAGCCT\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGCCTTGTAAACAGGAGATCCTGGGTTCGAATCCCAGTAGAGCCT',\n",
       "  'insosine_idx': []}}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trna_obj.modomics_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'hg19_HEK239vsK562_test//tmp/'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(<bound method TRNA.__dd of <mimseq.tRNAtools.TRNA object at 0x10fbc7e90>>,\n",
       "            {'Escherichia_coli_str_K_12_substr_MG1655_tRNA-eColiLys-TTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGTCGTTAGCTCAGTTGGTAGAGCAGTTGACTTTTAATCAATTGGtCGCAGGTTCGAATCCTGCACGACCCA',\n",
       "                          'species': 'Escherichia coli',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGtCCTGGGTTCGATCCCCAGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCACGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-11-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGATGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-12-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-13-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-14-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCAATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-15-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGTGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGTGGGATCGATGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-16-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCCCAAGTGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-17-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCGCAAGTGGTAGAGTGCTTGCTTAGCATGCAAGAGGtAGTGGGATCGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-18-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTGGCTCAAGCGGTAGAGCGCTTGCTTAGCATGCAAGAGGtAGCAGGATCGACGCCTGCACTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-19-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGATTAGCTCAAGCGGTAGAGCGCCTGCTTAGCATGCAAGAGGtAGCAGGATCGATGCCTGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCCGGGTTCAATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-20-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGATCAGTGGTAGGGCGCACGCTTAGCATGCATGAGGcCCTGGGTCAATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-21-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGATTAGCTCAAGCGGTAGGGTGCCTGCTTAGCATGCAAGAGGtAGCAGGATCGACGCCTGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-22-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAAGCGGTAGAGCGCTCCCTTAGCATGCGAGAGGtAGCGGGATCGACGCCCCCATTCTCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-23-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGAcAGAGCACGTGGTTAGCATGCGTGAGGtCCTGCGTTCAACTTCCAGTATTTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGTACGAGGtCCCGGGTTCAATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTAGCATGCATGAGGtCCCGGGTTCGATCCCCAGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCTGGGTTCAATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGCGGTAGAGCGCGTGCTTAGCATGCACGAGGtCCTGGGTTCAATCCCCAATACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-7-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTAGCATGCACGAGGcCCCGGGTTCAATCCCTGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-8-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGATGCCCGCATCCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-9-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGAATTAGCTCAGGCGGTAGAGCGCTCGCTTAGCATGCGAGAGGtAGCGGGATCGACGCCCGCATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-AGC-chr6-93': defaultdict(None,\n",
       "                         {'seq': 'GAAGAAGTAGCTCAAGTGGTCGAGTGCTCGCTTAGCATGTGAGAGGtAGTGGGATTGATGCCCACATTCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTCGCATGTATGAGGtCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTCGCATGTATGAGGcCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCGCGCTTCGCATGTGTGAGGtCCCGGGTTCAATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCGTGCTTCGCATGTACGAGGcCCCGGGTTCGACCCCCGGCTCCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-CGC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGATCAGTGGTAGAGCGCATGCTTCGCATGTACGAGGtCCCTGGTTCAATCCCTGGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-GGC-chr19-3': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTGTGGCTTTGTGGGTtAAAGTACTTGTCTGGCAAACAGGAGAtGCTGGAtTTCAATCCCAGCAGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGtCCCGGGTTCGATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGtCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCCCGGGTTCGATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATGTAGCTCAGTGGTAGAGCGCATGCTTTGCACGTATGAGGcCCCGGGTTCAATCCCCGGCATCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCTCGGGTTCGATCCCCGACACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCACATGCTTTGCATGTGTGAGGcCCCGGGTTCGATCCCCGGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-7-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGCATGCTTTGCATGTATGAGGcCTCGGTTCGATCCCCGACACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-8-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCGGATGCTTTGCATGTATGAGAcTTTGGGTTGGATCCCCAGCACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Ala-TGC-9-1': defaultdict(None,\n",
       "                         {'seq': 'GTGGATGTAGTTTAGTGGTAGAACGCGCGCTTTGCATGTATGAGGtCCCGGTTTCGATCCCTGGCGTTTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCCAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ACG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-ACG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGGCCAGTGGCGCAATGGAtAACGCGTCTGACTACGGATCAGAAGAtTCTAGGTTCGACTCCTGGCTGGCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ACG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGATTCCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GACCCAGTGGCCTAATGGAtAAGGCATCAGCCTCCGGAGCTGGGGAtTGTGGGTTCGAGTCCCATCTGGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCCAGTGGCCTAATGGAtAAGGCACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCTGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCCAGTGGCCTAATGGAtAAGGCACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCTGGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCCGGTGGCCTAATGGAtAAGGCATTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCACCCGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCCAGTGGCCTAATGGAtAAGGCATTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTCCCATCTGGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-CCT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCCAGTGGCCTGATGGAtAAGGTACTGGCCTCCTAAGCCAGGGAtTGTGGGTTCGAGTTCCACCTGGGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGCGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGCAGGTTCGAGTCCTGCCGCGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTCCGTGGTTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GACCGCGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAGTCCCTTCGTGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-4-1': defaultdict(None,\n",
       "                         {'seq': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTTCGTGGTTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-5-1': defaultdict(None,\n",
       "                         {'seq': 'GACCACGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAGAAGAtTGAGGGTTCGAATCCCTTCGTGGTTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCG-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGTGTGGCCTAATGGAtAAGGCGTCTGACTTCGGATCAAAAGAtTGCAGGTTTGAGTTCTGCCACGGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTCCGGGTTCGAGTCCCGGCGGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGTGGGTTCGAATCCCACCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGTGGGTTCGAGTCCCACCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATGGAcgAGCGCGCTGGACTTCTAATCCAGAGGtTCCGGGTTCGAGTCCCGGCAGAGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTGTGGCGCAATGGAtAGCGCATTGGACTTCTAATTCAAAGGtTGCGGGTTCGAGTCCCTCCAGAGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Arg-TCT-chr5-4': defaultdict(None,\n",
       "                         {'seq': 'GGCAGCATAGCAGAGTGGTtCAGGTTACAGGTTCTAGATGTAAACTGAGTTCAAATCCCAGTTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-ATT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTcAGAGCGTTCGGCTATTAACCGAACGGtGAGTAGTTCAAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGATCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-10-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAGGtTGGCGGTTCGAACCCACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-11-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCCGCTGTTAACCGAAAGCtTGGTGGTTCGAGCCCACCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-12-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAAGtTGGTGGTTCGAACACACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-13-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCGGCTGTTAACCATAAGGtTGGTGGTTAGAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-14-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCCGCTGTTAACCGAAAGCtTGGTGGTTTGAGCCCACCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-15-1': defaultdict(None,\n",
       "                         {'seq': 'ATCTCCGTGGAGCAATTGGTtAGCGCGTTCGGCTGTTAACCGGAAAGtTGGTGGTTCGAGCCTACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-16-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTTGACTGTTAACTGAAAGGtTGGTGGTGCAAGCCCATCCAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-17-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCTGTAGCGCGATCGGTtAGCGCCTTCGGCTGTTAAACGAAAGGtTGGTGGTTCGTTCCCACCCCGGGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-18-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGTGCAATCGGTtAGCGCGTTCGGCTGTTAACCATAAGGtTGGTGGTTACAGACCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-19-1': defaultdict(None,\n",
       "                         {'seq': 'ATCTCCGTGGAGCAATTGGTtAGCGCGTTCGGCCGTTAACCGGAAAGtTGGTGGTTCGAGCCTACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-21-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCACAATCGGTtAGCGCGTTCGGCTGTTAATCTAGAGGtTGGTGGTTAGAGCCCACTGAGGGATGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-22-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCACAATCGGTtAGAGCGTTCGGCTGTTAATCTAAAGGtTGGTGGCTAGAACCCACTGAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCATTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACCGAAAGAtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACTGAAAGGtTGGTGGTTCGAGCCCACCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATGGGTtAGCGCGTTCGGCTGTTAACCGAAAGGtTGGTGGTTCGAGCCCATCCAGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-7-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGTAGTCGGTtAGCGCGTTCGGCTGTTAACCGAAAAGtTGGTGGTTCGAGCCCACCCAGGAACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-8-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGCtAGCGCGTTTGGCTGTTAACTAAAAGGtTGGTGGTTCGAACCCACCCAGAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asn-GTT-9-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTCTGTGGCGCAATCGGTtAGCGCGTTCGGCTGTTAACTGAAAGGtTAGTGGTTCGAGCCCACCCGGGGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_tRNA-Asp-ATC-chr6-103': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGTGGTAGAGAGTGTACTTATCATGCACGAGGtCTTGGGctGATTcCCCAGTACCTCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-1-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTCGTTAGTATAGTGGTtAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCAATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-2-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTCGTTAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-3-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTCGTTAGTATAGTGGTgAGTGTCCCCGTCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCGACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-4-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTCGTTAGTATGGTGGTgAGTATCCCTGCCTGTCACGCGGGAGaCCGGGGTTCGATTCCCCAACGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-5-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTCATCAGTATAGTGGTgAGTATCCCCGCCTGTCACGCGGGAGaCTGGGGTTCGATTCCCTGAGGAGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-6-1': defaultdict(None,\n",
       "                         {'seq': 'TTCTTGTTAATATAGTGGTgAGTATTCCCACCTGTCATGCGGGAGACGGGGTTCAATTCCCTGATGGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Asp-GTC-8-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTTGTTAGTATAGTGGTgAGTGTTTCTGCCTGTCATGTGGAGACTGGAGTTTGAGTCCCCAACAGGGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_tRNA-Cys-ACA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTGTATAGCTCAGTGGTAGAGCATTTGACTACAGCTTTAAAGTCATAtGTAGcTGGGTTCAAATCCTGTTTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ACA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGCATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-11-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTTAGCGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-12-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTTAGGGGTAGAGCATTTGACTGCAGATCAAAAGGtCCCTGGTTCAAATCCAGGTGCCCCTTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-13-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-14-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAAGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-15-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCTCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-16-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCACTTGACTGCAGATCAAGAAGtCCTTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-17-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGATATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-18-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGTTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-19-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAAATCAAGAGGtCCCTGATTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-20-1': defaultdict(None,\n",
       "                         {'seq': 'GGGCGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-21-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCACAGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCTGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-22-1': defaultdict(None,\n",
       "                         {'seq': 'GGGCGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCAGTTCAAATCTGGGTGCCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-23-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCACAGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGTTACTCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-24-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTAGGGCTCAGGGAtAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCGAATCTAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-25-1': defaultdict(None,\n",
       "                         {'seq': 'GGTATATCTCAGGGGGcAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTGAAATCCGGGTGCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCACTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGTGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTGTAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-7-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGTGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-8-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCCGGTTCAAATCCGGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-9-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr11-21': defaultdict(None,\n",
       "                         {'seq': 'GAAGGTATAGCTCAGGGGTAGAGCATTTGACTGCAGATCAAGAAAATACTCCTTTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr12-14': defaultdict(None,\n",
       "                         {'seq': 'GAGGGTATAGCTCAGAGGTAGAGCACTGGACTGCAGCTTCAGTTCAGCTGCCTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr5-2': defaultdict(None,\n",
       "                         {'seq': 'AGGGGTATAGCTCAGTGGTAGAGCATTTGACTGCAGATCAAAAAAAAAAAGTTTCCCCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr7-25': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTGAGGAGTAAACCATTTGGCTGCAGATCAAGTGGtCCCTGGTTCAAATATGGGTGCCCTCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chr8-6': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAACATTAGACTGCAAATCTGAAGaTGGAGGTTAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Cys-GCA-chrX-3': defaultdict(None,\n",
       "                         {'seq': 'GTGGGTATAGCTCAGAGGTAGAGCATTCAACTGCAGATCGAAGTGgCTGTACCATTTTGCATTCCCACCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTgACCACTTTGGACTCTGAATACAGTGATCAGAGTTCAAGTCTCACTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-11-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTgAGGGCTTTGGACTCTGACTACAGTGaTCAGAGTTCAAGTCTCAGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-12-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTaAGCACCCTGGACTCTGAATCCAGCAaCCAGAGTTCCAGTCTCAGCGtGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-13-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGTGTAGTCTACTGGTTAAACGCTTGGgCTCTGACATTAAcGtCCTGGGTTCAAATCCCAGCTTTGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-16-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCAGTGGTgcAGGGCATGGCTCTGAcACCTGGCGGCCTGGGTTCAAATCCCAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTgAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTaAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCGAGTCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTtAGCACTCTGGACTCTGAATCCGGTAaTCCGAGTTCAAATCTCGGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCCCATGGTGTAATGGTcAGCACTCTGGACTCTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-7-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTaAGCACTCTGGACTCTGAATCCAGCCATCTGAGTTCGAGTCTCTGTGGAACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-8-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTgAGCACTTTGGACTCTGAATACAGTGATCAGAGTTCAAGTCTCACTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-9-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGgGTTAATGGTgAGCACCCTGGACTCTGAATCAAGCGaTCCGAGTTCAAATCTCGGTGGTACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-CTG-chr19-4': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCATGGTGTAATGGTgAGCACTCTGGACTCTGAATCCAGAAATACATTCAAAGAATTAAGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCAaTCCGAGTTCGAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCCCATGGTGTAATGGTtAGCACTCTGGACTTTGAATCCAGCGaTCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGTCCCATGGTGTAATGGTtAGCACTCTGGGCTTTGAATCCAGCAaTCCGAGTTCGAATCTTGGTGGGACCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTGTGTACCTCAGTGGGcAAGGGTATGGACTTTGAAGCCAGACTaTTTGGGTTCAAATCCCAGCTTGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-6-1': defaultdict(None,\n",
       "                         {'seq': 'GACCATGTGGCCTAAGGGAaAAGACATCTCACTTTGGGTCAGAAGAtTGAGGGTTCAAGTCCTTTCATGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Gln-TTG-chr8-10': defaultdict(None,\n",
       "                         {'seq': 'GGTACAGTGTTAAAGGGGagaAAAATTGCTGACTTTGAATaCAGTAGaCCTAGGTTTGAATCCTGGCTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-1-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-2-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCTGGTGGTCTAGTGGTtAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGAAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-3-1': defaultdict(None,\n",
       "                         {'seq': 'CCCCTGGTGGTCTAGTGCTtAGGATTCGGTGCTCTCACCGCTGCTGCCTGCGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-CTC-5-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTTGATGTCTAGTGGTtAGGATTTGGTGCTCTCACTGCAGCAGCCTGGGTTCATTTCTCAGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-1-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCATATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGTGGCCCGGGTTCGACTCCCGGTATGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-10-1': defaultdict(None,\n",
       "                         {'seq': 'TCCGTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCTGCAGCTCGAGTTCGATTCCTGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-11-1': defaultdict(None,\n",
       "                         {'seq': 'CCCTGTGGTCTAGTGGCtAAGACTTTGTGCTTTCATTGCTGCAtCCTAGGTTCAATTCCCAGTCAGGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-2-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCACATGGTCTAGCGGTtAGGATTCCTGGTTTTCACCCAGGCGGCCCGGGTTCGACTCCCGGTGTGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-3-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCGCGGCCCGGGTTCGATTCCCGGCCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-4-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTGTAGTGGTgAGCACAGCTGCCTTTCAAGCAGTTAaCGCGGGTTCGATTCCCGGGTAACGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-6-1': defaultdict(None,\n",
       "                         {'seq': 'TCCTTGGTGGTCTAGTGGCtAGGATTCGGTGCTTTCACCTGTGCGGCCCGGGTTCAATTCCCGATGAAGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-7-1': defaultdict(None,\n",
       "                         {'seq': 'TGTCTGGTGGTCAAGTGGCtAGGATTTGGCGCTTTCACTGCCGCGGCCCGCGTTCGATTCCCGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-8-1': defaultdict(None,\n",
       "                         {'seq': 'TCCCTGGTGGTCTAGTGGCtAGGATTCGGCGCTTTCACCGCCTGCAGCTCGAGTTCGATTCCTGGTCAGGGAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Glu-TTC-9-1': defaultdict(None,\n",
       "                         {'seq': 'GCAATGGTGGTTCAGTGGTAGAATTCTCGCCTTTCACACAGGAGaCCCGGGTTCAATTCCTGACCCATGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTCCCACGCGGGAGaCCCGGGTTCAATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCGCCGCTGGTGTAGTGGTATCATGCAAGATTCCCATTCTTGCGaCCCGGGTTCGATTCCCGGGCGGCGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCATTGGTGGTTCAATGGTAGAATTCTCGCCTCCCACGCAGGAGaCCCAGGTTCGATTCCTGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTTTAGTGGTAGAATTCTCGCCTCCCATGCGGGAGaCCCGGGTTCAATTCCCGGCCACTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTTGGTGGTGCAGTGGTAGAATTCTCGCCTCCCACGTGGGAGaCCCGGGTTCAATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGTGGTTCAGTGGTAGAATTCTCGCCTCCCACGCGGGAGaCCCGGGTTTAATTCCCGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-7-1': defaultdict(None,\n",
       "                         {'seq': 'GTGGTCTAGTGGTtAGGATTCAGCGCTCCCACCGCCGCAGCCCGGGTTCGATTCCCGGtCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-8-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTCAGTGGTTTAGTGGTgGAATTCCTGCCTCCCATGCACGAGaTCCGTGTTCAACTCCTGGTTGGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-chr1-102': defaultdict(None,\n",
       "                         {'seq': 'GCGTCAGTGgTTTTAGTGGTgGAATTCCTGCCTCCCATGCACGAGaTCCGTGTTCAACTCCTGGTTGGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-CCC-chr1-16': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGCAGTTCAGTGGTAGAATTCTCGCCTCCCACCCGGGAGaCCTGGATTCCATTTCCGGCAAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCATGGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTTGATTCCCGGCCAGTGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCATAGGTGGTTCAGTGGTAGAATTCTTGCCTGCCACGCAGGAGGCCCAGGTTTGATTCCTGGCCCATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCATGCGGGCGGCCGGGCTTCGATTCCTgGCCAATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-GCC-6-1': defaultdict(None,\n",
       "                         {'seq': 'GCATGGGTGATTCAGTGGTAGAATTTTCACCTGCCATGCAGGAGGTCCAGGTTCATTTCCTGGCCTATGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTATAGTGGTtAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTATAGTGGTgAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTATAGTGGTaAGCATAGCTGCCTTCCAAGCAGTTGaCCCGGGTTCGATTCCCGGCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCC'}),\n",
       "             'Homo_sapiens_tRNA-Gly-TCC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCGTTGGTGGTATAGTGGTgAGCATAGTTGCCTTCCAAGCAGTTGaCCCGGGCTCGATTCCCGCCCAACGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCC'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGTGATCGTATAGTGGTtAGTACTCTGCGTTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTG'}),\n",
       "             'Homo_sapiens_tRNA-His-GTG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCATGATCGTATAGTGGTtAGTACTCTGCGCTGTGGCCGCAGCAaCCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTG'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGCGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAACTTAATTGGTtAGAGCGTGGTGCTAATAATGTCAAGGtTGCGGGTTGGATCCCCGAACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-11-1': defaultdict(None,\n",
       "                         {'seq': 'TGGCTGATGaGCTCAGCTGGTgGGAGCATGGTGTTAATGAGGCTGAGGtCGTGGGTTCAATCCCCACCGGGCTATCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGAACCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGGTGCTAATAACGCTAAGGtCGCGGGTTCGATCCCCGTACTGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-7-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTcAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-8-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTCGGCtAGAGCGTGGTGCTAATAACGCCAAGGtCGCGGGTTCGATCCCCGTACGGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-9-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTGGTTAGTTCAGTTGGTtAGAGCGTGGTGCTAATAACGCCAAGGtCGTGGGTTCGATCCCCATATCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-chr6-37': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCGGTTGCTtAGAGTGTTGTGTTAATTACGCtAAAGGtCGTGGGTTTGATCCTCTTACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-AAT-chr6-83': defaultdict(None,\n",
       "                         {'seq': 'GTTAGCTTAGCTGGTgAGAGCGTGCTGCTAATAACACCAAGGtCGTGGTTTCTATCCCATtACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTaAGAGCGTGGTGCTGATAACACCAAGGtCGCGGGCTCGACTCCCGCACCGGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-chrX-5': defaultdict(None,\n",
       "                         {'seq': 'GGTCGGTTATCTCAGTTGGTgAGCGCGTGGTGCTGATACCGCCATGGtCGCGGGCTTGATCCCCGCACCGGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-GAT-chrX-9': defaultdict(None,\n",
       "                         {'seq': 'GGTCGGTTATCTCAGTTGGTgAGCGCGTGGTGCTGATACCGCCATGGtCGCAGGCTTGATCCCCGCACCGGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGATCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCGAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAT'}),\n",
       "             'Homo_sapiens_tRNA-Ile-TAT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCTCCAGTGGCGCAATCGGTtAGCGCGCGGTACTTATAATGCCGAGGtTGTGAGTTCAAGCCTCACCTGGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAT'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCAAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGTGGTctAAGACGCTGGATTAAGGCTCCAGTCTCTTCGGGGGCGTGGGTTTGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGCCAGTGGCTCAATGGAtAATGCGTCTGACTAAGAATCAGAAGAtTCCAGCCTTGACTCCTGGCTGGCTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGGGTGGCCGAGCGGTctAAGGCACTGTATTAAGACTCCAGTCTCTTCAGAGGCATGGGTTTGAATCCCACTGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-AAG-7-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAGCGGTctAAGGCTCCGGATTAAGGCGCCGGTGTCTTCGGAGgCATGGGTTCGAATTCCACCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGTATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGCATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGTGGTctAAGGCGCCAGACTCAAGTTCTGGTCTCCGGATGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-6-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGCAGTcttAAGGCGCTGCGTTCAAATCGCACCCTCCGCTGGAGGCGTGGGTTCGAATCCCACTTTTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-7-1': defaultdict(None,\n",
       "                         {'seq': 'CATAGGGTAGTGGCtAAGAACCTAAACTCTAAATTTAGATGtCCTGAGTTCAAATCCCAGCTGTATGCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-chr1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCAGTGTcAGGTAGGGCATGGATTTGGGAGCCAGAAAgCCCAGGTTCAAATCCTGACTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAA-chr14-11': defaultdict(None,\n",
       "                         {'seq': 'GGCAGTGTAGCCAAGAGGCcaggAAGGTGCACCCTGGAGCCCCACTGCCTAGGTTCAAATCCCAGCTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGATGGCCGAGCGGTctAAGGCGCTGCGTTCAGGTCGCAGTCTCCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-chr9-3': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATGGTGTAATGGTgAGCACTCTGGACTCAGAATCCAGAAGtAGTgCTGGAACAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-CAG-chr9-7': defaultdict(None,\n",
       "                         {'seq': 'GTCAGGGTGGCTGAGCAGTctGAGGGGCTGCGTTCAGGTCGCAGTCTGCCCTGGAGGCGTGGGTTCGAATCCCACTCCTGAAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-1-1': defaultdict(None,\n",
       "                         {'seq': 'ACCAGGATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGACATATGTCCGCGTGGGTTCGAACCCCACTCCTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-2-1': defaultdict(None,\n",
       "                         {'seq': 'ACCGGGATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGGCTGGTGCCCGCGTGGGTTCGAACCCCACTCTCGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-3-1': defaultdict(None,\n",
       "                         {'seq': 'ACCAGAATGGCCGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGATTCATATCCGCGTGGGTTCGAACCCCACTTCTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-4-1': defaultdict(None,\n",
       "                         {'seq': 'ACCGGGATGGCTGAGTGGTtAAGGCGTTGGACTTAAGATCCAATGGACAGGTGTCCGCGTGGGTTCGAGCCCCACTCCCGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-5-1': defaultdict(None,\n",
       "                         {'seq': 'ACTCATTTGGCTGAGTGGTtAAGGCATTGGACTTAAGATCCAATGGAGTAGTGGCTGTGTGGGTTTAAACCCCACTACTGGTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-chr1-9': defaultdict(None,\n",
       "                         {'seq': 'GAGAAAGTcATCGTAGTTACGAAGTTGGCTTAAACCCAGTTTtGGGAGGTTCAATTCCTTCCTTTCTCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAA-chr11-12': defaultdict(None,\n",
       "                         {'seq': 'ACCAGGATGGCCAAGTAGTTaAAGGCACTGGACTTAAGAGCCAATGGACATATGTCTGTGTGGGTTTGAACCCCACTCCTGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGTGTGGCCGAGCGGTctAAGGCGCTGGATTTAGGCTCCAGTCTCTTCGGGGGCGTGGGTTCGAATCCCACCACTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGCGGTctAAGGCGCTGGATTTAGGCTCCAGTCTCTTCGGAGGCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGCGTGGCCGAGTGGTctAAGGCGCTGGATTTAGGCTCCAGTCATTTCGATGgCGTGGGTTCGAATCCCACCGCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAG'}),\n",
       "             'Homo_sapiens_tRNA-Leu-TAG-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAGTGTGGTTGAATGGTctAAGGCACTGAATTTAGGCTCCAGTCTCTTTGGGGaCGTGGGTTTAAATCCCACTGCTGCAACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAG'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-10-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCAGCTAGCTCAGTCGGTAGAGCATAAGACTCTTAATCTCAGGGtTGTGGATTCGTGCCCCATGCTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-11-1': defaultdict(None,\n",
       "                         {'seq': 'CTGCAGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCATGGGTTCGTGCCCCATGTTGGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-12-1': defaultdict(None,\n",
       "                         {'seq': 'CCAGCATGTCTCAGTCGGTATAGTGTGAGACTCTTAATCTCAGGGtCGTGGGTTCAAGCCCCACATTGGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-13-1': defaultdict(None,\n",
       "                         {'seq': 'GTCTAGCTAGATCAGTTGGTAGAGCATAAGACTCTTAATCTCAGGGtCATGGGTTTGAGCCCTACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-14-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCAGCTAGCTCAGCCGGTAGAGCACAAGACTCTTAATCTCAGGGtCGTGGGTTTGAGCCCTGTGTTGAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-15-1': defaultdict(None,\n",
       "                         {'seq': 'CCGAATAGCTTAGTTGATgAAGCGTGAGACTCTTAATCTCAGGGtAGTGGGTTCAAGCCCCACATTGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-16-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGCTACCTCAGTTGGTAGAGCATGGGACTCTTAATCCCAGAGtcAGTGGGTTCAAGCCTCACATTGAGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGAGACCCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGCTAGCTCAGTCGATAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCGCACGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGCTAGCTCAGTCGGTAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTTGAGCCCCATGTTGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-7-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCAGCTAGCTCAGTCGGTAGAGCATGAGACTCTTAATCTCAGGGtCATGGGTTTGAGCCCCACGTTTGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-8-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGCTAGCTCAGTCGGCAAAGCATGAGACTCTTAATCTCAGGGtCGTGGGCTCGAGCTCCATGTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-9-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGACTACCTCAGTCGGTgGAGCATGGGACTCTTCATCCCAGGGtTGTGGGTTCGAGCCCCACATTGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr15-5': defaultdict(None,\n",
       "                         {'seq': 'GCACAGTCAGTcagtAGAGCATGGGACTCTTAATCCCAGGGtCGTGGGTTCAAGCCCCACCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr16-1': defaultdict(None,\n",
       "                         {'seq': 'CCCCGGCTGGCTCAGTCAGTAGATCATGAGACTCTTAATCTCAGGGtCGTGGGTTCACGCCCCACACTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-CTT-chr7-30': defaultdict(None,\n",
       "                         {'seq': 'GCGCTAGTCAGTAGAGCATGAGACTCTTAATCTCAGGGtCGTGGGTTCGAGCCCCACATCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGATAGCTCAGTTGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-10-1': defaultdict(None,\n",
       "                         {'seq': 'GCCAGGATAGTTCAGGTGGTAGAGCATCAGACTTTTAACCTGAGGGtTCAGGGTTCAAGTCTCTGTTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-11-1': defaultdict(None,\n",
       "                         {'seq': 'ACCCAGATAGCTCAGTCAGTAGAGCATCAGACTTTTAATCTGAGGGtCCAAGGTTCATGTCCCTTTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-12-1': defaultdict(None,\n",
       "                         {'seq': 'ACCTGGGTAGCTTAGTTGGTAGAGCATTGGACTTTTAATTTGAGGGcCCAGGTTTCAAGTCCCTGTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-13-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGGTAGCTCAGTCGGTAGAGCTaTCAGACTTTTAGCCTGAGGAtTCAGGGTTCAATCCCTTGCTGGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-14-1': defaultdict(None,\n",
       "                         {'seq': 'GATAGCTCAGTTGATAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCATGTCCCTGTTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-15-1': defaultdict(None,\n",
       "                         {'seq': 'GTTGGGGTAACTCAGTTGGTAGAGTAGCAGACTTTTCATCTGAGGGtCCAGGGTTTAAGTCCATGTCCAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGATAGCTCAGTTGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGATAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCGGGGTTCAAGTCCCTGTTCGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGGTAGCTCAGTCGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCCTGTCCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-7-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGATAGCTCAGTTGGTAGAACATCAGACTTTTAATCTGACGGtGCAGGGTTCAAGTCCCTGTTCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-8-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGAGAGCTCAGTGGGTAGAGCATCAGACTTTTAATCTGAGGGtCCAGGGTTCAAGTCCTCGTTCGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-9-1': defaultdict(None,\n",
       "                         {'seq': 'ACCTGGGTAGCTCAGTAGGTAGAACATCAGACTTTTAATCTGAGGGtCTAGGGTTCAAGTCCCTGTCCAGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Lys-TTT-chr3-2': defaultdict(None,\n",
       "                         {'seq': 'GCCTGGATAGCTCCTTCGGTAGAGCATCATcagACTTTTAATGTGAGGGtCCAGGGTTCAAGTTCCTGTTTGGGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTCGTTAGCGCAGTAGGTAGCGCGTCAGTCTCATAATCTGAAGGtCGTGAGTTCGATCCTCACACGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCTCTTAGCGCAGTGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTCCTTAGCGCAGTAGGCAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAACCTCAGAGGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCTCTTAGCGCAGCGGGcAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCGAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCTCTTAGCGCAGCTGGCAGCGCGTCAGTCTCATAATCTGAAGGtCCTGAGTTCAAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GCCTCGTTAGCGCAGTAGGCAGCGCGTCAGTCTCATAATCTGAAGGtCGTGAGTTCGAGCCTCACACGGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Met-CAT-7-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCTCTTAGTGCAGCTGGCAGCGCGTCAGTTTCATAATCTGAAAGtCCTGAGTTCAAGCCTCAGAGAGGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTGTCTAGTGGCtAGGATTCGGTGCTGAAAGCGTCACGgCCCGGGTTCGATTCCCGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-11-1': defaultdict(None,\n",
       "                         {'seq': 'GCTGAGATAGCTCGGTTGGGAGGGCATCAGCCTGAAGATCTAAAGGaGACTGGTTCAATTCTGGGTTTTGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAGATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACCGAAGATCTtAAAGGtCCCTGGTTCAATCCCGGGTTTCGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-5-1': defaultdict(None,\n",
       "                         {'seq': 'GCCAAAATTGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGtCCCTGGTTCGATCCCGGGTTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-6-1': defaultdict(None,\n",
       "                         {'seq': 'GCTGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTTAAAGttCCCTGGTTCAACCCTGGGTTTCAGCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-7-1': defaultdict(None,\n",
       "                         {'seq': 'GCCGAAATAGCTCAATTGGGAGAGTGTTAGACTGAAGATCTTCTGCAGGtCTCTGGTTCAATTCCGGGTTTCGACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Phe-GAA-8-1': defaultdict(None,\n",
       "                         {'seq': 'GCCAAAATAGCTCAGCTGGGAGAGCATTAGACTGAAGATCTAAAGGtCTCTGGTTTGATCCTGGGTTTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGATGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTAGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-AGG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTgTGGTTCTCGCTTAGGGCGGGAGACCCAAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-CGG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTCGGGTGTGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-GGG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCAGTGTGGCATAGTGGTtagaAATGTGCGCTCTGGGGCTGCTGAtCCCAGGCTCAAACCCTGGCGCTGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGTGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGGTTTGGGTCCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGG'}),\n",
       "             'Homo_sapiens_tRNA-Pro-TGG-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCGTTGGTCTAGGGGTATGATTCTCGCTTTGGGTGCGAGAGGtCCCGGGTTCAAATCCCGGACGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGG'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGATGATCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGCGACAGAGTGGTTCAATTCCACCTTTCGGGCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCA'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCTCGGATGATCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGTGACAGAGTGGTTCAATTCCACCTTTGTAGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCA'}),\n",
       "             'Homo_sapiens_tRNA-SeC-TCA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GACCGTGTGGCCTTAATGGAtAAGGTGTCTGACTTCAGATCAGAAGAtTGAGGGTTTGAGTCCCTTTGTGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-ACT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCGGTTAGCTCAGTTGGTtAGAGCGTGCTGCTACTAATGCCAGGGtCGAGGTTTCGATCCCCGTACGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ACT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTAGAAATCCATTGGGGTTTCCCCACGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGTGATGGACTAGAAACCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGGTGTATGGCTCAGGGGTAGAGAATTTGACTAGAGATCAAGAGGtCCCTGGTTCAAATCCAGGTGCCCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-6-1': defaultdict(None,\n",
       "                         {'seq': 'AGTTGTAGCTGAGTGGTtAAGGCAACGAGCTAGAAATTCGTTGGTTTCTCTCTgTGCAGGTTTGAATCCTGCTAATTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-AGA-chr11-8': defaultdict(None,\n",
       "                         {'seq': 'CAAGAAATTCATAGAGGTTATGGGATTGGCTAGAAACCAGTTTcAGGAGGTTCGATTCCTTCCTTTTTGGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCTGTGATGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCTGTGATGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTCTCCCCGCGCAGGTTCAAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GCTGTGATGGCCGAGTGGTtAAGGTGTTGGACTCGAAATCCAATGGGGGTTCCCCGCGCAGGTTCAAATCCTGCTCACAGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-CGA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTCACGGTGGCCGAGTGGTtAAGGCGTTGGACTCGAAATCCAATGGGGTTTCCCCGCACAGGTTCGAATCCTGTTCGTGACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCACCTTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTTTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GACGAGGTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACACGTGGGTTCGAATCCCATCCTCGTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-GCT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GAGGCCTGGCCGAGTGGTtAAGGCGATGGACTGCTAATCCATTGTGCTCTGCACGCGTGGGTTCGAATCCCATCCTCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCAGCGATGGCCGAGTGGTtAAGGCGTTGGACTTGAAATCCAATGGGGTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTCTCCCCGCGCAGGTTCGAATCCTGCCGACTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGA'}),\n",
       "             'Homo_sapiens_tRNA-Ser-TGA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTAGTCGTGGCCGAGTGGTtAAGGCGATGGACTTGAAATCCATTGGGGTTTCCCCGCGCAGGTTCGAATCCTGTCGGCTACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGA'}),\n",
       "             'Homo_sapiens_tRNA-Sup-CTA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCAGGGTAGGGTAGAGGTtAAAACCATGGATTCTAGAGCCAGATGGGTTCAAATCCCGGCTCTGCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTA'}),\n",
       "             'Homo_sapiens_tRNA-Sup-CTA-chr5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCAGTGTGATGTGGAGGAagggAGCATGTACTCTACTTCAGAGGGgCTTGGGTTCAAATCCCAGCTCTGCCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CTA'}),\n",
       "             'Homo_sapiens_tRNA-Sup-TTA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GCCCGGATAGTTCAGTTGGTAGAGCATCAGACTTAATCAGAGGGtCCAGGGTTCAAGTCCCTGTTTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTA'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCGCCGTGGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCGTAGCTTAGTTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGACTCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTTCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGAGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCGCCGTGGCTTAGCTGGTtAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCCTGTGGCTTAGCTGGTcAAAGCGCCTGTCTAGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-AGT-7-1': defaultdict(None,\n",
       "                         {'seq': 'AGCACCATGGCTTAGCTGGTtAAAGCACCTGTCTAGTAAACAGGAGAtCCTGAGtTTCAATTcCAATGGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTCGTAAACAGGAGAtCCTGGGTTCGACTCCCAGTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCGCGGTGGCCAAGTGGTAAGGCGTCGGTCTCGTAAACCGAAGAtCACGGGTTCGAACCCCGTCCGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTGTGGCTTAGTTGGCtAAAGCGCCTGTCTCGTAAACAGGAGAtCCTGGGTTCGAATCCCAGCGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGCGCGGTGGCCAAGTGGTAAGGCGTCGGTCTCGTAAACCGAAGAtCGCGGGTTCGAACCCCGTCCGTGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCCTGTAGCTCAGCGGTtGGAGCGCTGGTCTCGTAAACCtAGGGGtCGTGAGTTCAAATCTCACCAGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCAGAGTGGTGCAGCGGAAGCGTGCTGGGCCCGTAACCCAGAGGtCAATGGATCGAAGCCATCCTTGGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-CGT-chr1-118': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTCGTAAGCAGGAGGATCGCTTGAGCCCCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCTATGGCTTAGTTGGTtAAAGCGCCTGTCTTGTAAACAGGAGAtCCTGGGTTCGAATCCCAGTAGAGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCATAGCTCAGTGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCGATCCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCATAGCTCAGGGGTtAGAGCGCTGGTCTTGTAAACCAGGGGtCGCGAGTTCAATTCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-4-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-5-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCCTATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Thr-TGT-6-1': defaultdict(None,\n",
       "                         {'seq': 'GGCTCCATAGCTCAGGGGTtAGAGCACTGGTCTTGTAAACCAGGGtCGCGAGTTCAAATCTCGCTGGGGCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGCCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-2-1': defaultdict(None,\n",
       "                         {'seq': 'GACCTCGTGGCGCAATGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAGTCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-3-1': defaultdict(None,\n",
       "                         {'seq': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-4-1': defaultdict(None,\n",
       "                         {'seq': 'GACCTCGTGGCGCAACGGTAGCGCGTCTGACTCCAGATCAGAAGGcTGCGTGTTCGAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-5-1': defaultdict(None,\n",
       "                         {'seq': 'GACCTCGTGGCGCAACGGCAGCGCGTCTGACTCCAGATCAGAAGGtTGCGTGTTCAAATCACGTCGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Trp-CCA-chr7-31': defaultdict(None,\n",
       "                         {'seq': 'GGCCTCATGGTGCAACAGTAGTGTGTCTGACTCCAGATCAGAAGGtTGTATGTTCAAATCACGTAGGGGTCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CCA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr1-11': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCTTTAGACTATAAATCTAAAGAGAGATTGAGCTCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr1-52': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGACTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr11-9': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-ATA-chr9-2': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCATTAGACTATAAATCTAAAGaCAGAGGTCAAGGCCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'ATA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-1-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-10-1': defaultdict(None,\n",
       "                         {'seq': 'GGGGGTATAGCTCAGGGCtAGAGCTtTTTGACTGTAGAGCAAGAGGtCCCTGGTTCAAATCCAGGTTCTCCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-11-1': defaultdict(None,\n",
       "                         {'seq': 'TATAGCTCAGTGGTAGAGCATTTAACTGTAGATCAAGAGGtCCCTGGATCAACTCTGGGTGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-12-1': defaultdict(None,\n",
       "                         {'seq': 'GTCAGTGTTGCACAACGGTtaAGTGAAGAGGCTGTAAACCCAGACTGGATGGGTTCAATTCCCATCTCTGCCGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-2-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-3-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGGAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-4-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCAGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-5-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-6-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGGAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-7-1': defaultdict(None,\n",
       "                         {'seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCAATTCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-8-1': defaultdict(None,\n",
       "                         {'seq': 'CTTTCGATAGCTCAGTTGGTAGAGCGGAGGACTGTAGATCCTTAGGtCGCTGGTTCGAATCCGGCTCGAAGGACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-9-1': defaultdict(None,\n",
       "                         {'seq': 'TCTTCAATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATTCTTAGGTGCTGGTTTGATTCCGACTTGGAGAGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr10-3': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGCAAGCATTGGACTGTAAATCTAAAGaCAGATGTTGAGCCATCTTTTTAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr14-8': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGaCAGGGGCTAAGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr2-12': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGCAAGCATTAGACTGTAAATCTAAAGaCAGAGGTTAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr21-2': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGaCAGGGGTTAAGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCATTAGACTGTAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTTCCTCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-2': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGCAAGCATTAGACTGTAAATCTGAAAaCAGAGGTCAAAGgTCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr7-6': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCATTAGACTGTAAATCTAAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr8-7': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGACTGAATAAGCCTTAGACTGTAAATCTGAAGaCAGAGGTCAAGGCCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr9-10': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAGCATTGGACTGTAAATCTAAAGaCAGAGGTCAAGACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Tyr-GTA-chr9-4': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTAAAGCATTAGACTGTAAATCTAAGGaCAGAGGCTAAACCTCTTTTTACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTcATCACGTTCGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCCTGGATCAAAACCAGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTAACACGCGAAAGGtCCGCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTTGCCTAACACGCGAAAGGtCCCCGGTTCGAAACCGGGCAGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-7-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCATAGTGTACTGGTtATCACATTCACCTAACACGCGAAAGGtCCTTGGTTTGAAACCAGGCAGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-AAC-chr6-84': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCATAGCGTAGTAATtATCAAGTTTGCCTAACACGCGAAATGtCCCCGGTTCAAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'AAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-10-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCTGTGGTGTAGTGGTtATCATGTTCGCCTCACACGAGAAAAGtCCCTGATTCGAGACTGGGTGGGAACGCCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-11-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCTGTGGTGTAGTGGTtATTATGTTCGCTTCACATATGAAAGGtCTCTGGTTCGAGACTGCGTGGGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GCTTCTGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACCGGGCAGAAGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGCGGTtATCACATTCGCCTCACACGCGAAAGGtCCCCGGTTCGATCCCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTCGAAACTGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-5-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATCACGTTCGCCTCACACGCGtAAAGGtCCCCGGTTCGAAACCGGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-6-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGGAGTGGTtATCACGTTCGCCTCACACGCGAAAGGtCCCCGGTTTGAAACCAGGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-7-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTAGTGTAGTGGTtATTATGTTCGCCTCACACGCGAAAAGtCCCCGGTTCGAAATCAGGCGGGAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-8-1': defaultdict(None,\n",
       "                         {'seq': 'GCACTGGTGGTTCAGTGGTAGAATTCTCGCCTCACACGCGGGACACCCGGGTTCAATTCCCGGTCAAGGCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-CAC-9-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCTGTAGTATGGTGGTtATCACGTTAGTCTCACACGTGAAAGGtCCCTGGTTCGAAACCAGGTGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATAGTGTAGTGGTtATCACGTCTGCTTTACACGCAGAAGGtCCTGGGTTCGAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-2-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATAGTGTAGCGGTtATCACGTCTGCTTTACACGCAGAAGGtCCTGGGTTCGAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-3-1': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATAGTGTAGTGGTtATCACATCTGCTTTACACGCAGAAGGtCCTGGGTTCAAGCCCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-4-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTCCGTGGTGTAGTGGTtATCACATTCGCCTTACACGCGAAAGGtCCTCGGGTCGAAACCGAGCGGAAACACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-chr17-24': defaultdict(None,\n",
       "                         {'seq': 'GATTCTATAGTGTAGCAGTtATCACATCTGCTTTACATGCAGAAGGtCCTAGGTTTGAGCCGCGGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-Val-TAC-chr2-6': defaultdict(None,\n",
       "                         {'seq': 'GGTTCCATAGTGTAACAGTtATCATGTCTGCTTTACACACTGAAGGtCCTGGGTTCAAACTCCAGTGGAACCACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-2-1': defaultdict(None,\n",
       "                         {'seq': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGtCGATGGATCTAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_tRNA-iMet-CAT-3-1': defaultdict(None,\n",
       "                         {'seq': 'AGCAGAGTGGTGCAGTGGAAGCATACctaTGGGCCCATAACCCAGAGGtTGATGGATGGAAACCATCCTCTGCTACCA',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'type': 'cyto',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ala-TGC-1-1': defaultdict(None,\n",
       "                         {'seq': 'AAGGGCTTAGCTTAATTAAAGTGGCTGATTTGCGTTCAGTTGATGCAGAGTGGGGTTTTGCAGTCCTTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGC'}),\n",
       "             'Homo_sapiens_mito_tRNA-Arg-TCG-1-1': defaultdict(None,\n",
       "                         {'seq': 'TGGTATATAGTTTAAACAAAACGAATGATTTCGACTCATTAAATTATGATAATCATATTTACCAACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCG'}),\n",
       "             'Homo_sapiens_mito_tRNA-Asn-GTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'TAGATTGAAGCCAGTTGATTAGGGTGCTTAGCTGTTAACTAAGTGTTTGTGGGTTTAAGTCCCATTGGTCTAGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Asp-GTC-1-1': defaultdict(None,\n",
       "                         {'seq': 'AAGGTATTAGAAAAACCATTTCATAACTTTGTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTC'}),\n",
       "             'Homo_sapiens_mito_tRNA-Cys-GCA-1-1': defaultdict(None,\n",
       "                         {'seq': 'AGCTCCGAGGTGATTTTCATATTGAATTGCAAATTCGAAGAAGCAGCTTCAAACCTGCCGGGGCTTCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Gln-TTG-1-1': defaultdict(None,\n",
       "                         {'seq': 'TAGGATGGGGTGTGATAGGTGGCACGGAGAATTTTGGATTCTCAGGGATGGGTTCGATTCTCATAGTCCTAGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTG'}),\n",
       "             'Homo_sapiens_mito_tRNA-Glu-TTC-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTTCTTGTAGTTGAAATACAACGATGGTTTTTCATATCATTGGTCGTGGTTGTAGTCCGTGCGAGAATACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTC'}),\n",
       "             'Homo_sapiens_mito_tRNA-Gly-TCC-1-1': defaultdict(None,\n",
       "                         {'seq': 'ACTCTTTTAGTATAAATAGTACCGTTAACTTCCAATTAACTAGTTTTGACAACATTCAAAAAAGAGTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCC'}),\n",
       "             'Homo_sapiens_mito_tRNA-His-GTG-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTAAATATAGTTTAACCAAAACATCAGATTGTGAATCTGACAACAGAGGCTTACGACCCCTTATTTACCCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTG'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ile-GAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Leu-TAG-1-1': defaultdict(None,\n",
       "                         {'seq': 'ACTTTTAAAGGATAACAGCTATCCATTGGTCTTAGGCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAG'}),\n",
       "             'Homo_sapiens_mito_tRNA-Leu-TAA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTTAAGATGGCAGAGCCCGGTAATCGCATAAAACTTAAAACTTTACAGTCAGAGGTTCAATTCCTCTTCTTAACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Lys-TTT-1-1': defaultdict(None,\n",
       "                         {'seq': 'CACTGTAAAGCTAACTTAGCATTAACCTTTTAAGTTAAAGATTAAGAGAACCAACACCTCTTTACAGTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TTT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Met-CAT-1-1': defaultdict(None,\n",
       "                         {'seq': 'AGTAAGGTCAGCTAAATAAGCTATCGGGCCCATACCCCGAAAATGTTGGTTATACCCTTCCCGTACTACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'CAT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Phe-GAA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTTTATGTAGCTTACCTCCTCAAAGCAATACACTGAAAATGTTTAGACGGGCTCACATCACCCCATAAACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GAA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Pro-TGG-1-1': defaultdict(None,\n",
       "                         {'seq': 'CAGAGAATAGTTTAAATTAGAATCTTAGCTTTGGGTGCTAATGGTGGAGTTAAAGACTTTTTCTCTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGG'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ser-GCT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GAGAAAGCTCACAAGAACTGCTAACTCATGCCCCCATGTCTAACAACATGGCTTTCTCACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GCT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Ser-TGA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GAAAAAGTCATGGAGGCCATGGGGTTGGCTTGAAACCAGCTTTGGGGGGTTCGATTCCTTCCTTTTTTGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Thr-TGT-1-1': defaultdict(None,\n",
       "                         {'seq': 'GTCCTTGTAGTATAAACTAATACACCAGTCTTGTAAACCGGAGATGAAAACCTTTTTCCAAGGACACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TGT'}),\n",
       "             'Homo_sapiens_mito_tRNA-Trp-TCA-1-1': defaultdict(None,\n",
       "                         {'seq': 'AGAAATTTAGGTTAAATACAGACCAAGAGCCTTCAAAGCCCTCAGTAAGTTGCAATACTTAATTTCTGCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TCA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Tyr-GTA-1-1': defaultdict(None,\n",
       "                         {'seq': 'GGTAAAATGGCTGAGTGAAGCATTGGACTGTAAATCTAAAGACAGGGGTTAGGCCTCTTTTTACCCCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'GTA'}),\n",
       "             'Homo_sapiens_mito_tRNA-Val-TAC-1-1': defaultdict(None,\n",
       "                         {'seq': 'CAGAGTGTAGCTTAACACAAAGCACCCAACTTACACTTAGGAGATTTCAACTTAACTTGACCGCTCTGACCA',\n",
       "                          'type': 'mito',\n",
       "                          'species': 'Homo sapiens',\n",
       "                          'modified': [],\n",
       "                          'InosinePos': [],\n",
       "                          'anticodon': 'TAC'})})"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trna_obj.tRNA_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'hg19_test'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_dict = trna_obj.tRNA_dict\n",
    "mod_dict = trna_obj.modomics_dict\n",
    "# species_set = species_set\n",
    "tmp_dir = temp_dir\n",
    "\n",
    "# For later use in clustering\n",
    "# anticodon_list = list({tRNA['anticodon'] for tRNA in tRNA_dict.values()})\n",
    "\n",
    "seq_records = defaultdict()\n",
    "tRNAbed_fh = open(args.outdir + args.name + \"_maturetRNA.bed\",\"w\")\n",
    "for tRNA_id, tRNA in tRNA_dict.items():\n",
    "    # Initialize list of modified sites for each tRNA\n",
    "    tRNA['modified_idx'] = list()\n",
    "    tRNA['insosine_idx'] = list()\n",
    "    with open(tmp_dir + tRNA_id + \".fa\", \"w\") as fh_tRNA:\n",
    "        print('>{}\\n{}'.format(tRNA_id, tRNA['seq']), file=fh_tRNA)\n",
    "    anticodon = re.search('.*tR(NA|X)-.*?-(.*?)-', tRNA_id).group(2)\n",
    "    tRNA['anticodon'] = anticodon\n",
    "    # Find matches to modomics where anticodons match\n",
    "    # and types are the same \n",
    "    # (here regex is used to match anticodons with \".\" in modomics\n",
    "    # to all possible matching sequences from input tRNAs)\n",
    "    match = dict()\n",
    "    for key, mod in mod_dict.items():\n",
    "        if re.match(\"^\" + mod['anticodon'] + \"+$\", anticodon) and tRNA['type'] == mod['type']:\n",
    "            match[key] = mod\n",
    "    if len(match) > 0:\n",
    "        with open(tmp_dir + \"modomics_match.fasta\", \"w\") as match_fh:\n",
    "            for key, mod in match.items():  \n",
    "                print('>{}\\n{}'.format(key, mod['unmod_seq']), file=match_fh)\n",
    "\n",
    "        # Call BLAST\n",
    "        blastn_cline = NcbiblastnCommandline(query=fh_tRNA.name, subject=match_fh.name, task='blastn-short', out=tmp_dir + \"blast_temp.xml\", outfmt=5)\n",
    "        blastn_cline()\n",
    "\n",
    "        # Parse XML result and store hit with highest bitscore    \n",
    "        blast_rec = NCBIXML.read(open(tmp_dir + \"blast_temp.xml\", \"r\"))\n",
    "        maxbit = 0\n",
    "        tophit = ''\n",
    "        for aln in blast_rec.alignments:\n",
    "            for hsp in aln.hsps:\n",
    "                if (hsp.bits > maxbit) and (hsp.align_length / aln.length == 1) and (hsp.identities / aln.length == 1):\n",
    "                    maxbit = hsp.bits\n",
    "                    tophit = aln.title.split(' ')[0]\n",
    "\n",
    "        # Return list of all modified positions for the match as long as there is only 1, add to tRNA_dict\n",
    "        if tophit:\n",
    "            tRNA['modified_idx'].append(match[tophit]['modified_idx'])\n",
    "            tRNA['insosine_idx'].append(match[tophit]['insosine_idx'])\n",
    "\n",
    "    # Build seqrecord list for writing\n",
    "    seq_records[str(tRNA_id)] = SeqRecord(Seq(tRNA['seq'].upper()), id = str(tRNA_id))\n",
    "    print('{}\\t0\\t{}\\t{}\\t1000\\t+\\n'.format(tRNA_id, str(len(tRNA['seq'])), tRNA_id), file=tRNAbed_fh)\n",
    "tRNAbed_fh.close()\n",
    "\n",
    "match_count = sum(len(tRNA['modified_idx'])+len(tRNA['insosine_idx']) for tRNA in tRNA_dict.values())\n",
    "log.info(\"{} total tRNA gene sequences (undetermined and nmt sequences excluded)\".format(len(tRNA_dict)))\n",
    "log.info(\"{} sequences with a match to Modomics dataset\".format(match_count))\n",
    "\n",
    "with open(str(out_dir + args.name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "    SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "import re, copy, sys, os, shutil, subprocess, logging, glob\n",
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "from mimseq.ssAlign import aligntRNA, extraCCA, tRNAclassifier, tRNAclassifier_nogaps, getAnticodon, clusterAnticodon\n",
    "\n",
    "def modsToSNPIndex(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, experiment_name, out_dir, double_cca, snp_tolerance = False, cluster = False, cluster_id = 0.95, posttrans_mod_off = False, pretrnas = False, local_mod = False):\n",
    "# Builds SNP index needed for GSNAP based on modificaiton data for each tRNA and clusters tRNAs\n",
    "\n",
    "    nomatch_count = 0\n",
    "    match_count = 0\n",
    "    total_snps = 0\n",
    "    total_inosines = 0\n",
    "    snp_records = list()\n",
    "    seq_records = defaultdict()\n",
    "    anticodon_list = list()\n",
    "    tRNAbed = open(out_dir + experiment_name + \"_maturetRNA.bed\",\"w\")\n",
    "\n",
    "    ''' Contained in TRNA class\n",
    "    # generate modomics_dict and tRNA_dict\n",
    "    tRNA_dict, modomics_dict, species = tRNAparser(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, posttrans_mod_off, double_cca, pretrnas, local_mod)\n",
    "    temp_dir = out_dir + \"/tmp/\"\n",
    "\n",
    "    try:\n",
    "        os.mkdir(temp_dir)\n",
    "    except FileExistsError:\n",
    "        log.warning(\"Temp folder present - previous run interrupted? Overwriting old temp files...\\n\")\n",
    "\n",
    "\n",
    "    ###################################################\n",
    "    ## Main code for matching and SNP index building ##\n",
    "    ###################################################\n",
    "\n",
    "    # remove spurious CCA seen in some sequences (i.e. genomically encoded or artifact from gtRNAdb or tRNAScan-SE prediction)\n",
    "    # to do this, remove previously added CCA (intronRemover()) that falls out of canonical structure using ssAlign module\n",
    "\n",
    "    with open(out_dir + 'tmpRNAseqs.fa', 'w') as fh_out:\n",
    "        for header, tRNA in tRNA_dict.items():\n",
    "            print('>{}\\n{}'.format(header, tRNA['seq']), file=fh_out)\n",
    "\n",
    "    stkname = align_tRNA(fh_out.name, outdir, repo_path)\n",
    "    wrong_seqs = wrong_cca(stkname)\n",
    "    # Clip the end of the wrong CCA sequences\n",
    "    for header in wrong_seqs:\n",
    "        tRNA_dict[header]['seq'] = tRNA_dict[header]['seq'][:-3]\n",
    "\n",
    "    os.remove(fh_out.name)\n",
    "    '''\n",
    "\n",
    "    \n",
    "    # match each sequence in tRNA_dict to value in modomics_dict using BLAST\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "    log.info(\"\\n+------------------------+ \\\n",
    "        \\n| Beginning SNP indexing |\\\n",
    "        \\n+------------------------+\")    \n",
    "\n",
    "\n",
    "    for seq in tRNA_dict:\n",
    "        # Initialise list of modified sites for each tRNA\n",
    "        tRNA_dict[seq]['modified'] = []\n",
    "        tRNA_dict[seq]['InosinePos'] = []\n",
    "        temp_tRNAFasta = open(temp_dir + seq + \".fa\",\"w\")\n",
    "        temp_tRNAFasta.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "        temp_tRNAFasta.close()\n",
    "        anticodon = re.search('.*tR(NA|X)-.*?-(.*?)-', seq).group(2)\n",
    "        tRNA_dict[seq]['anticodon'] = anticodon\n",
    "        if not anticodon in anticodon_list:\n",
    "            anticodon_list.append(anticodon)\n",
    "        # find initial possible matches to modomics where anticodons match and types are the same (here regex is used to match anticodons with \".\" in modomics to all possible matching sequences from input tRNAs)\n",
    "        match = {k:v for k,v in modomics_dict.items() if re.match(\"^\" + v['anticodon'] + \"+$\", anticodon) and tRNA_dict[seq]['type'] == v['type']}\n",
    "        if len(match) >= 1:\n",
    "            temp_matchFasta = open(temp_dir + \"modomicsMatch.fasta\",\"w\")\n",
    "            for i in match:    \n",
    "                temp_matchFasta.write(\">\" + i + \"\\n\" + match[i]['unmod_sequence'] + \"\\n\")\n",
    "            temp_matchFasta.close()\n",
    "\n",
    "            #blast\n",
    "            blastn_cline = NcbiblastnCommandline(query = temp_tRNAFasta.name, subject = temp_matchFasta.name, task = 'blastn-short', out = temp_dir + \"blast_temp.xml\", outfmt = 5)\n",
    "            blastn_cline()\n",
    "\n",
    "            #parse XML result and store hit with highest bitscore    \n",
    "            blast_record = NCBIXML.read(open(temp_dir + \"blast_temp.xml\",\"r\"))\n",
    "            maxbit = 0\n",
    "            tophit = ''\n",
    "            for alignment in blast_record.alignments:\n",
    "                for hsp in alignment.hsps:\n",
    "                    if (hsp.bits > maxbit) and (hsp.align_length / alignment.length == 1) and (hsp.identities / alignment.length == 1):\n",
    "                        maxbit = hsp.bits\n",
    "                        tophit = alignment.title.split(' ')[0]\n",
    "            \n",
    "            # return list of all modified positions for the match as long as there is only 1, add to tRNA_dict\n",
    "            if tophit:\n",
    "                match_count += 1\n",
    "                tRNA_dict[seq]['modified'] = match[tophit]['modified']\n",
    "                tRNA_dict[seq]['InosinePos'] = match[tophit]['InosinePos']\n",
    "            elif len(tophit) == 0:\n",
    "                nomatch_count += 1\n",
    "        if len(match) == 0:\n",
    "            nomatch_count += 1\n",
    "\n",
    "        # Build seqrecord list for writing\n",
    "        seq_records[str(seq)] = SeqRecord(Seq(tRNA_dict[seq]['sequence'].upper()), id = str(seq))\n",
    "\n",
    "        tRNAbed.write(seq + \"\\t0\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t\" + seq + \"\\t1000\\t+\\n\" )\n",
    "\n",
    "    tRNAbed.close()\n",
    "\n",
    "    log.info(\"{} total tRNA gene sequences (undetermined and nmt sequences excluded)\".format(len(tRNA_dict)))\n",
    "    log.info(\"{} sequences with a match to Modomics dataset\".format(match_count))\n",
    "\n",
    "    with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "        SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "    # if clustering is not activated then write full gff and report on total SNPs written \n",
    "    if not cluster:\n",
    "        coverage_bed = tRNAbed.name\n",
    "        mod_lists = defaultdict(list)\n",
    "        Inosine_lists = defaultdict(list)\n",
    "        with open(out_dir + experiment_name + \"_tRNA.gff\",\"w\") as tRNAgff, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:    \n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for seq in tRNA_dict:\n",
    "                mod_lists[seq] = tRNA_dict[seq]['modified']\n",
    "                Inosine_lists[seq] = tRNA_dict[seq]['InosinePos']\n",
    "                tRNAgff.write(seq + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + seq + \"'\\n\")\n",
    "                isoacceptor_group = '-'.join(seq.split(\"-\")[:-2])\n",
    "                isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        # generate Stockholm alignment file for all tRNA transcripts and parse additional mods file\n",
    "        aligntRNA(temptRNATranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "        # add additional SNPs from extra file to list of modified positions, and ensure non-redundancy with set()\n",
    "        # index SNPs\n",
    "        # Format for SNP index (space separated):\n",
    "        # >snpID chromosomeName:position(1-based) RefMod\n",
    "        # e.g. >rs111 Homo_sapiens_nmt_tRNA-Leu-TAA-1-1_exp0:29 GN\n",
    "        for seq in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[seq]['modified'] = list(set(tRNA_dict[seq]['modified'] + additionalMods_sub[seq]['mods']))\n",
    "                mod_lists[seq] = list(set(mod_lists[seq] + additionalMods_sub[seq]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[seq])\n",
    "\n",
    "            # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "            # Position is 1-based for iit_store i.e. pos + 1\n",
    "            for (index, pos) in enumerate(mod_lists[seq]):\n",
    "                #if \"Gln-TTG\" in seq and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for seq in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[seq]['InosinePos'] = list(set(tRNA_dict[seq]['InosinePos'] + additionalInosines_sub[seq]['InosinePos']))\n",
    "                Inosine_lists[seq] = list(set(Inosine_lists[seq] + additionalInosines_sub[seq]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[seq])\n",
    "            total_snps += len(Inosine_lists[seq])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[seq]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions only if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for seq in Inosine_lists:\n",
    "                for pos in Inosine_lists[seq]:\n",
    "                    seq_records[seq].seq = seq_records[seq].seq[0:pos] + \"G\" + seq_records[seq].seq[pos+1:]\n",
    "\n",
    "        with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "            SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False\n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))\n",
    "        # empty mismatch dict to avoid error when returning it from this function\n",
    "        mismatch_dict = defaultdict(list)\n",
    "        insert_dict = defaultdict(dd_list)\n",
    "        del_dict = defaultdict(dd_list)\n",
    "        mod_lists = {tRNA:list() for tRNA in tRNA_dict.keys()}\n",
    "        Inosine_lists = {tRNA:data['InosinePos'] for tRNA, data in tRNA_dict.items()}\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list)\n",
    "        cluster_dict = defaultdict(list)\n",
    "\n",
    "    ##########################\n",
    "    # Cluster tRNA sequences #\n",
    "    ##########################\n",
    "\n",
    "    elif cluster:\n",
    "\n",
    "        log.info(\"**** Clustering tRNA sequences ****\")\n",
    "        log.info(\"Clustering tRNA sequences by {:.0%} similarity...\".format(cluster_id))\n",
    "        # dictionary of final centroid sequences\n",
    "        final_centroids = defaultdict()\n",
    "        # get dictionary of sequences for each anticodon and write to fastas\n",
    "        for anticodon in anticodon_list:\n",
    "            seq_set = {k:{'sequence':v['sequence'],'modified':v['modified']} for k,v in tRNA_dict.items() if v['anticodon'] == anticodon}\n",
    "            with open(temp_dir + anticodon + \"_allseqs.fa\",\"w\") as anticodon_seqs:\n",
    "                for sequence in seq_set:\n",
    "                    anticodon_seqs.write(\">\" + sequence + \"\\n\" + seq_set[sequence]['sequence'] + \"\\n\")\n",
    "            # run usearch on each anticodon sequence fatsa to cluster\n",
    "            cluster_cmd = [\"usearch\", \"-cluster_fast\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]\n",
    "            #cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"--sortedby\", \"other\" ,\"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]            #cluster_cmd = \"usearch -cluster_fast \" + temp_dir + anticodon + \"_allseqs.fa -sort length -id \" + str(cluster_id) + \" -centroids \" + temp_dir + anticodon + \"_centroids.fa -uc \" + temp_dir + anticodon + \"_clusters.uc &> /dev/null\"\n",
    "            subprocess.check_call(cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # sort clusters by size (i.e. number of members in cluster)\n",
    "            sort_cmd = [\"usearch\", \"-sortbysize\", temp_dir + anticodon + \"_centroids.fa\", \"-fastaout\", temp_dir + anticodon + \"_centroids_sort.fa\"]\n",
    "            subprocess.check_call(sort_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # recluster based on sorted by size clusters\n",
    "            final_cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_centroids_sort.fa\", \"-sortedby\", \"size\", \"-id\", str(cluster_id), \"-centroids\", temp_dir + anticodon + \"_centroidsFinal.fa\"]\n",
    "            subprocess.check_call(final_cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "        # combine centroids files into one file\n",
    "        for filename in glob.glob(temp_dir + \"*_centroidsFinal.fa\"):\n",
    "            with open(filename, \"r\") as fileh:\n",
    "                with open(temp_dir + \"all_centroids.fa\", \"a\") as outh:\n",
    "                    outh.write(fileh.read())\n",
    "        centroids = SeqIO.parse(temp_dir + \"all_centroids.fa\", \"fasta\")\n",
    "        for centroid in centroids:\n",
    "            centroid.id = centroid.id.split(\";\")[0]\n",
    "            final_centroids[centroid.id] = SeqRecord(Seq(str(centroid.seq).upper()), id = centroid.id) \n",
    "\n",
    "        # read cluster files, get nonredudant set of mod positions of all members of a cluster, create snp_records for writing SNP index\n",
    "        cluster_pathlist = Path(temp_dir).glob(\"**/*_clusters.uc\")\n",
    "        mod_lists = defaultdict(list) # stores non-redundant sets of mismatches and mod positions for clusters\n",
    "        Inosine_lists = defaultdict(list) # stores positions of inosines for clusters\n",
    "        snp_records = list()\n",
    "        cluster_dict = defaultdict(list) # info about clusters and members\n",
    "        mismatch_dict = defaultdict(list) # dictionary of mismatches only (not mod positions - required for exclusion from misincorporation analysis in mmQuant)\n",
    "        insert_dict = defaultdict(dd_list) # dictionary of insertions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        del_dict = defaultdict(dd_list) # dictionary of deletions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list) # for each cluster, list of members that mismatch at each position corresonding to mismatch dict - used for splitting clusters into isodecoders (splitClusters.py)\n",
    "        cluster_num = 0\n",
    "        total_snps = 0\n",
    "        total_inosines = 0\n",
    "        clusterbed = open(out_dir + experiment_name + \"_clusters.bed\",\"w\")\n",
    "        coverage_bed = clusterbed.name\n",
    "        clustergff = open(out_dir + experiment_name + \"_tRNA.gff\",\"w\")\n",
    "        for path in cluster_pathlist:\n",
    "            with open(str(path),\"r\") as cluster_file:\n",
    "                for line in cluster_file:\n",
    "                    line = line.strip()\n",
    "\n",
    "                    # Handle cluster centroids and initialise modified positions list\n",
    "                    if line.split(\"\\t\")[0] == \"S\":\n",
    "                        cluster_num += 1\n",
    "                        cluster_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        mod_lists[cluster_name] = tRNA_dict[cluster_name][\"modified\"]\n",
    "                        Inosine_lists[cluster_name] = tRNA_dict[cluster_name]['InosinePos']\n",
    "                        clusterbed.write(cluster_name + \"\\t0\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t\" + cluster_name + \"\\t1000\\t+\\n\" )\n",
    "                        clustergff.write(cluster_name + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + cluster_name + \"'\\n\")\n",
    "                        cluster_dict[cluster_name].append(cluster_name)\n",
    "                \n",
    "                    # Handle members of clusters\n",
    "                    elif line.split(\"\\t\")[0] == \"H\":\n",
    "                        member_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        cluster_name = line.split(\"\\t\")[9].split(\";\")[0]\n",
    "                        compr_aln = line.split(\"\\t\")[7]\n",
    "                        # if member of cluster is 100% identical (i.e. \"=\" in 8th column of cluster file)\n",
    "                        if compr_aln == \"=\":\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + tRNA_dict[member_name][\"modified\"]))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + tRNA_dict[member_name][\"InosinePos\"]))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "                        \n",
    "                        # if there are insertions or deletions in the centroid, edit member or centroid sequences to ignore these positions\n",
    "                        # and edit modified positions list in order to make non-redundant positions list, similar to next else statement\n",
    "                        elif re.search(\"[ID]\", compr_aln):\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            pos = 0\n",
    "                            adjust_pos_del = 0\n",
    "                            adjust_pos_ins = 0\n",
    "                            insertion_pos = list()\n",
    "                            deletion_pos = list()\n",
    "                            aln_list = re.split('(.*?[A-Z])', compr_aln)\n",
    "                            aln_list = list(filter(None, aln_list))\n",
    "\n",
    "                            for phrase in aln_list:\n",
    "                                \n",
    "                                if (\"M\" in phrase) and (phrase.split(\"M\")[0] != \"\"):\n",
    "                                    pos += int(phrase.split(\"M\")[0])\n",
    "                                elif (\"M\" in phrase) and (phrase.split(\"M\")[0] == \"\"):\n",
    "                                    pos += 1\n",
    "\n",
    "                                if (\"I\" in phrase) and (phrase.split(\"I\")[0] != \"\"):\n",
    "                                    insert_len = int(phrase.split(\"I\")[0])\n",
    "\n",
    "                                    for i in range(insert_len):\n",
    "                                        insertion_pos.append(pos+i)\n",
    "                                        #pos += 1\n",
    "                                elif (\"I\" in phrase) and (phrase.split(\"I\")[0] == \"\"):\n",
    "                                    insertion_pos.append(pos)\n",
    "                                    #pos += 1\n",
    "\n",
    "                                if (\"D\" in phrase) and (phrase.split(\"D\")[0] != \"\"):\n",
    "                                    delete_len = int(phrase.split(\"D\")[0])\n",
    "\n",
    "                                    for i in range(delete_len):\n",
    "                                        deletion_pos.append(pos)\n",
    "                                        pos += 1\n",
    "                                elif (\"D\" in phrase) and (phrase.split(\"D\")[0] == \"\"):\n",
    "                                    deletion_pos.append(pos)\n",
    "                                    pos += 1\n",
    "\n",
    "                            for delete in deletion_pos:\n",
    "                                adjust_pos_len = 0\n",
    "                                for insert in insertion_pos:\n",
    "                                    if insert < delete:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                new_delete = delete + adjust_pos_del + adjust_pos_len\n",
    "                                member_seq = member_seq[ :new_delete] + member_seq[new_delete+1: ]\n",
    "                                del_dict[cluster_name][new_delete].append(member_name)\n",
    "                                adjust_pos_del -= 1\n",
    "\n",
    "                            for index, insert in enumerate(insertion_pos):\n",
    "                                adjust_pos_len = 0\n",
    "                                for delete in deletion_pos:\n",
    "                                    if delete < insert:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                #if index != 0:\n",
    "                                #    if insert == insertion_pos[index-1] + 1:\n",
    "                                #        adjust_pos_ins -= 1\n",
    "                                new_insert = insert + adjust_pos_len + adjust_pos_ins\n",
    "                                member_seq = member_seq[ :new_insert] + cluster_seq[insert] + member_seq[new_insert: ]\n",
    "                                #insert_dict[cluster_name][new_insert-1].append(member_name)\n",
    "                                insert_dict[cluster_name][new_insert].append(member_name)\n",
    "                                #cluster_seq = cluster_seq[ :new_insert] + cluster_seq[new_insert+1: ]\n",
    "\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "                        # handle members that are not exact sequence matches but have no indels either\n",
    "                        # find mismatches and build non-redundant set\n",
    "                        else:\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "        clusterbed.close()\n",
    "\n",
    "        # Write cluster information to tsv and get number of unique sequences per cluster (i.e. isodecoders) for read count splitting\n",
    "        with open(out_dir + experiment_name + \"clusterInfo.txt\",\"w\") as clusterInfo, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:\n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            clusterInfo.write(\"tRNA\\tcluster_num\\tcluster_size\\tparent\\n\")\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for key, value in cluster_dict.items():\n",
    "                cluster_num = list(cluster_dict.keys()).index(key)\n",
    "                for member in value:\n",
    "                    clusterInfo.write(\"{}\\t{}\\t{}\\t{}\\n\".format(member, cluster_num, len(value), key))\n",
    "                    isoacceptor_group = '-'.join(member.split(\"-\")[:-2])\n",
    "                    isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        # generate Stockholm alignment file for cluster transcripts and process additional mods file\n",
    "        aligntRNA(clusterTranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "\n",
    "        log.info(\"{} clusters created from {} tRNA sequences\".format(len(cluster_dict),len(tRNA_dict)))\n",
    "\n",
    "        # update mod_lists with additional mods and write SNP index\n",
    "        for cluster in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[cluster]['modified'] = list(set(tRNA_dict[cluster]['modified'] + additionalMods_sub[cluster]['mods']))\n",
    "                mod_lists[cluster] = list(set(mod_lists[cluster] + additionalMods_sub[cluster]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(mod_lists[cluster]):\n",
    "                # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "                # Position is 1-based for iit_store i.e. pos + 1\n",
    "                #if \"Gln-TTG\" in cluster and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for cluster in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[cluster]['InosinePos'] = list(set(tRNA_dict[cluster]['InosinePos'] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "                Inosine_lists[cluster] = list(set(Inosine_lists[cluster] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[cluster])\n",
    "            total_snps += len(Inosine_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[cluster]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for cluster in Inosine_lists:\n",
    "                for pos in Inosine_lists[cluster]:\n",
    "                    final_centroids[cluster].seq = final_centroids[cluster].seq[0:pos] + \"G\" + final_centroids[cluster].seq[pos+1:]\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # rewrite edited cluster transcripts\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False        \n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))        \n",
    "    \n",
    "    # write outputs for indexing \n",
    "    with open(out_dir + experiment_name + \"_modificationSNPs.txt\", \"w\") as snp_file:\n",
    "        for item in snp_records:\n",
    "            snp_file.write('{}\\n'.format(item))\n",
    "    \n",
    "    shutil.rmtree(temp_dir)\n",
    "    \n",
    "    # Return coverage_bed (either tRNAbed or clusterbed depending on --cluster) for coverage calculation method\n",
    "    return(coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers = modsToSNPIndex(trnas, trnaout, mito_trnas, modifications, name, out, double_cca, snp_tolerance, cluster, cluster_id, posttrans, pretrnas, local_mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=False, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='hg19_HEK239vsK562', posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
