{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "#from mimseq import version\n",
    "#from mimseq.tRNAtools import modsToSNPIndex, generateGSNAPIndices, newModsParser, tidyFiles\n",
    "#from mimseq.tRNAmap import mainAlign\n",
    "#from mimseq.getCoverage import getCoverage, plotCoverage\n",
    "#from mimseq.mmQuant import generateModsTable, plotCCA\n",
    "#from mimseq.ssAlign import structureParser, modContext \n",
    "#from mimseq.splitClusters import splitIsodecoder, unsplitClusters, getIsodecoderSizes, writeIsodecoderTranscripts\n",
    "import sys, os, subprocess, logging, datetime, copy\n",
    "import argparse\n",
    "from pyfiglet import figlet_format\n",
    "from collections import defaultdict\n",
    "\n",
    "log = logging.getLogger(__name__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def restrictedFloat(x):\n",
    "## Method for restricting cluster_id and cov_diff argument to float between 0 and 1\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0 or x > 1.0:\n",
    "            raise argparse.ArgumentTypeError('{} not in range 0.0 - 1.0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "def restrictedFloat2(x):\n",
    "## Method for restricting min-cov argument to float between 0 and 1, or int greater than 1\n",
    "\n",
    "    try:\n",
    "        x = float(x)\n",
    "        if x < 0.0:\n",
    "            raise argparse.ArgumentTypeError('{} not greater than 0'.format(x))\n",
    "        return x\n",
    "    except ValueError:\n",
    "        raise argparse.ArgumentTypeError('{} not a real number'.format(x))\n",
    "\n",
    "\n",
    "\n",
    "parser = argparse.ArgumentParser(description = 'Custom high-throughput tRNA sequencing alignment and quantification pipeline\\\n",
    "    based on modification induced misincorporation cDNA synthesis.', add_help = True, usage = \"%(prog)s [options] sample data\")\n",
    "\n",
    "inputs = parser.add_argument_group(\"Input files\")\n",
    "inputs.add_argument('-s','--species', metavar='species', required = not ('-t' in sys.argv), dest = 'species', help = \\\n",
    "    'Species being analyzed for which to load pre-packaged data files (prioritized over -t, -o and -m). Options are: Hsap, Hsap38, Mmus, Scer, Spom, Dmel, Drer, Ecol', \\\n",
    "    choices = ['Hsap','Hsap38','Mmus','Scer','Spom','Dmel', 'Drer', 'Ecol'])\n",
    "inputs.add_argument('-t', '--trnas', metavar='genomic tRNAs', required = False, dest = 'trnas', help = \\\n",
    "    'Genomic tRNA fasta file, e.g. from gtRNAdb or tRNAscan-SE. Already avalable in data folder for a few model organisms.')\n",
    "inputs.add_argument('-o', '--trnaout', metavar = 'tRNA out file', required = (not '--species' or '-s' in sys.argv) or ('-t' in sys.argv), \n",
    "    dest = 'trnaout', help = 'tRNA.out file generated by tRNAscan-SE (also may be available on gtRNAdb). Contains information about tRNA features, including introns.')\n",
    "inputs.add_argument('-m', '--mito-trnas', metavar = 'mitochondrial tRNAs', required = False, dest = 'mito', \\\n",
    "    help = 'Mitochondrial tRNA fasta file. Should be downloaded from mitotRNAdb for species of interest. Already available in data folder for a few model organisms.')\n",
    "\n",
    "options = parser.add_argument_group(\"Program options\")\n",
    "options.add_argument('--pretRNAs', required = False, dest = 'pretrnas', action = 'store_true',\\\n",
    "    help = \"Input reference sequences are pretRNAs. Enabling this option will disable the removal of intron sequences and addition of 3'-CCA to generate \\\n",
    "    mature tRNA sequences. Useful for mapping and discovering pretRNA sequence reads.\")\n",
    "options.add_argument('--no-cluster', required = False, dest = 'cluster', action = 'store_false',\\\n",
    "    help = 'Disable usearch sequence clustering of tRNAs by isodecoder which drastically reduces the rate of multi-mapping reads. Default is enabled.')\n",
    "options.add_argument('--cluster-id', metavar = 'clustering identity threshold', dest = 'cluster_id', type = restrictedFloat, nargs = '?', default = 0.97,\\\n",
    "    required = False, help = 'Identity cutoff for usearch clustering between 0 and 1. Default is 0.97.')\n",
    "options.add_argument('--deconv-cov-ratio', metavar='deconvolution coverage threshold', dest='cov_diff', type = restrictedFloat, nargs = '?', default=0.5,\\\n",
    "    required=False, help=\"Threshold for ratio between coverage at 3' end and mismatch used for deconvolution. Coverage reductions greater than the threshold will result in non-deconvoluted sequences. \\\n",
    "        Default is 0.5 (i.e. less than 50%% reduction required for deconvolution).\")\n",
    "options.add_argument('--threads', metavar = 'thread number', required = False, dest = 'threads', type = int, \\\n",
    "    help = 'Set processor threads to use during read alignment and read counting.')\n",
    "options.add_argument('--posttrans-mod-off', required = False, dest = 'posttrans', action = 'store_true', \\\n",
    "    help = \"Disable post-transcriptional modification of tRNAs, i.e. addition of 3'-CCA and 5'-G (His) to mature sequences. Disable for certain \\\n",
    "    prokaryotes (e.g. E. coli) where this is genomically encoded. Leave enabled (default) for all eukaryotes.\")\n",
    "options.add_argument('--control-condition', metavar = 'control condition', required = True, dest = 'control_cond', \\\n",
    "    help = 'Name of control/wild-type condition as per user defined group specified in sample data input. This must exactly match the group name \\\n",
    "    specified in sample data. This is used for differential expression analysis so that results are always in the form mutant/treatment vs WT/control. REQUIRED')\n",
    "options.add_argument('--no-cca-analysis', required = False, dest = 'cca', action = 'store_false',\\\n",
    "    help = \"Disable analysis of 3'-CCA ends. When enabled, this calculates proportions of CC vs CCA ending reads per cluster and performs DESeq2 analysis. \\\n",
    "    Useful for comparing functional to non-functional mature tRNAs. Default is enabled.\")\n",
    "options.add_argument('--double-cca', required = False, dest = 'double_cca', action = \"store_true\",\\\n",
    "    help = \"Enable analysis of 3'-CCACCA tagging for tRNA degradation pathway. Note that this will alter the output of the CCA analysis pipeline.\")\n",
    "options.add_argument('--local-modomics', required=False, dest = 'local_mod', action='store_true',\\\n",
    "    help = \"Disable retrieval of Modomics data from online. Instead use older locally stored data. Warning - this leads\\\n",
    "        to usage of older Modomics data!\")\n",
    "\n",
    "align = parser.add_argument_group(\"GSNAP alignment options\")\n",
    "align.add_argument('--max-mismatches', metavar = 'allowed mismatches', required = False, dest = 'mismatches', type = float, \\\n",
    "    help = 'Maximum mismatches allowed. If specified between 0.0 and 1.0, then treated as a fraction of read length. Otherwise, treated as \\\n",
    "    integer number of mismatches. Default is an automatic ultrafast value calculated by GSNAP; see GSNAP help for more info.')\n",
    "align.add_argument('--remap-mismatches', metavar = 'allowed mismatches for remap', required = False, dest = 'remap_mismatches', type = float,\\\n",
    "    help = 'Maximum number of mismatches allowed during remapping of all reads. Treated similarly to --max-mismatches. This is important to control misalignment of reads to similar clusters/tRNAs \\\n",
    "    Note that the SNP index will be updated with new SNPs from the first round of alignment and so this should be relatively small to prohibit misalignment.')\n",
    "align.add_argument('--no-snp-tolerance', required = False, dest = 'snp_tolerance', action = 'store_false',\\\n",
    "    help = 'Disable GSNAP SNP-tolerant read alignment, where known modifications from Modomics are mapped as SNPs. Default is enabled.')\n",
    "\n",
    "\n",
    "outputs = parser.add_argument_group(\"Output options\")\n",
    "outputs.add_argument('-n', '--name', metavar = 'experiment name', required = True, dest = 'name', help = \\\n",
    "    'Name of experiment. Note, output files and indices will have this as a prefix. REQUIRED')\n",
    "outputs.add_argument('--out-dir', metavar = 'output directory', required = False, dest = 'out', help = \\\n",
    "    'Output directory. Default is current directory. Cannot be an existing directory.')\n",
    "outputs.add_argument('--keep-temp', required = False, dest='keep_temp', action = 'store_true', help = \\\n",
    "    'Keeps multi-mapping and unmapped bam files from GSNAP alignments. Default is false.')\n",
    "\n",
    "bedtools = parser.add_argument_group(\"Bedtools coverage options\")\n",
    "bedtools.add_argument('--min-cov', metavar = 'Minimum coverage per cluster', required = False, dest = 'min_cov', type = restrictedFloat2, default=0.0005, \\\n",
    "    help = \"Minimum coverage per cluster required to include this cluster in coverage plots, modification analysis, and 3'-CCA analysis. \\\n",
    "    Can be a fraction of total mapped reads between 0 and 1, or an integer of absolute coverage. Any cluster not meeting the threshold in 1 or more sample will be excluded. \\\n",
    "    Note that all clusters are included for differential expression analysis with DESeq2. Default = 0.0005 (0.05%% mapped reads).\")\n",
    "bedtools.add_argument('--max-multi', metavar = 'Bedtools coverage multithreading', required = False, dest = 'max_multi', type = int, \\\n",
    "    help = 'Maximum number of bam files to run bedtools coverage on simultaneously. Increasing this number reduces processing time\\\n",
    "    by increasing number of files processed simultaneously. However, depending on the size of the bam files to process and\\\n",
    "    available memory, too many files processed at once can cause termination of mim-tRNAseq due to insufficient memory. If\\\n",
    "    mim-tRNAseq fails during coverage calculation, lower this number. Increase at your own discretion. Default is 3.')\n",
    "\n",
    "remapping = parser.add_argument_group(\"Analysis of unannotated modifications and realignment\")\n",
    "remapping.add_argument('--remap', required = False, dest = 'remap', action = 'store_true',\\\n",
    "    help = 'Enable detection of unannotated (potential) modifications from misincorporation data. These are defined as having a total misincorporation rate\\\n",
    "    higher than the threshold set with --misinc-thresh. These modifications are then appended to already known ones, and read alignment is reperformed.\\\n",
    "    Very useful for poorly annotated species in Modomics. Due to realignment and misincorporation parsing, enabling this option slows the analysis down considerably.')\n",
    "remapping.add_argument('--misinc-thresh', metavar = 'threshold for unannotated mods', dest = 'misinc_thresh', type = restrictedFloat, nargs = '?', default = 0.1,\\\n",
    "    required = False, help = 'Threshold of total misincorporation rate at a position in a cluster used to call unannotated modifications. Value between 0 and 1, default is 0.1  (10%% misincorporation).')\n",
    "\n",
    "parser.add_argument('--version', action='version', version='%(prog)s {}'.format(version.__version__), help = 'Show version number and exit')\n",
    "parser.add_argument('sampledata', help = 'Sample data sheet in text format, tab-separated. Column 1: full path to fastq (or fastq.gz). Column 2: condition/group.')\n",
    "\n",
    "parser.set_defaults(threads=1, out=\"./\", max_multi = 3, min_cov = 0, mito = '', cov_diff = 0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "args = parser.parse_args('--species Hsap --cluster-id 0.95 --threads 1 --min-cov 2000 --max-mismatches 0.1 --control-condition HEK293T -n hg19_test --out-dir hg19_HEK239vsK562 --max-multi 1 --remap --remap-mismatches 0.075 sampleData_HEKvsK562.txt'.split())\n",
    "\n",
    "\n",
    "try:\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(__file__))\n",
    "except NameError:\n",
    "    filepath = '/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq/mimseq.py'\n",
    "    args.repo_path = os.path.dirname(os.path.realpath(filepath))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           _                 _   ____  _   _    _                   \n",
      " _ __ ___ (_)_ __ ___       | |_|  _ \\| \\ | |  / \\   ___  ___  __ _ \n",
      "| '_ ` _ \\| | '_ ` _ \\ _____| __| |_) |  \\| | / _ \\ / __|/ _ \\/ _` |\n",
      "| | | | | | | | | | | |_____| |_|  _ <| |\\  |/ ___ \\\\__ \\  __/ (_| |\n",
      "|_| |_| |_|_|_| |_| |_|      \\__|_| \\_\\_| \\_/_/   \\_\\___/\\___|\\__, |\n",
      "                                                                 |_|\n",
      "\n",
      " Modification-induced misincorporation analysis of tRNA sequencing data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(figlet_format('mim-tRNAseq', font='standard'))\n",
    "print(\" Modification-induced misincorporation analysis of tRNA sequencing data\\n\")\n",
    "if args.pretrnas:\n",
    "    if args.cca:\n",
    "        log.warning(\"Disabling CCA analysis in pre-tRNA mode...\")\n",
    "        args.cca = False\n",
    "    if args.cluster:\n",
    "        log.warning(\"Disabling tRNA clustering in pre-tRNA mode...\")\n",
    "        args.cluster = False\n",
    "# Check that control_cond exists in sample data\n",
    "conditions = list()\n",
    "with open(args.sampledata, \"r\") as sampleData:\n",
    "    for line in sampleData:\n",
    "        line = line.strip()\n",
    "        if not line.startswith(\"#\"):\n",
    "            conditions.append(line.split(\"\\t\")[1])\n",
    "if args.control_cond not in conditions:\n",
    "    raise argparse.ArgumentTypeError('{} not a valid condition in {}'.format(args.control_cond, args.sampledata))\n",
    "if not args.species and not (args.trnas or args.trnaout):\n",
    "    parser.error('Must specify valid --species argument or supply -t (tRNA sequences) and -o (tRNAscan out file)!')                        \n",
    "else:\n",
    "    if args.species:\n",
    "        if args.species == 'Hsap':\n",
    "            args.trnas = args.repo_path + \"/data/hg19-eColitK/hg19_eColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg19-eColitK/hg19_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg19-eColitK/hg19-mitotRNAs.fa\"\n",
    "        if args.species == 'Hsap38':\n",
    "            args.trnas = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/hg38-eColitK/hg38-tRNAs-detailed.out\"\n",
    "            args.mito = args.repo_path + \"/data/hg38-eColitK/hg38-mitotRNAs.fa\"\n",
    "        if args.species == 'Scer':\n",
    "            args.trnas = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColitK.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/sacCer3-eColitK/sacCer3_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/sacCer3-eColitK/sacCer3-mitotRNAs.fa\"\n",
    "        if args.species == 'Mmus':\n",
    "            args.trnas = args.repo_path + \"/data/mm10-eColitK/mm10_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/mm10-eColitK/mm10_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/mm10-eColitK/mm10-mitotRNAs.fa\"\n",
    "        if args.species == 'Spom':\n",
    "            args.trnas = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_972H-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/schiPomb-eColitK/schiPomb_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/schiPomb-eColitK/schiPomb-mitotRNAs.fa\"\n",
    "        if args.species == 'Dmel':\n",
    "            args.trnas = args.repo_path + \"/data/dm6-eColitK/dm6_eColitK-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/dm6-eColitK/dm6_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/dm6-eColitK/dm6-mitotRNAs.fa\"\n",
    "        if args.species == 'Drer':\n",
    "            args.trnas = args.repo_path + \"/data/danRer11-eColitK/danRer11_eColitK_filtered.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/danRer11-eColitK/danRer11_eschColi-tRNAs.out\"\n",
    "            args.mito = args.repo_path + \"/data/danRer11-eColitK/danRer11-mitotRNAs.fa\"\n",
    "        if args.species == 'Ecol':\n",
    "            args.trnas = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.fa\"\n",
    "            args.trnaout = args.repo_path + \"/data/eschColi-K_12_MG1655-tRNAs/eschColi_K_12_MG1655-tRNAs.out\"\n",
    "            args.mito = ''\n",
    "    else:\n",
    "        args.species = args.trnas.split(\"/\")[-1].split(\".\")[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "trnas, trnaout, name, species, out, cluster, cluster_id, cov_diff, posttrans, control_cond, threads, max_multi, snp_tolerance, keep_temp, cca, double_cca, min_cov, mismatches, remap, remap_mismatches, misinc_thresh, mito_trnas, pretrnas, local_mod, sample_data = args.trnas, args.trnaout, args.name, args.species, args.out, args.cluster, args.cluster_id, args.cov_diff, args.posttrans, args.control_cond, args.threads, args.max_multi, args.snp_tolerance, args.keep_temp, args.cca, args.double_cca, args.min_cov, args.mismatches, args.remap, args.remap_mismatches, args.misinc_thresh, args.mito, args.pretrnas, args.local_mod, args.sampledata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=False, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='hg19_HEK239vsK562', posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, repo_path='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq', sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "map_round = 1 #first round of mapping\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/mimseq'"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args.repo_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Here make tmp folder, start logging etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_dir = out_dir + \"/tmp/\"\n",
    "\n",
    "try:\n",
    "    os.mkdir(temp_dir)\n",
    "except FileExistsError:\n",
    "    log.warning(\"Temp folder present - previous run interrupted? Overwriting old temp files...\\n\")\n",
    "\n",
    "###################################################\n",
    "## Main code for matching and SNP index building ##\n",
    "###################################################\n",
    "\n",
    "# remove spurious CCA seen in some sequences (i.e. genomically encoded or artifact from gtRNAdb or tRNAScan-SE prediction)\n",
    "# to do this, remove previously added CCA (intronRemover()) that falls out of canonical structure using ssAlign module\n",
    "\n",
    "with open(out_dir + 'temptRNAseqs.fa', 'w') as tempSeqs:\n",
    "    for seq in tRNA_dict:\n",
    "        tempSeqs.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "\n",
    "aligntRNA(tempSeqs.name, out_dir)\n",
    "extra_cca = extraCCA()\n",
    "\n",
    "for record in extra_cca:\n",
    "    tRNA_dict[record]['sequence'] = tRNA_dict[record]['sequence'][:-3]\n",
    "\n",
    "os.remove(tempSeqs.name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "import re, copy, sys, os, shutil, subprocess, logging, glob\n",
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "import json\n",
    "\n",
    "\n",
    "#log = logging.getLogger(__name__)\n",
    "\n",
    "class TRNA():\n",
    "    '''Object for tRNAs, their sequences, modifications and other information.'''\n",
    "    \n",
    "    def __init__(self, cmd_args):\n",
    "        self.inp_repo_path = cmd_args.repo_path\n",
    "        self.inp_gtRNAdb = cmd_args.trnas\n",
    "        self.inp_tRNAscan_out = cmd_args.trnaout\n",
    "        self.inp_mitotRNAs = cmd_args.mito\n",
    "        self.inp_ptm_off = cmd_args.posttrans\n",
    "        self.inp_double_cca = cmd_args.double_cca\n",
    "        self.inp_pretrnas = cmd_args.pretrnas\n",
    "        self.inp_local_mod = cmd_args.local_mod\n",
    "        \n",
    "        # Read intron boundaries into dict\n",
    "        self.intron_dict = self.read_introns(cmd_args.trnaout)\n",
    "        # Read tRNAs, minus introns, into dict\n",
    "        self.tRNA_dict = self.read_tRNAs(cmd_args.trnas, cmd_args.mito, cmd_args.posttrans, cmd_args.double_cca, cmd_args.pretrnas)\n",
    "        # Read table of tRNA modifications into dict\n",
    "        self.mod_dict = self.mod_parser(cmd_args.repo_path)\n",
    "        # Get species of input tRNA seqs to subset full Modomics table\n",
    "        self.species_set = {val['species'] for val in self.tRNA_dict.values()}\n",
    "        # Download and process Modomics data\n",
    "        self.modomics_dict = self.fetch_modomics(cmd_args.repo_path, cmd_args.local_mod)\n",
    "        \n",
    "\n",
    "    # Private methods for default dict.\n",
    "    # Used to create nested defaultdicts.\n",
    "    # Can be done with lambda functions (e.g. tRNA_dict = defaultdict(lambda: defaultdict()))\n",
    "    # But lambda functions cannot be pickled, and pickling is required for parallelization with multiprocessing.\n",
    "    def __dd(self):\n",
    "        return(defaultdict())\n",
    "    def __dd_list(self):\n",
    "        return(defaultdict(list))\n",
    "\n",
    "    def read_introns(self, trnaout_path):\n",
    "        '''Build dictionary of intron locations.'''\n",
    "        intron_dict = {}\n",
    "        with open(trnaout_path, 'r') as trnaout_file:\n",
    "            for line in trnaout_file:\n",
    "                cols = line.split()\n",
    "                if line.startswith(\"chr\"):\n",
    "                    tRNA_ID = cols[0] + \".trna\" + cols[1]\n",
    "                    tRNA_start = int(cols[2])\n",
    "                    intron_start = int(cols[6])\n",
    "                    intron_stop = int(cols[7])\n",
    "                    # If intron boundaries are not 0, i.e. there is an intron then add to dict\n",
    "                    if (intron_start > 0) & (intron_stop > 0):\n",
    "                        if tRNA_start > intron_start: # reverse strand\n",
    "                            intron_start = tRNA_start - intron_start\n",
    "                            intron_stop = tRNA_start - intron_stop + 1 # python 0 indexing\n",
    "                        else: # forward strand\n",
    "                            intron_start = intron_start - tRNA_start\n",
    "                            intron_stop = intron_stop - tRNA_start + 1 # python 0 indexing\n",
    "                        intron_dict[tRNA_ID] = {'intron_start': intron_start, 'intron_stop': intron_stop}\n",
    "\n",
    "            # log.info(\"{} introns registered...\".format(len(intron_dict)))\n",
    "            print(\"{} introns registered...\".format(len(intron_dict)))\n",
    "            return(intron_dict)\n",
    "\n",
    "    def read_tRNAs(self, trna_path, mitotrna_path, ptm_off, double_cca, pretrnas):\n",
    "        '''Read tRNA sequences from input files, remove introns and add CCA.'''\n",
    "        def remove_intron (intron_dict, seq_id, seq_obj, ptm_off, double_cca):\n",
    "            '''Use intron_dict to remove introns plus add CCA and 5' G for His (if eukaryotic).'''\n",
    "            # Find a match, slice intron and add G and CCA\n",
    "            ID = re.search(\"tRNAscan-SE ID: (.*?)\\).|\\((chr.*?)-\", seq_obj.description).groups()\n",
    "            ID = list(filter(None, ID))[0]\n",
    "            if ID in intron_dict:\n",
    "                seq = str(seq_obj.seq[:intron_dict[ID]['intron_start']] + seq_obj.seq[intron_dict[ID]['intron_stop']:])\n",
    "            else:\n",
    "                seq = str(seq_obj.seq)\n",
    "            if not ptm_off:\n",
    "                if double_cca:\n",
    "                    seq = seq + 'CCACCA'\n",
    "                else:\n",
    "                    seq = seq + 'CCA'\n",
    "                if '-His-' in seq_id:\n",
    "                    seq = 'G' + seq\n",
    "            return(seq)\n",
    "\n",
    "        # Build dict of sequences from input tRNA fasta\n",
    "        # Start with cyto\n",
    "        tRNA_dict = defaultdict(self.__dd)\n",
    "        intron_count = 0\n",
    "        seq_set = set()\n",
    "        for seq_id, seq_obj in SeqIO.to_dict(SeqIO.parse(trna_path, \"fasta\")).items():\n",
    "            # Only add to dict if not undetermined sequence (Und)\n",
    "            # or nuclear-encoded mitochondrial tRNA (mnt)\n",
    "            if not (re.search('Und', seq_id) or re.search('nmt', seq_id)):\n",
    "                if not pretrnas:\n",
    "                    tRNAseq = remove_intron(self.intron_dict, seq_id, seq_obj, ptm_off, double_cca)\n",
    "                    intron_count += 1 if len(tRNAseq) < len(seq_obj.seq) else 0\n",
    "                else:\n",
    "                    tRNAseq = str(seq_obj.seq)\n",
    "                if tRNAseq not in seq_set: # Toss away duplicate tRNA sequences\n",
    "                    tRNA_dict[seq_id]['sequence'] = tRNAseq\n",
    "                    tRNA_dict[seq_id]['species'] = ' '.join(seq_id.split('_')[0:2])\n",
    "                    tRNA_dict[seq_id]['type'] = 'cyto'\n",
    "                    seq_set.add(tRNAseq)\n",
    "\n",
    "        # Then add mito (if given)\n",
    "        if mitotrna_path:\n",
    "            mito_count = defaultdict(int) # for unique tRNA naming\n",
    "            # Read each mito tRNA, edit sequence header to match nuclear genes as above and add to tRNA_dict\n",
    "            for seq_id, seq_obj in SeqIO.to_dict(SeqIO.parse(mitotrna_path, 'fasta')).items():\n",
    "                id_parts = seq_id.split(\"|\")\n",
    "                anticodon = id_parts[4]\n",
    "                amino_acid = id_parts[3][0:3] # e.g. extract Leu from Leu1/Leu2\n",
    "                mito_count[anticodon] += 1\n",
    "                new_id = id_parts[1] + \"_mito_tRNA-\" + amino_acid + \"-\" + id_parts[4] + \"-\" + str(mito_count[anticodon]) + \"-1\"\n",
    "                if double_cca:\n",
    "                    tRNAseq = str(seq_obj.seq) + \"CCACCA\"\n",
    "                else:\n",
    "                    tRNAseq = str(seq_obj.seq) + \"CCA\"\n",
    "                if tRNAseq not in seq_set: # Toss away duplicate tRNA sequences\n",
    "                    tRNA_dict[new_id]['sequence'] = tRNAseq\n",
    "                    tRNA_dict[new_id]['type'] = 'mito'\n",
    "                    tRNA_dict[new_id]['species'] = ' '.join(id_parts[1].split('_')[0:2])\n",
    "                    seq_set.add(tRNAseq)\n",
    "\n",
    "            num_cyto = sum(1 for val in tRNA_dict.values() if val['type'] == 'cyto')\n",
    "            num_mito = sum(1 for val in tRNA_dict.values() if val['type'] == 'mito')\n",
    "            # log.info(\"Removed {} introns (including for duplicate tRNAs)\".format(intron_count))\n",
    "            # log.info(\"{} cytosolic and {} mitochondrial tRNA sequences imported\".format(num_cytosilic, num_mito))\n",
    "            print(\"Removed {} introns (including for duplicate tRNAs)\".format(intron_count))\n",
    "            print(\"{} cytosolic and {} mitochondrial unique tRNA sequences imported\".format(num_cyto, num_mito))\n",
    "            return(tRNA_dict)\n",
    "\n",
    "    def mod_parser(self, repo_path):\n",
    "        '''Parse tRNA and modifications.'''\n",
    "        mod_path = Path(repo_path + '/modifications.txt')\n",
    "        try:\n",
    "            _ = mod_path.resolve(strict=True)\n",
    "        except FileNotFoundError:\n",
    "            raise Exception('Expected modifications file but found none at this location: {}'.format(mod_path))\n",
    "        \n",
    "        # Read modifications and build dict\n",
    "        with open(mod_path, 'r', encoding='utf-8') as mod_file:\n",
    "            mod_dict = dict()\n",
    "            for line in mod_file:\n",
    "                if not line.startswith(\"#\"):\n",
    "                    name, abbr, ref, mod = list(map(str.strip, line.split('\\t')))\n",
    "                    # Replace unknown modifications with reference of N\n",
    "                    if not ref:\n",
    "                        ref = 'N'\n",
    "                    if mod:\n",
    "                        mod_dict[mod] = {'name':name, 'abbr':abbr, 'ref':ref}\n",
    "        return(mod_dict)\n",
    "\n",
    "    def fetch_modomics(self, repo_path, local_mod):\n",
    "        '''Download Modomics modified tRNA data (or use local version).'''\n",
    "        def unmodify_seq(seq, mod_dict):\n",
    "            '''Change modified bases into standard ACGT in input sequence.'''\n",
    "            new_seq = list()\n",
    "            for base in seq:\n",
    "                # For insertions ('_') make reference N - this is not described in the modifications table\n",
    "                if base == '_':\n",
    "                    unmod_base = 'N'\n",
    "                else:\n",
    "                    unmod_base = mod_dict[base]['ref']\n",
    "                    # Change queuosine to G (reference is preQ0base in modification file)\n",
    "                    if unmod_base == 'preQ0base':\n",
    "                        unmod_base = 'G'\n",
    "                new_seq.append(unmod_base)\n",
    "            return(''.join(new_seq).replace('U', 'T'))\n",
    "\n",
    "        #log.info(\"Downloading modomics database...\")\n",
    "        print(\"Downloading modomics database...\")\n",
    "        # Get full Modomics modified tRNA data from web\n",
    "        if not local_mod:\n",
    "            try:\n",
    "                response = requests.get(\"http://www.genesilico.pl/modomics/api/sequences?&RNAtype=tRNA\")\n",
    "                response.raise_for_status()\n",
    "                modomics_json = response.json()\n",
    "                #log.info(\"Modomics retrieved...\")\n",
    "                print(\"Modomics retrieved...\")\n",
    "            except Exception as http_err:\n",
    "                #log.error(\"Failed to download Modomics database! Error: {}. Check status of Modomics webpage. Using local Modomics files...\".format(http_err))\n",
    "                print(\"Failed to download Modomics database! Error: {}. Check status of Modomics webpage. Using local Modomics files...\".format(http_err))\n",
    "                modomics_path = repo_path + '/data/modomics.json'\n",
    "                with open(modomics_path, encoding=\"utf-8\") as mod_fh:\n",
    "                    modomics_json = json.load(mod_fh)\n",
    "            except Exception as err:\n",
    "                modomics_path = repo_path + '/data/modomics.json'\n",
    "                #log.error(\"Error in loading local Modomics files: {}\".format(err))\n",
    "                print(\"Error in loading local Modomics files: {}\".format(err))\n",
    "                raise Exception('Failed to open and read local Modomics JSON file: {}'.format(modomics_path))\n",
    "        else:\n",
    "            #log.warning(\"Retrieval of Modomics database disabled. Using local files instead...\")\n",
    "            print(\"Retrieval of Modomics database disabled. Using local files instead...\")\n",
    "            modomics_path = repo_path + '/data/modomics.json'\n",
    "            with open(modomics_path, encoding=\"utf-8\") as mod_fh:\n",
    "                modomics_json = json.load(mod_fh)\n",
    "\n",
    "        # Build Modomics_dict from JSON data\n",
    "        modomics_dict = dict()\n",
    "        species_count = defaultdict(int)\n",
    "        log.info(\"Parsing Modomics JSON data...\")\n",
    "        for entry in modomics_json.values():\n",
    "            mod_species = entry['organism']\n",
    "            if mod_species in self.species_set:\n",
    "                species_count[mod_species] += 1\n",
    "                anticodon = entry['anticodon']\n",
    "                new_anticodon = unmodify_seq(anticodon, self.mod_dict)\n",
    "                if \"N\" in new_anticodon:\n",
    "                    continue\n",
    "                # Check amino acid name in modomics - set to iMet if equal to Ini to match gtRNAdb\n",
    "                amino_acid = entry['subtype']\n",
    "                if amino_acid == 'Ini':\n",
    "                    amino_acid = 'iMet'\n",
    "\n",
    "                # Unique names for duplicates\n",
    "                curr_id = str(mod_species.replace(' ','_') + '_tRNA-' + amino_acid + '-' + new_anticodon)\n",
    "                duplicate_id = 0\n",
    "                while curr_id in modomics_dict and duplicate_id > 0:\n",
    "                    duplicate_id += 1\n",
    "                    curr_id = \"-\".join(curr_id.split('-')[0:3]) + '-' + str(duplicate_id)\n",
    "\n",
    "                tRNA_type = 'cyto' if 'cytosol' in entry['organellum'] else 'mito'\n",
    "                seq = entry['seq'].replace('U', 'T').replace('-', '')\n",
    "                unmod_seq = unmodify_seq(seq, self.mod_dict)\n",
    "                # Return list of modified nucl and inosines indices and add to modomics_dict\n",
    "                # add unmodified seq to modomics_dict by lookup to modifications\n",
    "                mods = list('''\"KR'OYW⊆X*[''')\n",
    "                mod_idx = [i for i, base in enumerate(seq) if base in mods]\n",
    "                insosine_idx = [i for i, base in enumerate(seq) if base == 'I']\n",
    "                modomics_dict[curr_id] = {'seq':seq, 'type':tRNA_type, 'anticodon':new_anticodon, 'modified_idx':mod_idx, 'unmod_seq':unmod_seq, 'insosine_idx':insosine_idx}\n",
    "\n",
    "        for s in self.species_set:\n",
    "            #log.info('Number of Modomics entries for {}: {}'.format(s, species_count[s]))\n",
    "            print('Number of Modomics entries for {}: {}'.format(s, species_count[s]))\n",
    "        return(modomics_dict)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [],
   "source": [
    "args.local_mod = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34 introns registered...\n",
      "Removed 33 introns (including for duplicate tRNAs)\n",
      "399 cytosolic and 22 mitochondrial unique tRNA sequences imported\n",
      "Downloading modomics database...\n",
      "Retrieval of Modomics database disabled. Using local files instead...\n",
      "Number of Modomics entries for Escherichia coli: 43\n",
      "Number of Modomics entries for Homo sapiens: 45\n"
     ]
    }
   ],
   "source": [
    "trna_obj = TRNA(args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Escherichia_coli_tRNA-Ala-TGC': {'seq': 'GGGGCTATAGCTCAGCDGGGAGAGCGCCTGCTTVGCACGCAGGAG7TCTGCGGTPCGATCCCGCATAGCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGGCTATAGCTCAGCTGGGAGAGCGCCTGCTTTGCACGCAGGAGGTCTGCGGTTCGATCCCGCATAGCTCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ala-GGC': {'seq': 'GGGGCTATAGCTCAGCDGGGAGAGCGCTTGCATGGCATGCAAGAG7TCAGCGGTPCGATCCCGCTTAGCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGGCTATAGCTCAGCTGGGAGAGCGCTTGCATGGCATGCAAGAGGTCAGCGGTTCGATCCCGCTTAGCTCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Arg-ACG': {'seq': 'GCATCCG4AGCTCAGCDGGADAGAGTACTCGG%TICG/ACCGAGCG7XCGGAGGTPCGAATCCTCCCGGATGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'ACG',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GCATCCGTAGCTCAGCTGGATAGAGTACTCGGCTACGAACCGAGCGGTCGGAGGTTCGAATCCTCCCGGATGCACCA',\n",
       "  'insosine_idx': [34]},\n",
       " 'Escherichia_coli_tRNA-Arg-CCG': {'seq': 'GCGCCCGTAGCTCAGCDGGADAGAGCGCTGCC%TCCGKAGGCAGAG7TCTCAGGTPCGAATCCTGTCGGGCGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGCCCGTAGCTCAGCTGGATAGAGCGCTGCCCTCCGGAGGCAGAGGTCTCAGGTTCGAATCCTGTCGGGCGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Arg-TCT': {'seq': 'GCGCCCTTAGCTCAGTTGGATAGAGCAACGAC%T{CT6AGPCGTGGGCCGCAGGTPCGAATCCTGCAGGGCGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGCCCTTAGCTCAGTTGGATAGAGCAACGACCTTCTAAGTCGTGGGCCGCAGGTTCGAATCCTGCAGGGCGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Asn-GTT': {'seq': 'TCCTCTG4AGTTCAGDCGGDAGAACGGCGGACTQTT6APCCGTAT7TCACTGGTPCGAGTCCAGTCAGAGGAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TCCTCTGTAGTTCAGTCGGTAGAACGGCGGACTGTTAATCCGTATGTCACTGGTTCGAGTCCAGTCAGAGGAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Asp-GTC': {'seq': 'GGAGCGG4AGTTCAGDCGGDDAGAATACCTGCCT⊄TC/CGCAGGGG7TCGCGGGTPCGAGTCCCGPCCGTTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGAGCGGTAGTTCAGTCGGTTAGAATACCTGCCTGTCACGCAGGGGGTCGCGGGTTCGAGTCCCGTCCGTTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Cys-GCA': {'seq': 'GGCGCGT4AACAAAGCGGDDATGTAGCGGAPTGCA*APCCGTCTAGTCCGGTPCGACTCCGGAACGCGCCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCA',\n",
       "  'modified_idx': [35],\n",
       "  'unmod_seq': 'GGCGCGTTAACAAAGCGGTTATGTAGCGGATTGCAAATCCGTCTAGTCCGGTTCGACTCCGGAACGCGCCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gln-TTG': {'seq': 'TGGGGTA4CGCCAAGC#GDAAGGCACCGGTJT$TG/PACCGGCATTCCCTGGTPCGAATCCAGGTACCCCAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TGGGGTATCGCCAAGCGGTAAGGCACCGGTTTTTGATACCGGCATTCCCTGGTTCGAATCCAGGTACCCCAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gln-CTG': {'seq': 'TGGGGTA4CGCCAAGC#GDAAGGCACCGGAJTCTG/PPCCGGCATTCCGAGGTPCGAATCCTCGTACCCCAGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TGGGGTATCGCCAAGCGGTAAGGCACCGGATTCTGATTCCGGCATTCCGAGGTTCGAATCCTCGTACCCCAGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Glu-TTC': {'seq': 'GTCCCCT4CGTCPAGAGGCCCAGGACACCGCCCTSTC/CGGCGGTAACAGGGGTPCGAATCCCCTAGGGGACGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GTCCCCTTCGTCTAGAGGCCCAGGACACCGCCCTTTCACGGCGGTAACAGGGGTTCGAATCCCCTAGGGGACGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-CCC': {'seq': 'GCGGGCG4AGTTCAATGGDAGAACGAGAGCTTCCCAAGCTCTATACGAGGGTPCGATTCCCTTCGCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGCGTAGTTCAATGGTAGAACGAGAGCTTCCCAAGCTCTATACGAGGGTTCGATTCCCTTCGCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-TCC': {'seq': 'GCGGGCATCGTATAATGGCTATTACCTCAGCCT{CCAAGCTGATGATGCGGGTPCGATTCCCGCTGCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGCATCGTATAATGGCTATTACCTCAGCCTTCCAAGCTGATGATGCGGGTTCGATTCCCGCTGCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Gly-GCC': {'seq': 'GCGGGAA4AGCTCAGDDGGDAGAGCACGACCTTGCCAAGGTCGGG7TCGCGAGTPCGAGTCTCGTTTCCCGCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCGGGAATAGCTCAGTTGGTAGAGCACGACCTTGCCAAGGTCGGGGTCGCGAGTTCGAGTCTCGTTTCCCGCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-His-GTG': {'seq': 'GGTGGCTA4AGCTCAGDDGGDAGAGCCCTGGATTQTG/PPCCAGTT7TCGTGGGTPCGAATCCCATTAGCCACCCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTG',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGGCTATAGCTCAGTTGGTAGAGCCCTGGATTGTGATTCCAGTTGTCGTGGGTTCGAATCCCATTAGCCACCCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ile-GAT': {'seq': 'AGGCTTGTAGCTCAGGDGGDDAGAGCGCACCCCTGAT6AGGGTGAG7XCGGTGGTPCAAGTCCACPCAGGCCTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAT',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'AGGCTTGTAGCTCAGGTGGTTAGAGCGCACCCCTGATAAGGGTGAGGTCGGTGGTTCAAGTCCACTCAGGCCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ile-CAT': {'seq': 'GGCCCCT4AGCTCAGT#GDDAGAGCAGGCGACT}AT6APCGCTTG7XCGCTGGTPCAAGTCCAGCAGGGGCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GGCCCCTTAGCTCAGTGGTTAGAGCAGGCGACTCATAATCGCTTGGTCGCTGGTTCAAGTCCAGCAGGGGCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-CAG': {'seq': 'GCGAAGGTGGCGGAADD#GDAGACGCGCTAGCTTCAGKPGPTAGTGTCCTTACGGACGTGGGGGTPCAAGTCCCCCCCCTCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGAAGGTGGCGGAATTGGTAGACGCGCTAGCTTCAGGTGTTAGTGTCCTTACGGACGTGGGGGTTCAAGTCCCCCCCCTCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-GAG': {'seq': 'GCCGAGGTGGTGGAADD#GDAGACACGCTACCTTGAGKPGGTAGTGCCCAATAGGGCTTACGGGTPCAAGTCCCGTCCTCGGTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCGAGGTGGTGGAATTGGTAGACACGCTACCTTGAGGTGGTAGTGCCCAATAGGGCTTACGGGTTCAAGTCCCGTCCTCGGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-TAA': {'seq': 'GCCCGGA4GGTGGAADC#GDAGACACAAGGGAPT)AA*APCCCTCGGCGTTCGCGCTGTGCGGGTPCAAGTCCCGCTCCGGGTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCCGGATGGTGGAATCGGTAGACACAAGGGATTTAAAATCCCTCGGCGTTCGCGCTGTGCGGGTTCAAGTCCCGCTCCGGGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-CAA': {'seq': 'GCCGAAG4GGCGAAADC#GDAGACGCAGTTGAPTBAA*APCAACCGTAGAAATACGTGCCGGTPCGAGTCCGGCCTTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCCGAAGTGGCGAAATCGGTAGACGCAGTTGATTCAAAATCAACCGTAGAAATACGTGCCGGTTCGAGTCCGGCCTTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Lys-TTT': {'seq': 'GGGTCGTTAGCTCAGDDGGDAGAGCAGTTGACTSTT6APCAATTG7XCGCAGGTPCGAATCCTGCACGACCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GGGTCGTTAGCTCAGTTGGTAGAGCAGTTGACTTTTAATCAATTGGTCGCAGGTTCGAATCCTGCACGACCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Met-CAT': {'seq': 'GGCTACG4AGCTCAGDD#GDDAGAGCACATCACTMAT6APGATGGG7XCACAGGTPCGAATCCCGTCGTAGCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GGCTACGTAGCTCAGTTGGTTAGAGCACATCACTCATAATGATGGGGTCACAGGTTCGAATCCCGTCGTAGCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Phe-GAA': {'seq': 'GCCCGGA4AGCTCAGDCGGDAGAGCAGGGGAPTGAA*APCCCCGT7XCCTTGGTPCGATTCCGAGTCCGGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAA',\n",
       "  'modified_idx': [36, 46],\n",
       "  'unmod_seq': 'GCCCGGATAGCTCAGTCGGTAGAGCAGGGGATTGAAAATCCCCGTGTCCTTGGTTCGATTCCGAGTCCGGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Pro-CGG': {'seq': 'CGGTGATTGGCGCAGCCTGGDAGCGCACTTCGTTCGGKACGAAGGG7TCGGAGGTPCGAATCCTCTATCACCGACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'CGGTGATTGGCGCAGCCTGGTAGCGCACTTCGTTCGGGACGAAGGGGTCGGAGGTTCGAATCCTCTATCACCGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Sec-TCA': {'seq': '_AAGATCG4CGTCTCCGGDGAGGCGGCTGGACTTCA+ATCCAGTTGGGGCCGC_GCGGTCCCGGGCAGGTPCGACTCCTGTGATCTT_GCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'NAAGATCGTCGTCTCCGGTGAGGCGGCTGGACTTCAAATCCAGTTGGGGCCGCNGCGGTCCCGGGCAGGTTCGACTCCTGTGATCTTNGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-TGA': {'seq': 'GGAAGTG4GGCCGAGC#GDDGAAGGCACCGGTBTVGA*AACCGGCGACCCGAAAGGGTTCCAGAGTPCGAATCTCTGCGCTTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGAAGTGTGGCCGAGCGGTTGAAGGCACCGGTCTTGAAAACCGGCGACCCGAAAGGGTTCCAGAGTTCGAATCTCTGCGCTTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-CGA': {'seq': 'GGAGAGATGCCGGAGC#GCDGAACGGACCGGTCTCGA*AACCGGAGTAGGGGCAACTCTACCGGGGGTPCAAATCCCCCTCTCTCCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGAGAGATGCCGGAGCGGCTGAACGGACCGGTCTCGAAAACCGGAGTAGGGGCAACTCTACCGGGGGTTCAAATCCCCCTCTCTCCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-GCT': {'seq': 'GGTGAGG4GGCCGAGAGGCDGAAGGCGCTCCC%TGCT6AGGGAGTATGCGGTCAAAAGCTGCATCCGGGGTPCGAATCCCCGCCTCACCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGAGGTGGCCGAGAGGCTGAAGGCGCTCCCCTGCTAAGGGAGTATGCGGTCAAAAGCTGCATCCGGGGTTCGAATCCCCGCCTCACCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Ser-GGA': {'seq': 'GGTGAGG4GTCCGAGT#GDDGAAGGAGCACGCCTGGAAAGPGTGTATACGGCAACGTATCGGGGGTPCGAATCCCCCCCTCACCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGTGAGGTGTCCGAGTGGTTGAAGGAGCACGCCTGGAAAGTGTGTATACGGCAACGTATCGGGGGTTCGAATCCCCCCCTCACCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Thr-GGT': {'seq': 'GCTGATATAGCTCAGDDGGDAGAGCGCACCCTTGGTEAGGGTGAG7TCGGCAGTPCGAATCTGCCTATCAGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCTGATATAGCTCAGTTGGTAGAGCGCACCCTTGGTAAGGGTGAGGTCGGCAGTTCGAATCTGCCTATCAGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Trp-CCA': {'seq': 'AGGGGCG4AGTTCAADDGGDAGAGCACCGGTBTCCA*AACCGGGT7TTGGGAGTPCGAGTCTCTCCGCCCCTGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCA',\n",
       "  'modified_idx': [36],\n",
       "  'unmod_seq': 'AGGGGCGTAGTTCAATTGGTAGAGCACCGGTCTCCAAAACCGGGTGTTGGGAGTTCGAGTCTCTCCGCCCCTGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Tyr-GTA': {'seq': 'GGTGGGG44CCCGAGC#GCCAAAGGGAGCAGACTQPA*APCTGCCGTCACAGACTTCGAAGGTPCGAATCCTTCCCCCACCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTA',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GGTGGGGTTCCCGAGCGGCCAAAGGGAGCAGACTGTAAATCTGCCGTCACAGACTTCGAAGGTTCGAATCCTTCCCCCACCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Val-TAC': {'seq': 'GGGTGAT4AGCTCAGCDGGGAGAGCACCTCCCTVAC=AGGAGGGG7TCGGCGGTPCGATCCCGTCATCACCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGGTGATTAGCTCAGCTGGGAGAGCACCTCCCTTACAAGGAGGGGGTCGGCGGTTCGATCCCGTCATCACCCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Val-GAC': {'seq': 'GCGTTCA4AGCTCAGDDGGDDAGAGCACCACCTTGACATGGTGGGG7XCGTTGGTPCGAGTCCAATTGAACGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAC',\n",
       "  'modified_idx': [47],\n",
       "  'unmod_seq': 'GCGTTCATAGCTCAGTTGGTTAGAGCACCACCTTGACATGGTGGGGGTCGTTGGTTCGAGTCCAATTGAACGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-iMet-CAT': {'seq': 'CGCGGGG4GGAGCAGCCTGGDAGCTCGTCGGGBTCATAACCCGAAGATCGTCGGTPCAAATCCGGCCCCCGCAACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'CGCGGGGTGGAGCAGCCTGGTAGCTCGTCGGGCTCATAACCCGAAGATCGTCGGTTCAAATCCGGCCCCCGCAACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ala-AGC': {'seq': 'GGGGAATTALCTCAAADGGDAGAGCRCTCGCJTIGCOP#CGAGAG7TAGCGGGAPCG\"TGCCCGCATTCTCCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGC',\n",
       "  'modified_idx': [25, 36, 57],\n",
       "  'unmod_seq': 'GGGGAATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGCGAGAGGTAGCGGGATCGATGCCCGCATTCTCCACCA',\n",
       "  'insosine_idx': [33]},\n",
       " 'Homo_sapiens_tRNA-Glu-CTC': {'seq': 'TCCCTGGTGLTCPAGTGGDPAGGATTCGGCGCTCTCACCGCCGCGGC??GGG\\\\PCGATTCCCGGTCAGGGAACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CTC',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'TCCCTGGTGGTCTAGTGGTTAGGATTCGGCGCTCTCACCGCCGCGGCCCGGGTTCGATTCCCGGTCAGGGAACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Phe-GAA': {'seq': 'GCCGAAATALCTC\"GDDGGGAGAGCRPPAGABT#AAWAPCTAAAG7DC?CTGGTPCG\"TCCCGGGTTTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GAA',\n",
       "  'modified_idx': [13, 25, 36, 57],\n",
       "  'unmod_seq': 'GCCGAAATAGCTCAGTTGGGAGAGCGTTAGACTGAAGATCTAAAGGTCCCTGGTTCGATCCCGGGTTTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gly-CCC': {'seq': 'GCGBCLCTGGTGPAGTGGDATCATGCAAGAJTCCCATZCTTGCGAC??GGGTPCG\"TTCCCGGGCGGCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCC',\n",
       "  'modified_idx': [55],\n",
       "  'unmod_seq': 'GCGCCGCTGGTGTAGTGGTATCATGCAAGATTCCCATTCTTGCGACCCGGGTTCGATTCCCGGGCGGCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gly-GCC': {'seq': 'GCAJTLGTGGTTCAGTGGDAGAATTCTCGCCTGCCA?GCGGGAGG???GGGTPCG\"TTCCCGGCCAATGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCC',\n",
       "  'modified_idx': [55],\n",
       "  'unmod_seq': 'GCATTGGTGGTTCAGTGGTAGAATTCTCGCCTGCCACGCGGGAGGCCCGGGTTCGATTCCCGGCCAATGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-His-GTG': {'seq': ';GCCGTGATCGTAPAGDGGDDAGTACTCTGCGPTGTGGCCGCAGCAA??TCGGTPCG\"ATCCGAGTCACGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTG',\n",
       "  'modified_idx': [57],\n",
       "  'unmod_seq': 'GGCCGTGATCGTATAGTGGTTAGTACTCTGCGTTGTGGCCGCAGCAACCTCGGTTCGAATCCGAGTCACGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Met-CAT': {'seq': \"GCCCTCTTAGCGCAGTGGG'AGCGCGTCAGTCTCATAATCTGAAGGTCCTGAGTTCGAGCCTCAGAGAGGGCA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [19],\n",
       "  'unmod_seq': 'GCCCTCTTAGCGCAGTGGGCAGCGCGTCAGTCTCATAATCTGAAGGTCCTGAGTTCGAGCCTCAGAGAGGGCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Asn-GTT': {'seq': 'GTCTCTGTKLCGCAADCGGDXAGCGCRPPCGGCTQTT6ACCGAAAG7DTGGTGG.PCG\"GCCCACCCAGGGACGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTT',\n",
       "  'modified_idx': [8, 20, 26, 58],\n",
       "  'unmod_seq': 'GTCTCTGTGGCGCAATCGGTTAGCGCGTTCGGCTGTTAACCGAAAGGTTGGTGGNTCGAGCCCACCCAGGGACGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Gln-TTG': {'seq': 'GGTCCCATKGTGPAAT#GDDAGCACTCPGGABTNTGAAPCCAGCGAT??GAGPPCA\"ATCTCGGTGGGACCTCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TTG',\n",
       "  'modified_idx': [8, 56],\n",
       "  'unmod_seq': 'GGTCCCATGGTGTAATGGTTAGCACTCTGGACTTTGAATCCAGCGATCCGAGTTCAAATCTCGGTGGGACCTCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-TGA': {'seq': \"GCAGCGATGGCCGAGTGGTTAAGGCGTTGGA'TTGAAATCCAATGG'GTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GCAGCGATGGCCGAGTGGTTAAGGCGTTGGACTTGAAATCCAATGGCGTCTCCCCGCGCAGGTTCGAACCCTGCTCGCTGCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-iMet-CAT': {'seq': 'AGCAGAGTKLCGCAGCGGAAGCGTLCTGGGCCCAT6ACCCAGAG7D?GATGGATCG\"AACCATCCTCTGCTACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAT',\n",
       "  'modified_idx': [8, 56],\n",
       "  'unmod_seq': 'AGCAGAGTGGCGCAGCGGAAGCGTGCTGGGCCCATAACCCAGAGGTCGATGGATCGAAACCATCCTCTGCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Tyr-GTA': {'seq': 'CCTTCLATALCTCAGCDGGXAGAGCRRAGGACT9PAKA]CCTTAG7D?GCTGGTPCG\"TTCCGGCTCGAAGGACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GTA',\n",
       "  'modified_idx': [19, 25, 26, 36, 57],\n",
       "  'unmod_seq': 'CCTTCGATAGCTCAGCTGGTAGAGCGGAGGACTGTAGATCCTTAGGTCGCTGGTTCGATTCCGGCTCGAAGGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Sec-TCA': {'seq': 'GCCCGGAT_CCTCAGTGGTCTGGGGTGCAGGCTNCA+ACCTGTAGCTGTCTAGCGACAGAGTGGTPCA\"TTCCA_TTTCGGGCGCCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCA',\n",
       "  'modified_idx': [68],\n",
       "  'unmod_seq': 'GCCCGGATNCCTCAGTGGTCTGGGGTGCAGGCTTCAAACCTGTAGCTGTCTAGCGACAGAGTGGTTCAATTCCANTTTCGGGCGCCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ile-GAT': {'seq': 'AGAAATATKTCTGATAAAAGARPPACTTTGAT6GAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'GAT',\n",
       "  'modified_idx': [8, 21],\n",
       "  'unmod_seq': 'AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGGAGCTTAAACCCCCTTATTTCTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-TAA': {'seq': 'GTTAAGATKLCAGAGCCCGGDAATCGCAPAAAACTÊAAAACTTTACAGT?AGAGGTPCA\"TTCCTCTTCTTAACACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TAA',\n",
       "  'modified_idx': [8, 59],\n",
       "  'unmod_seq': 'GTTAAGATGGCAGAGCCCGGTAATCGCATAAAACTTAAAACTTTACAGTCAGAGGTTCAATTCCTCTTCTTAACACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-GCT': {'seq': \"GACGAGGTGGCCGAGTGGTTAAGGCGATGGACTGCTEATCCATTGT'CTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'GCT',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GACGAGGTGGCCGAGTGGTTAAGGCGATGGACTGCTAATCCATTGTCCTCTGCACGCGTGGGTTCGAATCCCACCCTCGTCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Lys-TTT': {'seq': 'CACTGTAA\"LCTAACTTAGCAPPAACCT∃TT6AGTTAAAGATTAAGAGAACCAA_CTCTTTACAGTGACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TTT',\n",
       "  'modified_idx': [8],\n",
       "  'unmod_seq': 'CACTGTAAAGCTAACTTAGCATTAACCTTTTAAGTTAAAGATTAAGAGAACCAANCTCTTTACAGTGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Pro-TGG': {'seq': 'CAGAGAAT\"GTTTAAATTAGAATCPPAGCPTTGGKPGCTAATGGTGGAGTPAAAGACTTTTTCTCTGACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TGG',\n",
       "  'modified_idx': [8, 34],\n",
       "  'unmod_seq': 'CAGAGAATAGTTTAAATTAGAATCTTAGCTTTGGGTGCTAATGGTGGAGTTAAAGACTTTTTCTCTGACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Asp-GTC': {'seq': 'AAGGTATT\"LAAAAACCATTTCAPAACTTTQTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'GTC',\n",
       "  'modified_idx': [8],\n",
       "  'unmod_seq': 'AAGGTATTAGAAAAACCATTTCATAACTTTGTCAAAGTTAAATTATAGGCTAAATCCTATATATCTTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-TAG': {'seq': 'ACTTTTAA\"LGATAACAGCTATCCAPPGGPCTTAGKCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "  'type': 'mito',\n",
       "  'anticodon': 'TAG',\n",
       "  'modified_idx': [8, 35],\n",
       "  'unmod_seq': 'ACTTTTAAAGGATAACAGCTATCCATTGGTCTTAGGCCCCAAAAATTTTGGTGCAACTCCAAATAAAAGTACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Leu-TAG': {'seq': 'GCGGGAGTGGCGAAADD#GTAGACGCACCAGATTVAGKPTCTGGCGCCGCAAGGTGTGCGAGTPCAAGTCTCGCCTCCCGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TAG',\n",
       "  'modified_idx': [37],\n",
       "  'unmod_seq': 'GCGGGAGTGGCGAAATTGGTAGACGCACCAGATTTAGGTTCTGGCGCCGCAAGGTGTGCGAGTTCAAGTCTCGCCTCCCGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Escherichia_coli_tRNA-Thr-TGT': {'seq': 'GCCGACTTAGCTCAGTAGGDAGAGCAACTGABTVGT6ATCAGTAG7TCACCAGTPCGATTCCGGTAGTCGGCACCA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GCCGACTTAGCTCAGTAGGTAGAGCAACTGACTTGTAATCAGTAGGTCACCAGTTCGATTCCGGTAGTCGGCACCA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-CAG': {'seq': \"GTCAGGATGGCCGAGCGGTCTAAGGCGCTGCGTTCAGGTCGCAGTC'CCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAG',\n",
       "  'modified_idx': [46],\n",
       "  'unmod_seq': 'GTCAGGATGGCCGAGCGGTCTAAGGCGCTGCGTTCAGGTCGCAGTCCCCCCTGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Leu-CAA': {'seq': 'GTCAGGATGGCCGAGTGGTCTAAGGCGCCAGAC?CAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CAA',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GTCAGGATGGCCGAGTGGTCTAAGGCGCCAGACCCAAGTTCTGGTCTCCAATGGAGGCGTGGGTTCGAATCCCACTTCTGACA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Arg-CCT': {'seq': \"GCCCCAGTGGCCTAATGGATAAGGCACTGGC'TCCTAAGCCAGGGATTGTGGGTTCGAGTCCCACCTGGGGTA\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CCT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GCCCCAGTGGCCTAATGGATAAGGCACTGGCCTCCTAAGCCAGGGATTGTGGGTTCGAGTCCCACCTGGGGTA',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Arg-TCT': {'seq': \"GGCTCCGTGGCGCAATGGATAGCGCATTGGA'TTCTAATTCAAAGGTTCCGGGTTCGAGTCCCGGCGGAGTCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TCT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCTCCGTGGCGCAATGGATAGCGCATTGGACTTCTAATTCAAAGGTTCCGGGTTCGAGTCCCGGCGGAGTCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-AGA': {'seq': \"GTAGTCGTGGCCGAGTGGTTAAGGCGATGGA'TAGAAATCCATTGG'GTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GTAGTCGTGGCCGAGTGGTTAAGGCGATGGACTAGAAATCCATTGGCGTTTCCCCGCGCAGGTTCGAATCCTGCCGACTACG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Ser-CGA': {'seq': \"GCTGTGATGGCCGAGTGGTTAAGGCGTTGGA'TCGAAATCCAATGG'GTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCG\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGA',\n",
       "  'modified_idx': [31, 46],\n",
       "  'unmod_seq': 'GCTGTGATGGCCGAGTGGTTAAGGCGTTGGACTCGAAATCCAATGGCGTCTCCCCGCGCAGGTTCGAATCCTGCTCACAGCG',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-AGT': {'seq': \"GGCGCCGTGGCTTAGTTGGTTAAAGCGCCTG'CTAGTAAACAGGAGATCCTGGGTTCGAATCCCAGCGGTGCCT\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'AGT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCGCCGTGGCTTAGTTGGTTAAAGCGCCTGCCTAGTAAACAGGAGATCCTGGGTTCGAATCCCAGCGGTGCCT',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-CGT': {'seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGTCTCGTAAACAGGAGATCCTGGGTTCGACTCCCAGTGGGGCCT',\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'CGT',\n",
       "  'modified_idx': [],\n",
       "  'unmod_seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGTCTCGTAAACAGGAGATCCTGGGTTCGACTCCCAGTGGGGCCT',\n",
       "  'insosine_idx': []},\n",
       " 'Homo_sapiens_tRNA-Thr-TGT': {'seq': \"GGCTCTATGGCTTAGTTGGTTAAAGCGCCTG'CTTGTAAACAGGAGATCCTGGGTTCGAATCCCAGTAGAGCCT\",\n",
       "  'type': 'cyto',\n",
       "  'anticodon': 'TGT',\n",
       "  'modified_idx': [31],\n",
       "  'unmod_seq': 'GGCTCTATGGCTTAGTTGGTTAAAGCGCCTGCCTTGTAAACAGGAGATCCTGGGTTCGAATCCCAGTAGAGCCT',\n",
       "  'insosine_idx': []}}"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trna_obj.modomics_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML\n",
    "import re, copy, sys, os, shutil, subprocess, logging, glob\n",
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "import pandas as pd\n",
    "import requests\n",
    "from requests.models import HTTPError\n",
    "from mimseq.ssAlign import aligntRNA, extraCCA, tRNAclassifier, tRNAclassifier_nogaps, getAnticodon, clusterAnticodon\n",
    "\n",
    "def modsToSNPIndex(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, experiment_name, out_dir, double_cca, snp_tolerance = False, cluster = False, cluster_id = 0.95, posttrans_mod_off = False, pretrnas = False, local_mod = False):\n",
    "# Builds SNP index needed for GSNAP based on modificaiton data for each tRNA and clusters tRNAs\n",
    "\n",
    "    nomatch_count = 0\n",
    "    match_count = 0\n",
    "    total_snps = 0\n",
    "    total_inosines = 0\n",
    "    snp_records = list()\n",
    "    seq_records = defaultdict()\n",
    "    anticodon_list = list()\n",
    "    tRNAbed = open(out_dir + experiment_name + \"_maturetRNA.bed\",\"w\")\n",
    "    # generate modomics_dict and tRNA_dict\n",
    "#    tRNA_dict, modomics_dict, species = tRNAparser(gtRNAdb, tRNAscan_out, mitotRNAs, modifications_table, posttrans_mod_off, double_cca, pretrnas, local_mod)\n",
    "    temp_dir = out_dir + \"/tmp/\"\n",
    "\n",
    "    try:\n",
    "        os.mkdir(temp_dir)\n",
    "    except FileExistsError:\n",
    "        log.warning(\"Temp folder present - previous run interrupted? Overwriting old temp files...\\n\")\n",
    "\n",
    "    ###################################################\n",
    "    ## Main code for matching and SNP index building ##\n",
    "    ###################################################\n",
    "\n",
    "    # remove spurious CCA seen in some sequences (i.e. genomically encoded or artifact from gtRNAdb or tRNAScan-SE prediction)\n",
    "    # to do this, remove previously added CCA (intronRemover()) that falls out of canonical structure using ssAlign module\n",
    "\n",
    "    with open(out_dir + 'temptRNAseqs.fa', 'w') as tempSeqs:\n",
    "        for seq in tRNA_dict:\n",
    "            tempSeqs.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "\n",
    "    aligntRNA(tempSeqs.name, out_dir)\n",
    "    extra_cca = extraCCA()\n",
    "\n",
    "    for record in extra_cca:\n",
    "        tRNA_dict[record]['sequence'] = tRNA_dict[record]['sequence'][:-3]\n",
    "\n",
    "    os.remove(tempSeqs.name)\n",
    "\n",
    "    # match each sequence in tRNA_dict to value in modomics_dict using BLAST\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "##### Up to here constitutes an object #####\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    log.info(\"\\n+------------------------+ \\\n",
    "        \\n| Beginning SNP indexing |\\\n",
    "        \\n+------------------------+\")    \n",
    "\n",
    "    for seq in tRNA_dict:\n",
    "        # Initialise list of modified sites for each tRNA\n",
    "        tRNA_dict[seq]['modified'] = []\n",
    "        tRNA_dict[seq]['InosinePos'] = []\n",
    "        temp_tRNAFasta = open(temp_dir + seq + \".fa\",\"w\")\n",
    "        temp_tRNAFasta.write(\">\" + seq + \"\\n\" + tRNA_dict[seq]['sequence'] + \"\\n\")\n",
    "        temp_tRNAFasta.close()\n",
    "        anticodon = re.search('.*tR(NA|X)-.*?-(.*?)-', seq).group(2)\n",
    "        tRNA_dict[seq]['anticodon'] = anticodon\n",
    "        if not anticodon in anticodon_list:\n",
    "            anticodon_list.append(anticodon)\n",
    "        # find initial possible matches to modomics where anticodons match and types are the same (here regex is used to match anticodons with \".\" in modomics to all possible matching sequences from input tRNAs)\n",
    "        match = {k:v for k,v in modomics_dict.items() if re.match(\"^\" + v['anticodon'] + \"+$\", anticodon) and tRNA_dict[seq]['type'] == v['type']}\n",
    "        if len(match) >= 1:\n",
    "            temp_matchFasta = open(temp_dir + \"modomicsMatch.fasta\",\"w\")\n",
    "            for i in match:    \n",
    "                temp_matchFasta.write(\">\" + i + \"\\n\" + match[i]['unmod_sequence'] + \"\\n\")\n",
    "            temp_matchFasta.close()\n",
    "\n",
    "            #blast\n",
    "            blastn_cline = NcbiblastnCommandline(query = temp_tRNAFasta.name, subject = temp_matchFasta.name, task = 'blastn-short', out = temp_dir + \"blast_temp.xml\", outfmt = 5)\n",
    "            blastn_cline()\n",
    "\n",
    "            #parse XML result and store hit with highest bitscore    \n",
    "            blast_record = NCBIXML.read(open(temp_dir + \"blast_temp.xml\",\"r\"))\n",
    "            maxbit = 0\n",
    "            tophit = ''\n",
    "            for alignment in blast_record.alignments:\n",
    "                for hsp in alignment.hsps:\n",
    "                    if (hsp.bits > maxbit) and (hsp.align_length / alignment.length == 1) and (hsp.identities / alignment.length == 1):\n",
    "                        maxbit = hsp.bits\n",
    "                        tophit = alignment.title.split(' ')[0]\n",
    "            \n",
    "            # return list of all modified positions for the match as long as there is only 1, add to tRNA_dict\n",
    "            if tophit:\n",
    "                match_count += 1\n",
    "                tRNA_dict[seq]['modified'] = match[tophit]['modified']\n",
    "                tRNA_dict[seq]['InosinePos'] = match[tophit]['InosinePos']\n",
    "            elif len(tophit) == 0:\n",
    "                nomatch_count += 1\n",
    "        if len(match) == 0:\n",
    "            nomatch_count += 1\n",
    "\n",
    "        # Build seqrecord list for writing\n",
    "        seq_records[str(seq)] = SeqRecord(Seq(tRNA_dict[seq]['sequence'].upper()), id = str(seq))\n",
    "\n",
    "        tRNAbed.write(seq + \"\\t0\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t\" + seq + \"\\t1000\\t+\\n\" )\n",
    "\n",
    "    tRNAbed.close()\n",
    "\n",
    "    log.info(\"{} total tRNA gene sequences (undetermined and nmt sequences excluded)\".format(len(tRNA_dict)))\n",
    "    log.info(\"{} sequences with a match to Modomics dataset\".format(match_count))\n",
    "\n",
    "    with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "        SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "    # if clustering is not activated then write full gff and report on total SNPs written \n",
    "    if not cluster:\n",
    "        coverage_bed = tRNAbed.name\n",
    "        mod_lists = defaultdict(list)\n",
    "        Inosine_lists = defaultdict(list)\n",
    "        with open(out_dir + experiment_name + \"_tRNA.gff\",\"w\") as tRNAgff, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:    \n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for seq in tRNA_dict:\n",
    "                mod_lists[seq] = tRNA_dict[seq]['modified']\n",
    "                Inosine_lists[seq] = tRNA_dict[seq]['InosinePos']\n",
    "                tRNAgff.write(seq + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[seq]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + seq + \"'\\n\")\n",
    "                isoacceptor_group = '-'.join(seq.split(\"-\")[:-2])\n",
    "                isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        # generate Stockholm alignment file for all tRNA transcripts and parse additional mods file\n",
    "        aligntRNA(temptRNATranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "        # add additional SNPs from extra file to list of modified positions, and ensure non-redundancy with set()\n",
    "        # index SNPs\n",
    "        # Format for SNP index (space separated):\n",
    "        # >snpID chromosomeName:position(1-based) RefMod\n",
    "        # e.g. >rs111 Homo_sapiens_nmt_tRNA-Leu-TAA-1-1_exp0:29 GN\n",
    "        for seq in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[seq]['modified'] = list(set(tRNA_dict[seq]['modified'] + additionalMods_sub[seq]['mods']))\n",
    "                mod_lists[seq] = list(set(mod_lists[seq] + additionalMods_sub[seq]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[seq])\n",
    "\n",
    "            # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "            # Position is 1-based for iit_store i.e. pos + 1\n",
    "            for (index, pos) in enumerate(mod_lists[seq]):\n",
    "                #if \"Gln-TTG\" in seq and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + tRNA_dict[seq]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for seq in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == seq and tRNA_dict[seq]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[seq]['InosinePos'] = list(set(tRNA_dict[seq]['InosinePos'] + additionalInosines_sub[seq]['InosinePos']))\n",
    "                Inosine_lists[seq] = list(set(Inosine_lists[seq] + additionalInosines_sub[seq]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[seq])\n",
    "            total_snps += len(Inosine_lists[seq])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[seq]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + seq + \"_snp\" + str(index) + \" \" + seq + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions only if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for seq in Inosine_lists:\n",
    "                for pos in Inosine_lists[seq]:\n",
    "                    seq_records[seq].seq = seq_records[seq].seq[0:pos] + \"G\" + seq_records[seq].seq[pos+1:]\n",
    "\n",
    "        with open(str(out_dir + experiment_name + '_tRNATranscripts.fa'), \"w\") as temptRNATranscripts:\n",
    "            SeqIO.write(seq_records.values(), temptRNATranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False\n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))\n",
    "        # empty mismatch dict to avoid error when returning it from this function\n",
    "        mismatch_dict = defaultdict(list)\n",
    "        insert_dict = defaultdict(dd_list)\n",
    "        del_dict = defaultdict(dd_list)\n",
    "        mod_lists = {tRNA:list() for tRNA in tRNA_dict.keys()}\n",
    "        Inosine_lists = {tRNA:data['InosinePos'] for tRNA, data in tRNA_dict.items()}\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list)\n",
    "        cluster_dict = defaultdict(list)\n",
    "\n",
    "    ##########################\n",
    "    # Cluster tRNA sequences #\n",
    "    ##########################\n",
    "\n",
    "    elif cluster:\n",
    "\n",
    "        log.info(\"**** Clustering tRNA sequences ****\")\n",
    "        log.info(\"Clustering tRNA sequences by {:.0%} similarity...\".format(cluster_id))\n",
    "        # dictionary of final centroid sequences\n",
    "        final_centroids = defaultdict()\n",
    "        # get dictionary of sequences for each anticodon and write to fastas\n",
    "        for anticodon in anticodon_list:\n",
    "            seq_set = {k:{'sequence':v['sequence'],'modified':v['modified']} for k,v in tRNA_dict.items() if v['anticodon'] == anticodon}\n",
    "            with open(temp_dir + anticodon + \"_allseqs.fa\",\"w\") as anticodon_seqs:\n",
    "                for sequence in seq_set:\n",
    "                    anticodon_seqs.write(\">\" + sequence + \"\\n\" + seq_set[sequence]['sequence'] + \"\\n\")\n",
    "            # run usearch on each anticodon sequence fatsa to cluster\n",
    "            cluster_cmd = [\"usearch\", \"-cluster_fast\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]\n",
    "            #cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_allseqs.fa\", \"-id\", str(cluster_id), \"--sortedby\", \"other\" ,\"-sizeout\" ,\"-centroids\", temp_dir + anticodon + \"_centroids.fa\", \"-uc\", temp_dir + anticodon + \"_clusters.uc\"]            #cluster_cmd = \"usearch -cluster_fast \" + temp_dir + anticodon + \"_allseqs.fa -sort length -id \" + str(cluster_id) + \" -centroids \" + temp_dir + anticodon + \"_centroids.fa -uc \" + temp_dir + anticodon + \"_clusters.uc &> /dev/null\"\n",
    "            subprocess.check_call(cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # sort clusters by size (i.e. number of members in cluster)\n",
    "            sort_cmd = [\"usearch\", \"-sortbysize\", temp_dir + anticodon + \"_centroids.fa\", \"-fastaout\", temp_dir + anticodon + \"_centroids_sort.fa\"]\n",
    "            subprocess.check_call(sort_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "            # recluster based on sorted by size clusters\n",
    "            final_cluster_cmd = [\"usearch\", \"-cluster_smallmem\", temp_dir + anticodon + \"_centroids_sort.fa\", \"-sortedby\", \"size\", \"-id\", str(cluster_id), \"-centroids\", temp_dir + anticodon + \"_centroidsFinal.fa\"]\n",
    "            subprocess.check_call(final_cluster_cmd, stdout = subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "        # combine centroids files into one file\n",
    "        for filename in glob.glob(temp_dir + \"*_centroidsFinal.fa\"):\n",
    "            with open(filename, \"r\") as fileh:\n",
    "                with open(temp_dir + \"all_centroids.fa\", \"a\") as outh:\n",
    "                    outh.write(fileh.read())\n",
    "        centroids = SeqIO.parse(temp_dir + \"all_centroids.fa\", \"fasta\")\n",
    "        for centroid in centroids:\n",
    "            centroid.id = centroid.id.split(\";\")[0]\n",
    "            final_centroids[centroid.id] = SeqRecord(Seq(str(centroid.seq).upper()), id = centroid.id) \n",
    "\n",
    "        # read cluster files, get nonredudant set of mod positions of all members of a cluster, create snp_records for writing SNP index\n",
    "        cluster_pathlist = Path(temp_dir).glob(\"**/*_clusters.uc\")\n",
    "        mod_lists = defaultdict(list) # stores non-redundant sets of mismatches and mod positions for clusters\n",
    "        Inosine_lists = defaultdict(list) # stores positions of inosines for clusters\n",
    "        snp_records = list()\n",
    "        cluster_dict = defaultdict(list) # info about clusters and members\n",
    "        mismatch_dict = defaultdict(list) # dictionary of mismatches only (not mod positions - required for exclusion from misincorporation analysis in mmQuant)\n",
    "        insert_dict = defaultdict(dd_list) # dictionary of insertions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        del_dict = defaultdict(dd_list) # dictionary of deletions in cluster parents - these are not recorded as mismatches but are needed in order to split clusters into isodecoders\n",
    "        cluster_perPos_mismatchMembers = defaultdict(dd_list) # for each cluster, list of members that mismatch at each position corresonding to mismatch dict - used for splitting clusters into isodecoders (splitClusters.py)\n",
    "        cluster_num = 0\n",
    "        total_snps = 0\n",
    "        total_inosines = 0\n",
    "        clusterbed = open(out_dir + experiment_name + \"_clusters.bed\",\"w\")\n",
    "        coverage_bed = clusterbed.name\n",
    "        clustergff = open(out_dir + experiment_name + \"_tRNA.gff\",\"w\")\n",
    "        for path in cluster_pathlist:\n",
    "            with open(str(path),\"r\") as cluster_file:\n",
    "                for line in cluster_file:\n",
    "                    line = line.strip()\n",
    "\n",
    "                    # Handle cluster centroids and initialise modified positions list\n",
    "                    if line.split(\"\\t\")[0] == \"S\":\n",
    "                        cluster_num += 1\n",
    "                        cluster_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        mod_lists[cluster_name] = tRNA_dict[cluster_name][\"modified\"]\n",
    "                        Inosine_lists[cluster_name] = tRNA_dict[cluster_name]['InosinePos']\n",
    "                        clusterbed.write(cluster_name + \"\\t0\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t\" + cluster_name + \"\\t1000\\t+\\n\" )\n",
    "                        clustergff.write(cluster_name + \"\\ttRNAseq\\texon\\t1\\t\" + str(len(tRNA_dict[cluster_name]['sequence'])) + \"\\t.\\t+\\t0\\tgene_id '\" + cluster_name + \"'\\n\")\n",
    "                        cluster_dict[cluster_name].append(cluster_name)\n",
    "                \n",
    "                    # Handle members of clusters\n",
    "                    elif line.split(\"\\t\")[0] == \"H\":\n",
    "                        member_name = line.split(\"\\t\")[8].split(\";\")[0]\n",
    "                        cluster_name = line.split(\"\\t\")[9].split(\";\")[0]\n",
    "                        compr_aln = line.split(\"\\t\")[7]\n",
    "                        # if member of cluster is 100% identical (i.e. \"=\" in 8th column of cluster file)\n",
    "                        if compr_aln == \"=\":\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + tRNA_dict[member_name][\"modified\"]))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + tRNA_dict[member_name][\"InosinePos\"]))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "                        \n",
    "                        # if there are insertions or deletions in the centroid, edit member or centroid sequences to ignore these positions\n",
    "                        # and edit modified positions list in order to make non-redundant positions list, similar to next else statement\n",
    "                        elif re.search(\"[ID]\", compr_aln):\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            pos = 0\n",
    "                            adjust_pos_del = 0\n",
    "                            adjust_pos_ins = 0\n",
    "                            insertion_pos = list()\n",
    "                            deletion_pos = list()\n",
    "                            aln_list = re.split('(.*?[A-Z])', compr_aln)\n",
    "                            aln_list = list(filter(None, aln_list))\n",
    "\n",
    "                            for phrase in aln_list:\n",
    "                                \n",
    "                                if (\"M\" in phrase) and (phrase.split(\"M\")[0] != \"\"):\n",
    "                                    pos += int(phrase.split(\"M\")[0])\n",
    "                                elif (\"M\" in phrase) and (phrase.split(\"M\")[0] == \"\"):\n",
    "                                    pos += 1\n",
    "\n",
    "                                if (\"I\" in phrase) and (phrase.split(\"I\")[0] != \"\"):\n",
    "                                    insert_len = int(phrase.split(\"I\")[0])\n",
    "\n",
    "                                    for i in range(insert_len):\n",
    "                                        insertion_pos.append(pos+i)\n",
    "                                        #pos += 1\n",
    "                                elif (\"I\" in phrase) and (phrase.split(\"I\")[0] == \"\"):\n",
    "                                    insertion_pos.append(pos)\n",
    "                                    #pos += 1\n",
    "\n",
    "                                if (\"D\" in phrase) and (phrase.split(\"D\")[0] != \"\"):\n",
    "                                    delete_len = int(phrase.split(\"D\")[0])\n",
    "\n",
    "                                    for i in range(delete_len):\n",
    "                                        deletion_pos.append(pos)\n",
    "                                        pos += 1\n",
    "                                elif (\"D\" in phrase) and (phrase.split(\"D\")[0] == \"\"):\n",
    "                                    deletion_pos.append(pos)\n",
    "                                    pos += 1\n",
    "\n",
    "                            for delete in deletion_pos:\n",
    "                                adjust_pos_len = 0\n",
    "                                for insert in insertion_pos:\n",
    "                                    if insert < delete:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                new_delete = delete + adjust_pos_del + adjust_pos_len\n",
    "                                member_seq = member_seq[ :new_delete] + member_seq[new_delete+1: ]\n",
    "                                del_dict[cluster_name][new_delete].append(member_name)\n",
    "                                adjust_pos_del -= 1\n",
    "\n",
    "                            for index, insert in enumerate(insertion_pos):\n",
    "                                adjust_pos_len = 0\n",
    "                                for delete in deletion_pos:\n",
    "                                    if delete < insert:\n",
    "                                        adjust_pos_len -= 1\n",
    "                                #if index != 0:\n",
    "                                #    if insert == insertion_pos[index-1] + 1:\n",
    "                                #        adjust_pos_ins -= 1\n",
    "                                new_insert = insert + adjust_pos_len + adjust_pos_ins\n",
    "                                member_seq = member_seq[ :new_insert] + cluster_seq[insert] + member_seq[new_insert: ]\n",
    "                                #insert_dict[cluster_name][new_insert-1].append(member_name)\n",
    "                                insert_dict[cluster_name][new_insert].append(member_name)\n",
    "                                #cluster_seq = cluster_seq[ :new_insert] + cluster_seq[new_insert+1: ]\n",
    "\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "                        # handle members that are not exact sequence matches but have no indels either\n",
    "                        # find mismatches and build non-redundant set\n",
    "                        else:\n",
    "                            cluster_seq = tRNA_dict[cluster_name][\"sequence\"]\n",
    "                            member_seq = tRNA_dict[member_name][\"sequence\"]\n",
    "                            mismatches = [i for i in range(len(member_seq)) if member_seq[i].upper() != cluster_seq[i].upper()]\n",
    "                            mismatch_dict[cluster_name] = list(set(mismatch_dict[cluster_name] + mismatches))\n",
    "                            for mismatch in mismatches:\n",
    "                                cluster_perPos_mismatchMembers[cluster_name][mismatch].append(member_name)\n",
    "                            member_mods = list(set(tRNA_dict[member_name][\"modified\"] + mismatches))\n",
    "                            member_Inosines = tRNA_dict[member_name][\"InosinePos\"]\n",
    "                            mod_lists[cluster_name] = list(set(mod_lists[cluster_name] + member_mods))\n",
    "                            Inosine_lists[cluster_name] = list(set(Inosine_lists[cluster_name] + member_Inosines))\n",
    "                            cluster_dict[cluster_name].append(member_name)\n",
    "\n",
    "        clusterbed.close()\n",
    "\n",
    "        # Write cluster information to tsv and get number of unique sequences per cluster (i.e. isodecoders) for read count splitting\n",
    "        with open(out_dir + experiment_name + \"clusterInfo.txt\",\"w\") as clusterInfo, open(out_dir + experiment_name + \"isoacceptorInfo.txt\",\"w\") as isoacceptorInfo:\n",
    "            isoacceptor_dict = defaultdict(int)\n",
    "            clusterInfo.write(\"tRNA\\tcluster_num\\tcluster_size\\tparent\\n\")\n",
    "            isoacceptorInfo.write(\"Isoacceptor\\tsize\\n\")\n",
    "            for key, value in cluster_dict.items():\n",
    "                cluster_num = list(cluster_dict.keys()).index(key)\n",
    "                for member in value:\n",
    "                    clusterInfo.write(\"{}\\t{}\\t{}\\t{}\\n\".format(member, cluster_num, len(value), key))\n",
    "                    isoacceptor_group = '-'.join(member.split(\"-\")[:-2])\n",
    "                    isoacceptor_dict[isoacceptor_group] += 1\n",
    "            for key, value in isoacceptor_dict.items():\n",
    "                isoacceptorInfo.write(key + \"\\t\" + str(value) + \"\\n\")\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        # generate Stockholm alignment file for cluster transcripts and process additional mods file\n",
    "        aligntRNA(clusterTranscripts.name, out_dir)\n",
    "        additionalMods, additionalInosines = additionalModsParser(species, out_dir)\n",
    "\n",
    "        log.info(\"{} clusters created from {} tRNA sequences\".format(len(cluster_dict),len(tRNA_dict)))\n",
    "\n",
    "        # update mod_lists with additional mods and write SNP index\n",
    "        for cluster in mod_lists:\n",
    "            additionalMods_sub = {k:v for k, v in additionalMods.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalMods_sub:\n",
    "                tRNA_dict[cluster]['modified'] = list(set(tRNA_dict[cluster]['modified'] + additionalMods_sub[cluster]['mods']))\n",
    "                mod_lists[cluster] = list(set(mod_lists[cluster] + additionalMods_sub[cluster]['mods']))\n",
    "\n",
    "            total_snps += len(mod_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(mod_lists[cluster]):\n",
    "                # Build snp_records as before but with cluster names and non-redundant sets of modifications\n",
    "                # Position is 1-based for iit_store i.e. pos + 1\n",
    "                #if \"Gln-TTG\" in cluster and pos + 1 == 34:\n",
    "                #    snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"C\")\n",
    "                #else:\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + tRNA_dict[cluster]['sequence'][pos].upper() + \"N\")\n",
    "\n",
    "        for cluster in Inosine_lists:\n",
    "            additionalInosines_sub = {k:v for k, v in additionalInosines.items() if k == cluster and tRNA_dict[cluster]['species'] in v['species']}\n",
    "            if additionalInosines_sub:\n",
    "                tRNA_dict[cluster]['InosinePos'] = list(set(tRNA_dict[cluster]['InosinePos'] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "                Inosine_lists[cluster] = list(set(Inosine_lists[cluster] + additionalInosines_sub[cluster]['InosinePos']))\n",
    "\n",
    "            total_inosines += len(Inosine_lists[cluster])\n",
    "            total_snps += len(Inosine_lists[cluster])\n",
    "\n",
    "            for (index, pos) in enumerate(Inosine_lists[cluster]):\n",
    "                # Add inosines to snp index (in addition to changing ref seqence - see below)\n",
    "                # Ensure only \"A\" SNPs are tolerated. That is, a G in the reference allows inosines while an A in the snp index allows unmodified reads\n",
    "                snp_records.append(\">\" + cluster + \"_snp\" + str(index) + \" \" + cluster + \":\" + str(pos + 1) + \" \" + \"GA\")\n",
    "\n",
    "\n",
    "        # edit ref seqs A to G at inosine positions if snp_tolerance is enabled\n",
    "        if snp_tolerance:\n",
    "            for cluster in Inosine_lists:\n",
    "                for pos in Inosine_lists[cluster]:\n",
    "                    final_centroids[cluster].seq = final_centroids[cluster].seq[0:pos] + \"G\" + final_centroids[cluster].seq[pos+1:]\n",
    "\n",
    "        Inosine_clusters = [cluster for cluster, inosines in Inosine_lists.items() if len(inosines) > 0]\n",
    "\n",
    "        # rewrite edited cluster transcripts\n",
    "        with open(str(out_dir + experiment_name + '_clusterTranscripts.fa'), \"w\") as clusterTranscripts:\n",
    "            SeqIO.write(final_centroids.values(), clusterTranscripts, \"fasta\")\n",
    "\n",
    "        if total_snps == 0:\n",
    "            snp_tolerance = False        \n",
    "\n",
    "        log.info(\"{:,} modifications written to SNP index\".format(total_snps))\n",
    "        log.info(\"{:,} A to G replacements in reference sequences for inosine modifications\".format(total_inosines))        \n",
    "    \n",
    "    # write outputs for indexing \n",
    "    with open(out_dir + experiment_name + \"_modificationSNPs.txt\", \"w\") as snp_file:\n",
    "        for item in snp_records:\n",
    "            snp_file.write('{}\\n'.format(item))\n",
    "    \n",
    "    shutil.rmtree(temp_dir)\n",
    "    \n",
    "    # Return coverage_bed (either tRNAbed or clusterbed depending on --cluster) for coverage calculation method\n",
    "    return(coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "coverage_bed, snp_tolerance, mismatch_dict, insert_dict, del_dict, mod_lists, Inosine_lists, Inosine_clusters, tRNA_dict, cluster_dict, cluster_perPos_mismatchMembers = modsToSNPIndex(trnas, trnaout, mito_trnas, modifications, name, out, double_cca, snp_tolerance, cluster, cluster_id, posttrans, pretrnas, local_mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Namespace(cca=True, cluster=True, cluster_id=0.95, control_cond='HEK293T', cov_diff=0.5, double_cca=False, keep_temp=False, local_mod=False, max_multi=1, min_cov=2000.0, misinc_thresh=0.1, mismatches=0.1, mito='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19-mitotRNAs.fa', name='hg19_test', out='hg19_HEK239vsK562', posttrans=False, pretrnas=False, remap=True, remap_mismatches=0.075, sampledata='sampleData_HEKvsK562.txt', snp_tolerance=True, species='Hsap', threads=1, trnaout='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eschColi-tRNAs.out', trnas='/Users/krdav/Google Drive/MCB/Sullivan_lab/tRNA_charging/mim-tRNAseq/data/hg19-eColitK/hg19_eColitK.fa')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "args"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
